{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import pickle\n",
    "\n",
    "from indra.literature.adeft_tools import universal_extract_text\n",
    "from indra.databases.hgnc_client import get_hgnc_name, get_hgnc_id\n",
    "\n",
    "from indra_db.util.content_scripts import get_text_content_from_pmids\n",
    "from indra_db.util.content_scripts import get_stmts_with_agent_text_like\n",
    "from indra_db.util.content_scripts import get_text_content_from_stmt_ids\n",
    "\n",
    "from adeft.discover import AdeftMiner\n",
    "from adeft.gui import ground_with_gui\n",
    "from adeft.modeling.label import AdeftLabeler\n",
    "from adeft.modeling.classify import AdeftClassifier\n",
    "from adeft.disambiguate import AdeftDisambiguator\n",
    "\n",
    "from adeft_indra.s3 import model_to_s3\n",
    "from adeft_indra.ground import gilda_ground"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [],
   "source": [
    "shortforms = ['LSK']\n",
    "genes = ['LCK']\n",
    "families = {}\n",
    "groundings = [f'HGNC:{get_hgnc_id(gene)}' for gene in genes]\n",
    "for family, members in families.items():\n",
    "    genes.extend(members)\n",
    "    groundings.append(f'FPLX:{family}')\n",
    "with open('../data/entrez_all_pmids.json', 'r') as f:\n",
    "    all_pmids = json.load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [],
   "source": [
    "entrez_texts = []\n",
    "entrez_refs = set()\n",
    "for gene, grounding in zip(genes, groundings):\n",
    "    try:\n",
    "        pmids = all_pmids[gene]\n",
    "    except KeyError:\n",
    "        continue\n",
    "    _, content = get_text_content_from_pmids(pmids)\n",
    "    entrez_texts.extend([(universal_extract_text(text), grounding)\n",
    "                          for text in content.values() if text])\n",
    "    entrez_refs.update(content.keys())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [],
   "source": [
    "miners = dict()\n",
    "all_texts = set()\n",
    "for shortform in shortforms:\n",
    "    stmts = get_stmts_with_agent_text_like(shortform)[shortform]\n",
    "    _, content = get_text_content_from_stmt_ids(stmts)\n",
    "    shortform_texts = [universal_extract_text(text, contains=shortforms)\n",
    "                       for ref, text in content.items() if text and ref not in entrez_refs]\n",
    "    miners[shortform] = AdeftMiner(shortform)\n",
    "    miners[shortform].process_texts(shortform_texts)\n",
    "    all_texts.union(shortform_texts)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "It's then necessary to check if Acromine produced the correct results. We must fix errors manually"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [],
   "source": [
    "top = miners['LSK'].top()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('+', 83.90816326530613),\n",
       " ('sca 1 + c kit +', 51.69230769230769),\n",
       " ('+ c kit +', 42.89830508474576),\n",
       " ('kit +', 37.56944444444444),\n",
       " ('lin − sca 1 + c kit +', 32.72222222222222),\n",
       " ('− sca 1 + c kit +', 26.526315789473685),\n",
       " ('+ ckit +', 21.21212121212121),\n",
       " ('lineage − sca 1 + c kit +', 18.571428571428573),\n",
       " ('lin − sca1 + c kit +', 14.125),\n",
       " ('+ kit +', 13.523809523809524),\n",
       " ('lin sca 1 + c kit +', 13.124999999999998),\n",
       " ('cells', 12.588235294117647),\n",
       " ('sca1 + c kit +', 11.333333333333334),\n",
       " ('sca1 + ckit +', 9.571428571428571),\n",
       " ('c kit +', 7.836065573770497),\n",
       " ('lin − sca 1 + ckit +', 6.0),\n",
       " ('hspcs', 5.5),\n",
       " ('of lin − sca 1 + c kit +', 5.428571428571429),\n",
       " ('− sca 1 + ckit +', 5.090909090909091),\n",
       " ('lin − sca1 + ckit +', 5.0),\n",
       " ('sca1 + kit +', 4.857142857142858),\n",
       " ('sca 1 + kit +', 4.8),\n",
       " ('progenitor', 4.0),\n",
       " ('lin − sca 1 + kit +', 4.0),\n",
       " ('lin − sca + ckit +', 3.6),\n",
       " ('lineage sca 1 + c kit +', 3.6),\n",
       " ('the lin − sca 1 + c kit +', 3.6),\n",
       " ('hspc', 3.0),\n",
       " ('lineage − sca 1 + ckit +', 3.0),\n",
       " ('+ cells', 2.8571428571428568),\n",
       " ('+ sca 1 +', 2.8571428571428568),\n",
       " ('c kit + sca 1 +', 2.8),\n",
       " ('of lin sca 1 + c kit +', 2.8),\n",
       " ('of hspcs', 2.5),\n",
       " ('negative sca 1 + c kit +', 2.5),\n",
       " ('expression in lineage − sca 1 + c kit +', 2.5),\n",
       " ('sca 1 + c kit + cells', 2.4),\n",
       " ('lin sca1 + c kit +', 2.0),\n",
       " ('of lin sca 1 + kit +', 2.0),\n",
       " ('the lin sca 1 + c kit +', 2.0),\n",
       " ('lineage lin − sca 1 + c kit +', 2.0),\n",
       " ('the absolute number of lin − sca1 + c kit +', 2.0),\n",
       " ('principal component analysis of fraction lt hsc ns gfp high and ns gfp low and hspc cells',\n",
       "  2),\n",
       " ('cells frequency and number are unaltered in nfil3 − − mouse bone marrow lin − ckit + sca1 +',\n",
       "  2),\n",
       " ('ing ryk on wnt5a mediated hematopoietic repopulation by ex vivo expanded lin − sca 1 + c kit +',\n",
       "  2),\n",
       " ('sorted hspc populations by taqman rt qpcr lin bm lineage negative bone marrow cells l − s + k +',\n",
       "  2),\n",
       " ('enance of the hsc pool a mean difference plots of mrna expression in lineage − sca 1 + c kit +',\n",
       "  2),\n",
       " ('pb mononuclear cells mncs b lineage lin − and lin + cells c lin − sca 1 + c kit +',\n",
       "  2),\n",
       " ('ckit +', 1.941176470588232),\n",
       " ('− sca1 + c kit +', 1.882352941176471),\n",
       " ('sca 1 + ckit +', 1.833333333333334),\n",
       " ('sca + ckit +', 1.7142857142857144),\n",
       " ('− sca1 + ckit +', 1.7142857142857144),\n",
       " ('− sca + ckit +', 1.666666666666667),\n",
       " ('− sca 1 + kit +', 1.666666666666667),\n",
       " ('low sca 1 + c kit +', 1.6),\n",
       " ('in lineage − sca 1 + c kit +', 1.6),\n",
       " ('positive', 1.5),\n",
       " ('progenitor cells', 1.5),\n",
       " ('− ckit + sca1 +', 1.5),\n",
       " ('lin sca 1 + kit +', 1.5),\n",
       " ('of lin − sca1 + c kit +', 1.5),\n",
       " ('− low sca 1 + c kit +', 1.5),\n",
       " ('hi', 1.3333333333333333),\n",
       " ('populations', 1.3333333333333333),\n",
       " ('kit positive', 1.3333333333333333),\n",
       " ('hematopoietic progenitor cells', 1.3333333333333333),\n",
       " ('− sca1 + kit +', 1.3333333333333333),\n",
       " ('lin − ckit + sca1 +', 1.3333333333333333),\n",
       " ('− c kit + sca 1 +', 1.3333333333333333),\n",
       " ('− sca 1 + c kit + cells', 1.3333333333333333),\n",
       " ('containing lin − sca 1 + c kit +', 1.3333333333333333),\n",
       " ('lineage − low sca 1 + c kit +', 1.3333333333333333),\n",
       " ('of lineage − sca 1 + c kit +', 1.3333333333333333),\n",
       " ('and flow cytometric analyses of lin sca 1 + c kit +', 1.3333333333333333),\n",
       " ('hsc', 1.0),\n",
       " ('pos', 1.0),\n",
       " ('lumbosacral kyphosis', 1.0),\n",
       " ('progenitor populations', 1.0),\n",
       " ('stem cells', 1.0),\n",
       " ('c kit hi', 1.0),\n",
       " ('c kit positive', 1.0),\n",
       " ('expansion of hspcs', 1.0),\n",
       " ('of hematopoietic progenitor cells', 1.0),\n",
       " ('+ c k it +', 1.0),\n",
       " ('+ sca 1 + cells', 1.0),\n",
       " ('lin sca + kit +', 1.0),\n",
       " ('lin sca1 + kit +', 1.0),\n",
       " ('sca 1 and c kit', 1.0),\n",
       " ('lin neg sca1 + ckit +', 1.0),\n",
       " ('lin − sca1 + kit +', 1.0),\n",
       " ('lineage negative sca1 + ckit +', 1.0),\n",
       " ('lineage sca 1 + cd117 +', 1.0),\n",
       " ('sca 1 + c kit high', 1.0),\n",
       " ('lin − ckit + sca 1 +', 1.0),\n",
       " ('lin − sca1 + + kit +', 1.0),\n",
       " ('lineage negative sca1 + c kit +', 1.0),\n",
       " ('lineage − low sca1 + ckit +', 1.0),\n",
       " ('of lin sca + c kit +', 1.0),\n",
       " ('of lin − sca + ckit +', 1.0),\n",
       " ('the lineage − sca + ckit +', 1),\n",
       " ('+ lin sca 1 + c kit +', 1.0),\n",
       " ('bm lin − sca1 + c kit +', 1.0),\n",
       " ('lin neg sca 1 + c kit +', 1.0),\n",
       " ('lin sca 1 + c kit + cells', 1.0),\n",
       " ('lineage negative sca 1 + c kit +', 1.0),\n",
       " ('of lineage sca 1 + c kit +', 1.0),\n",
       " ('hsc enriched lin sca 1 + c kit +', 1.0),\n",
       " ('lin − sca 1 + c kit + cells', 1.0),\n",
       " ('lineage marker negative sca 1 + c kit +', 1),\n",
       " ('the lin − c kit + sca 1 +', 1.0),\n",
       " ('hsc containing lin − sca 1 + c kit +', 1.0),\n",
       " ('hsc enriched lin − sca 1 + c kit +', 1.0),\n",
       " ('lineage negative lin − sca 1 + c kit +', 1.0),\n",
       " ('murine bone marrow lin − sca 1 + ckit +', 1),\n",
       " ('number of lineage − sca 1 + c kit +', 1.0),\n",
       " ('percentage of lin − sca 1 + c kit +', 1.0),\n",
       " ('populations of lin − sca 1 + c kit +', 1.0),\n",
       " ('the lineage − low sca 1 + c kit +', 1.0),\n",
       " ('within the lin − sca 1 + c kit +', 1.0),\n",
       " ('expansion of only lineage − sca 1 + c kit +', 1.0),\n",
       " ('for all experiments lineage − sca 1 + c kit +', 1),\n",
       " ('astic resorption with alendronate treatment resulted in expansion of hematopoietic progenitor cells',\n",
       "  1),\n",
       " ('cd150 + cd48 − lineage − sca 1 + c kit +', 1.0),\n",
       " ('revealed with secondary streptavidin allophycocyanin or phycoerythrin cy5 pe cy5 antibodies hsc',\n",
       "  1),\n",
       " ('sis of bone marrow revealed significant increased proliferation of hematopoietic progenitor cells',\n",
       "  1),\n",
       " ('group and flow cytometric analyses of lin sca 1 + c kit +', 1.0),\n",
       " ('ntation assays 39 similar mice infection with ehrlichia muris also harbored increased hpsc',\n",
       "  1),\n",
       " ('pt sacral slope ss lumbar lordosis ll sacrofemoral distance sfd and lumbosacral kyphosis',\n",
       "  1),\n",
       " ('tatistically significant decrease in the lineage negative sca1 positive c kit positive populations',\n",
       "  1),\n",
       " ('tly demonstrated the presence of a peculiar function resident hematopoietic stem progenitor cells',\n",
       "  1),\n",
       " ('we then analyzed the bm immature lineage c kit + sca 1 +', 1),\n",
       " ('deletion reversed leukocytosis and splenomegaly and preferential depletion primitive lin sca + kit +',\n",
       "  1),\n",
       " ('g pelvic incidence pelvic tilt pt sacral slope ss lumbar lordosis ll lumbosacral kyphosis',\n",
       "  1),\n",
       " ('ith za in both young and adult mice significant increased the hematopoietic progenitor populations',\n",
       "  1),\n",
       " ('n myeloid cmp and granulocyte macrophage progenitor cells gmp without affecting the stem cells',\n",
       "  1),\n",
       " ('of various hematopoietic cells lineage absolute number of lin negative sca1 positive kit positive',\n",
       "  1),\n",
       " ('resulted in significant synergistic increased in the incidence and the total number of progenitor',\n",
       "  1),\n",
       " ('udy of several sepsis report that although pseudomonas aeruginosa caused the expansion of hspcs',\n",
       "  1),\n",
       " ('zed that the myelosuppression occurring during sepsis was caused by dysfunctional expansion of hspc',\n",
       "  1),\n",
       " ('by competitive transplantation assays the authors noticed a marked reduction of csfr3 − − hspcs',\n",
       "  1),\n",
       " ('c kit cd117 fluorescein isothiocyanate conjugated sca1 bd biosciences lineage sca 1 + cd117 +',\n",
       "  1),\n",
       " ('cification as eslsk which manifest a transcripts profile that clustered with the native b6 lsk',\n",
       "  1),\n",
       " ('entative a lung and c bm facs plots of hematopoietic progenitor within the lsk compartment',\n",
       "  1),\n",
       " ('evel attribute in the form of the multidimensional template or tensor used local steering kernel',\n",
       "  1),\n",
       " ('notably inflammatory stress tnf α stimulation activation notch signaling in both wt and fa hspc',\n",
       "  1),\n",
       " ('onfirmed the surface marker profile of hscs following infection we measured the frequency of hspcs',\n",
       "  1),\n",
       " ('progenitor populations consistently increased these including the following lin − sca 1 + ckit +',\n",
       "  1),\n",
       " ('rformed murine bone marrow transplantation experiments with manipulated lineage − sca 1 + ckit +',\n",
       "  1),\n",
       " ('this further ruled out any possibility contamination from cells expression sca1 in the perturbed hsc',\n",
       "  1),\n",
       " ('y and imk populations of the 13 000 transcripts expression in our multipotent progenitor populations',\n",
       "  1),\n",
       " ('25 gene showed a significant difference in expression levels or frequency between lt hscs and hspcs',\n",
       "  1),\n",
       " ('are with evs control and wt transplantation whereas the difference in the frequency of stem progenitor',\n",
       "  1),\n",
       " ('c we performed limiting dilution competitive transplantation assays of murine lin sca1 + c kit +',\n",
       "  1),\n",
       " ('differentiation we performed genome wide gene expression analysis of lineage sca 1 + c kit +',\n",
       "  1),\n",
       " ('fig 2 a furthermore multipotential progenitor mpps and lineage negative sca 1 + c kit +',\n",
       "  1),\n",
       " ('function blocking antibodies both integrin were ubiquitously expression in fl lin sca 1 + kit +',\n",
       "  1),\n",
       " ('of cdkna1a in fancd2 deficient human lymphoblast cells line pd20 and mouse fancd2 − − hspc',\n",
       "  1),\n",
       " ('of committed erythroid myeloid progenitor and expansion of the lineage negative l sca1 + kit +',\n",
       "  1),\n",
       " ('oietic function of these cells interactions between culture cd166 + lineage sca 1 + cd117 +',\n",
       "  1),\n",
       " ('rmate control bone marrow fig 2b however the lineage negative sca 1 positive c kit positive',\n",
       "  1),\n",
       " ('rradiation irradiated caused rapid and prolonged depletion of immature lineage sca 1 + c kit +',\n",
       "  1),\n",
       " ('tlr4 and tlr2 are expression on the surface of lineage − low sca1 + ckit +',\n",
       "  1),\n",
       " ('try analysis confirmed the effect of diabetes on reduced the relative frequency of lineage neg sk',\n",
       "  1),\n",
       " ('< 0 05 a s1pr1 mrna expression in lineage committed lin + non committed lin neg hspcs',\n",
       "  1),\n",
       " ('adhesion to bm mononuclear cells flow cytometry analysis of lineage neg c kit pos sca 1 pos',\n",
       "  1),\n",
       " ('after the last injected demonstrated significant reduced percentage of lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('ak2 v617f mice showed significant expansion of hematopoietic stem cells hsc lin − sca1 + ckit +',\n",
       "  1),\n",
       " ('also drives the transient proliferation and expansion of bone marrow lin negative sca 1 + ckit +',\n",
       "  1),\n",
       " ('also transplantation in sublethally irradiated syngeneic recipient via both lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('bm cells percentage shown are of parental gated lower plots subgated dapi lin kit + sca +',\n",
       "  1),\n",
       " ('caused these animals die postnatally we first profile the hsc containing lineage − sca1 + kit +',\n",
       "  1),\n",
       " ('cells populations wnt5a has been shown to activation non canonical signaling in lin sca1 + ckit hi',\n",
       "  1),\n",
       " ('ed with pe anti cd45 1 we have previously shown that id1 expression in hsc multipotent progenitor',\n",
       "  1),\n",
       " ('esentative flow cytometry determined of cells surface expression of mpl in lin neg sca1 + kit +',\n",
       "  1),\n",
       " ('here we demonstrated genetic determined variation in the proliferation of lin − sca1 + + kit +',\n",
       "  1),\n",
       " ('itive hematopoietic cells the frequency of a phenotypic defined populations of lin sca + c kit +',\n",
       "  1),\n",
       " ('kemic cd45 2 + gfp negative donor cells did not change significant we conclude that both immature',\n",
       "  1),\n",
       " ('knockdown vectors were introduced by lentiviral transfer i e transduction to lin − sca + ckit +',\n",
       "  1),\n",
       " ('lt hscs a principal component analysis indicated that ns gfp high lt hscs are distinct from hspcs',\n",
       "  1),\n",
       " ('ng capacity 80 finally a recent report showed that il 27 exposure leads to expansion of hspcs',\n",
       "  1),\n",
       " ('oid cells bcap was expression in bm hematopoietic progenitor including lineage sca 1 + c kit +',\n",
       "  1),\n",
       " ('pom apom − − had increased proliferation of lin − sca1 + ckit + hematopoietic progenitor cells',\n",
       "  1),\n",
       " ('s dormancy is crucial to maintain the appropriate hsc pool in this studies we demonstrated that hspc',\n",
       "  1),\n",
       " ('s stemspan cc100 stem cells technologies vancouver bc mouse lineage negative sca 1 + c kit +',\n",
       "  1),\n",
       " ('total rna was isolated from 2 5x10 4 5x10 5 sorted lineage negative kit positive sca positive',\n",
       "  1),\n",
       " ('tp www peprotech com sorted cells were incubated overnight before reseeding at 5 × 10 4 cells',\n",
       "  1),\n",
       " ('16 a quantitative rt pcr analysis of pu 1 expression in isolated linckit + sca 1 + cells',\n",
       "  1),\n",
       " ('a representative flow cytometry plots of hspc for il 17 production lineage − sca 1 + ckit +',\n",
       "  1),\n",
       " ('aining antibiotics 100 u ml penicillin 100 μg ml streptomycin invitrogen lin − sca1 + c kit +',\n",
       "  1),\n",
       " ('and adhesion molecules as well as chemokine and chemokine receptor in lineage sca 1 + c kit +',\n",
       "  1),\n",
       " ('c identification of lineage negative lin sca 1 positive sca 1 + c kit positive c kit +',\n",
       "  1),\n",
       " ('c to the pb was initially analyses in a dose and time response assays and by quantifying progenitor',\n",
       "  1),\n",
       " ('d3100 or ischemia was evaluated by flow cytometric enumeration of circulating lin − sca 1 + ckit +',\n",
       "  1),\n",
       " ('determined the frequency of several hspc populations these including lineage lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('ed with the im and lbh combination demonstrated significant increased apoptosis of lin sca 1 + kit +',\n",
       "  1),\n",
       " ('eport that hypercholesterolemia increased the percentage of integrin β 2 + lin − sca 1 + ckit +',\n",
       "  1),\n",
       " ('eptor 5 tlr5 depend and nod like receptor c4 independent proliferation of lin sca 1 + kit +',\n",
       "  1),\n",
       " ('erformed qrt pcr and immunohistochemistry on purified murine hscs cd150 + lin − ckit + sca 1 +',\n",
       "  1),\n",
       " ('h our prior report of an adar1 required in the wild type mature lineage positive but not early',\n",
       "  1),\n",
       " ('her myeloid populations in chow fed animals fig 1 15 remarkably the lin − sca + ckit +',\n",
       "  1),\n",
       " ('kd knockdown bm bone marrow bl blood purification and culture of lineage − sca 1 + ckit +',\n",
       "  1),\n",
       " ('marrow il 27 predominantly and continuously promoted expansion of only lineage − sca 1 + c kit +',\n",
       "  1),\n",
       " ('nexin 2 interactions to alter hsc migration was determined for these studies lin − sca1 + ckit +',\n",
       "  1),\n",
       " ('ompartments in bone marrow and spleen as determined by increased frequency of lin − sca1 + ckit +',\n",
       "  1),\n",
       " ('onsible for pth actions in hematopoietic cells expansion pth increased the lin sca 1 + c kit +',\n",
       "  1),\n",
       " ('ost reconstituted with marrow progenitor transduce with migri icn1 4 b gfp + lin sca1 c kit +',\n",
       "  1),\n",
       " ('owever there was a statistically significant increased in the frequency of both lin − sca + ckit +',\n",
       "  1),\n",
       " ('peripheral nk cells analysis in recipient mice reconstituted by ryk deficient lsks lin − sca1 + kit +',\n",
       "  1),\n",
       " ('r cells hspcs abca1 and abcg1 deletion significant increased the number of hspcs lin sca + ckit +',\n",
       "  1),\n",
       " ('reventing gmp formation associated with accumulation of upstream cmp and the lin sca 1 + c kit +',\n",
       "  1),\n",
       " ('rogenitor fraction mp a cells populations that originates directly from lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('s of cd45 2 + lsk were measured by facs representative facs left panel the average cells number',\n",
       "  1),\n",
       " ('stem progenitor cells either by isolated large number of heterogeneous lineage − sca 1 + c kit +',\n",
       "  1),\n",
       " ('subsets we first examined multiple histone methylation marked in murine bm lin − sca1 + c kit +',\n",
       "  1),\n",
       " ('t al 2001 czabotar et al 2014 in mouse lineage marker negative sca 1 and c kit positive',\n",
       "  1),\n",
       " ('t3 + lymphoid primed multipotent progenitor lmpp fraction within the lin neg sca 1 + c kit high',\n",
       "  1),\n",
       " ('tion migration adherence and homing to the bm unexpectedly hsc enriched lin − sca 1 + kit +',\n",
       "  1),\n",
       " ('total hematopoietic cells fig 2a b hscs are phenotypic defined as lin − sca1 + c kit +',\n",
       "  1),\n",
       " ('w hematopoiesis g0s2 transcripts are enriched in two hsc populations lineage − sca 1 + c kit +',\n",
       "  1),\n",
       " ('− mice erythrocytes were lysed as previously described hscs that are lin low sca1 + c kit +',\n",
       "  1),\n",
       " ('18 19 however gfi1 − − and p 21 − − display dissimilar lineage negative sca1 + ckit +',\n",
       "  1),\n",
       " ('6 7 as evidenced by the apparent expansion of lineage negative lin neg sca 1 + c kit +',\n",
       "  1),\n",
       " ('an increased in akt activation in freshly isolated grp94 null hsc enriched lin sca 1 + c kit +',\n",
       "  1),\n",
       " ('and frequency of hematopoietic stem and progenitor cells as determined by the lin − sca1 + + kit +',\n",
       "  1),\n",
       " ('armacologic abt 263 loss of bcl xl expression and or activation did not alter bcr abl1 + stem cells',\n",
       "  1),\n",
       " ('atopoietic recovery after 5 fu treatment a we observed a higher frequency of lin − sca + ckit +',\n",
       "  1),\n",
       " ('clearly showed that stromal cells react to the presence of lineage negative lin − sca 1 + kit +',\n",
       "  1),\n",
       " ('cs therefore depend upon a perivascular niche for their maintenance lineage − low sca 1 + c kit +',\n",
       "  1),\n",
       " ('csf or amd3100 and in response to ischemic injury in mice lineage negative sca 1 + and ckit +',\n",
       "  1),\n",
       " ('ction is lost or compromised after hscs in the fetus have form lineage negative sca1 + c kit +',\n",
       "  1),\n",
       " ('dinal sections of mice transplantation with cfse labeled wt or klf5 deficient lin c kit + sca 1 +',\n",
       "  1),\n",
       " ('e have shown that dba congenic animals exhibit an increased percentage of lineage − sca1 + c kit +',\n",
       "  1),\n",
       " ('ed hematopoietic cells populations in blood and bm extended data fig 5a c lin − sca1 + ckit +',\n",
       "  1),\n",
       " ('ene ltd rehovot israel in some cases fluorescence activation cells sorted of lin sca 1 + c kit +',\n",
       "  1),\n",
       " ('etic stem cells the resulted showed that gata 2 over expression increased the lin sca1 + c kit +',\n",
       "  1),\n",
       " ('fig the contribution of ufbp1 deficient ufbp1 f f creert2 cells to lin sca 1 + c kit +',\n",
       "  1),\n",
       " ('first used imagestream to analyzed notch signaling in notch egfp expression in lin − sca1 + c kit +',\n",
       "  1),\n",
       " ('first we found that tnf α stimulation enhance notch signaling in fanca − − and fancc − − hscp',\n",
       "  1),\n",
       " ('fold lower levels compared to non hematopoietic stromal cells in hspc such as lin − sca1 + kit +',\n",
       "  1),\n",
       " ('h go289 significant reduced growth of mouse mll af9 leukemia cells while lin − sca1 + c kit +',\n",
       "  1),\n",
       " ('hscs ex vivo under defined culture conditions we culture mouse c kit + or lin sca1 + ckit +',\n",
       "  1),\n",
       " ('ification of c total bm cells d total splenocytes e total bm lineage − low sca1 + ckit +',\n",
       "  1),\n",
       " ('ing factor gm csf colonies although the proliferation and mobilization of lin c kit + sca 1 + cells',\n",
       "  1),\n",
       " ('models at simulated high altitude the bone marrow bm and spleen lineage marker sca 1 + c kit +',\n",
       "  1),\n",
       " ('nce of immune tolerance remain poorly defined here we report that mobilization lin sca 1 + c kit +',\n",
       "  1),\n",
       " ('nd a dramatic expansion of the stem and progenitor cells populations containing lin − sca 1 + kit +',\n",
       "  1),\n",
       " ('of bcl xl and that of it post transcripts modulator hnrnp a1 37 in mnc and stem cells enriched',\n",
       "  1),\n",
       " ('onsequences of β catenin activation in the hsc enriched lineage negative lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('or to vehicle control in bcr abl induced mouse leukemia cells bearing the phenotypic lin sca1 + kit +',\n",
       "  1),\n",
       " ('reestar inc hematopoietic stem cells populations hsc was defined as lin − il 7rα − ckit + sca1 +',\n",
       "  1),\n",
       " ('rial infection or immune data not shown and similar frequency of lin negative sca 1 + c kit +',\n",
       "  1),\n",
       " ('rogenitor the hsc enriched rag1 gfp − flt3 − lineage marker negative lin − sca1 + c kit hi',\n",
       "  1),\n",
       " ('row il 27 predominantly and continuously promoted the expansion of only lineage − sca 1 + c kit +',\n",
       "  1),\n",
       " ('s buffered salt solution without calcium or magnesium for detected of lineage − sca 1 + c kit +',\n",
       "  1),\n",
       " ('sc compartment has never been analyzed in the bm therefore we analyzed the lin neg sca1 + ckit +',\n",
       "  1),\n",
       " ('shown below the lsken and lsk cells here do not containing mkp and mep mm cells because the former',\n",
       "  1),\n",
       " ('shrna mediated tet2 knockdown led to an accumulation of lineage negative lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('significant reduction in bone marrow cellularity a and the number of lineage − sca 1 + c kit +',\n",
       "  1),\n",
       " ('small fraction of bone marrow cells that lack lineage marker lin − and expression sca 1 and c kit',\n",
       "  1),\n",
       " ('tch egfp fanca − − mice were isolated following by lineage + cells depletion linsca1 + c kit +',\n",
       "  1),\n",
       " ('tors expression sall4a or sall4b isoform were used to transduce mouse bone marrow lin sca1 + c kit +',\n",
       "  1),\n",
       " ('uced retrovirus expression pot1a specific shrnas shpot1a 1 and 2 into lineage − sca 1 + c kit +',\n",
       "  1),\n",
       " ('verexpression of these stat5 mutants but only when the most primitive cd34 − lin − ckit + sca1 +',\n",
       "  1),\n",
       " ('we evaluated the hspc compartment by flow cytometry the absolute number of lin − sca1 + c kit +',\n",
       "  1),\n",
       " ('a circadian rhythm associated 2 fold increased in the number of hsc enriched lin sca 1 + c kit +',\n",
       "  1),\n",
       " ('abca1 − − abcg1 − − mice display abnormal cells autonomous proliferation of lin − sca 1 + kit +',\n",
       "  1),\n",
       " ('abcg1 are both high expression in hspcs and their deletion increased the number of lin − sca + ckit +',\n",
       "  1),\n",
       " ('adouts of progenitor function fig 2 table 1 co culture of murine lin − sca 1 hi c kit hi',\n",
       "  1),\n",
       " ('ct of type i ifn on hsc activation besides ifn α ifn γ can also enhance the proliferation of hspcs',\n",
       "  1),\n",
       " ('ctor hdad gfp and hdad sb we detected gfp marked in bone marrow lineage lin − sca1 + ckit +',\n",
       "  1),\n",
       " ('d mature peripheral blood cells and death within two weeks remarkably the lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('described previously poglio et al 2012 2010 briefly 2 × 10 3 lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('e spleen after rg1 treatment fig 5 c rg1 treatment increased lineage lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('ed diabetes db db mice diabetes resulted in a decrease number of lineage − sca 1 + c kit +',\n",
       "  1),\n",
       " ('eed flow cytometric analysis paralleled the in vivo pet data as more lineage − c kit + sca 1 +',\n",
       "  1),\n",
       " ('emia we now demonstrated that the disease is transplantation used bcr abl + lin sca 1 + c kit +',\n",
       "  1),\n",
       " ('establish genetic pathways that are required by cml but not by normal stem cells the lin sca + kit +',\n",
       "  1),\n",
       " ('etic cells the majority of which were high enriched for expression in lineage − sca 1 + c kit +',\n",
       "  1),\n",
       " ('f prf − − mice with lentiviral gene therapy was performed transduce lin sca 1 + c kit + cells',\n",
       "  1),\n",
       " ('for isolated stem cells had been to obtain the lineage negative lin − c kit + sca 1 + fraction',\n",
       "  1),\n",
       " ('ge committed lin + non lineage committed lin neg cells and hspcs lin neg sca 1 + c kit +',\n",
       "  1),\n",
       " ('gure 1b 1c left panel and total number figure 1c right panel of lin − sca1 + c kit +',\n",
       "  1),\n",
       " ('haecs were exclude from total counted by their large size for lineage lin lo sca 1 + c kit +',\n",
       "  1),\n",
       " ('hscs purified as the cd48 − cd150 + slam cells subpopulation of primitive lin − sca1 + c kit +',\n",
       "  1),\n",
       " ('hspcs in the absence of ackr1 the size of the lineage negative lin sca 1 + c kit + cells',\n",
       "  1),\n",
       " ('hspcs resident in the adult bone marrow bm are containing within the lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('icle for 3 weeks bone marrow was isolated and flow cytometric analyses of lin sca 1 + c kit +',\n",
       "  1),\n",
       " ('in marker such as cd5 ter119 b220 cd11b gr 1 and their varying expression of sca 1 and c kit',\n",
       "  1),\n",
       " ('lethality irradiated 2x 550 rad c57bl 6 or cd45 1 host mice for sorted cells populations 1 × 10 5',\n",
       "  1),\n",
       " ('marrow cellularity e and absolute number of phenotypic cd150 + cd48 − lineage − sca 1 + c kit +',\n",
       "  1),\n",
       " ('mice were driven by lin28b expression in mc precursors themselves we sorted lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('n 58 in animals models deficient in abca1 and abcg1 there is expansion of lin sca 1 + kit +',\n",
       "  1),\n",
       " ('nitor cells hspcs including lt hscs can be identified within the lineage − low sca 1 + c kit +',\n",
       "  1),\n",
       " ('nstrate a cells autonomous role for hhex in b cells development we cocultured lin − sca 1 + kit +',\n",
       "  1),\n",
       " ('observed a significant reduction in the percentage of fetal hspcs within the lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('ometry lineage negative cells were gated and are plots in panel a to highlight the lin sca1 + kit +',\n",
       "  1),\n",
       " ('on demonstrated that tpo alone induced a three fold increased in the number of lineage sca1 + c kit +',\n",
       "  1),\n",
       " ('p cml suggested that lic activation is confined to bcr abl expression lineage lin − sca1 + c kit +',\n",
       "  1),\n",
       " ('poietic effector moser et al 2008 2009b facs purified k3 + + lineage − sca 1 + c kit +',\n",
       "  1),\n",
       " ('pth suggested that il 6 mediated the pth effect in vivo pth increased lin sca 1 + c kit +',\n",
       "  1),\n",
       " ('r we could engineer the genome of primary murine hspcs ex vivo we isolated lineage − sca1 + ckit +',\n",
       "  1),\n",
       " ('re counted the microarray data set gse69760 from bm and fetal liver lineage − sca 1 + c kit +',\n",
       "  1),\n",
       " ('re myeloid and lymphoid cells 1 2 3 4 the hsc containing lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('remain unclear here we showed that pad4 which is high expression in lineage − sca 1 + c kit +',\n",
       "  1),\n",
       " ('s of function mutation 77 oc oc mice display a 50 fold reduction in lin neg sca1 + ckit +',\n",
       "  1),\n",
       " ('se lethality was a resulted of bone marrow failure since transplantation of bm lin − sca1 + c kit +',\n",
       "  1),\n",
       " ('swirski and nahrendorf 2013 here we described a cd150 + cd48 − lineage − sca 1 + c kit +',\n",
       "  1),\n",
       " ('terleukin 1 receptor 1 il 1r1 expression on hematopoietic hsc and lin − sca 1 + c kit + cells',\n",
       "  1),\n",
       " ('udy of gene expression change of ug26 1b6 stromal cells in contact with lineage − sca 1 + kit +',\n",
       "  1),\n",
       " ('valuation of the percentage of hspcs in bm as expected the percentage of lin sca 1 + c kit +',\n",
       "  1),\n",
       " ('when hsc enriched lineage negative sca 1 positive c kit positive cells or lin − sca + 1c kit +',\n",
       "  1),\n",
       " ('with this possibility by 180 day post transplantation icn4 animals lack gfp + lin − sca1 − c kit +',\n",
       "  1),\n",
       " ('06 in the adult mouse all multipotent cells are containing in the lineage − low sca 1 + c kit +',\n",
       "  1),\n",
       " ('1 cells production but not those of th2 cells caused a rapid expansion of lineage sca 1 + c kit +',\n",
       "  1),\n",
       " ('alysis also showed a rapid and dramatic decrease in the number of immature lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('arhgap21 + − mice a b flow cytometry analysis showed increased frequency of lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('bs both fetal liver and bone marrow hematopoiesis by reduced the number of lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('c myc expression we report here that pad4 is expression in bone marrow lineage − sca 1 + c kit +',\n",
       "  1),\n",
       " ('ed a marked decrease in the frequency and absolute number of hsc enriched lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('elated tlr ligands on hspc from normal and lupus prone mice the hspc are lineage − sca 1 + ckit +',\n",
       "  1),\n",
       " ('f δψm see methods the key for the right three panel is shown in d the gated area for lsk cells',\n",
       "  1),\n",
       " ('imately give rise to all red and white blood cells including platelets the lin − c kit + sca 1 +',\n",
       "  1),\n",
       " ('injected on the same day with whole mononuclear bm cells or isolated lin − or lin − sca1 + ckit +',\n",
       "  1),\n",
       " ('ion at × 60 magnification a parp1 activation in hspcs in response to mmc lin − sca1 + c kit +',\n",
       "  1),\n",
       " ('l preparation and analysis were performed as following mouse hscs were defined as lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('levels of pro angiogenic sca 1 pos cd11b pos cells figure 7a and lin neg sca 1 pos c kit pos',\n",
       "  1),\n",
       " ('opulation of the bcr abl induced cml like disease has been identified as lineage − sca 1 + c kit +',\n",
       "  1),\n",
       " ('pc populations in the bone marrow the hspc pool size defined by the percentage of lin sca + c kit +',\n",
       "  1),\n",
       " ('troma cells 18 and negative regulates stem cells pool size and function of lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('uv fixable viability kit was used for dead cells exclusion a populations of lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('y metabolism we measured the rates of glycolysis and oxphos in hsc enriched lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('7 in the control of malaria infection il 27 acted on and expanded lineage lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('a flow cytometric analysis of hspc in bm on day 7 the absolute number of lin − sca1 + c kit +',\n",
       "  1),\n",
       " ('ction of pu 1 expression compared to wild type wt littermates 37 ± 8 in lin − sca 1 + ckit +',\n",
       "  1),\n",
       " ('cy seen in mef null mice so we examined p53 mrna expression in primitive lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('d bone marrow cells for gated please see fig 2a we found that hsc lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('e we investigate the role of p selectin plays on hsc and lsc containing lin low sca 1 + c kit +',\n",
       "  1),\n",
       " ('fig 2 a first there was a tenfold increased in the stem cells containing lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('he number of colonies form unit cells cfu cs white blood cells wbcs and lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('ith the higher dose of evs we observed a preferential uptake of aft evs by lin − sca1 + c kit +',\n",
       "  1),\n",
       " ('l suspension to exclude dead cells the stained bm cells were sorted for lin − low sca 1 + c kit +',\n",
       "  1),\n",
       " ('nd flcn ko donor bmc at day 21 after pipc treatment the absolute number of lin − sca1 + c kit +',\n",
       "  1),\n",
       " ('o lps with a significant increased in the frequency and absolute number of bm lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('of c kit + sca 1 + square gated are drawn on lin negative cells expression both c kit and sca 1',\n",
       "  1),\n",
       " ('s in the bim − − lsk cells supporting information fig 1b murine lin − sca 1 + c kit + cells',\n",
       "  1),\n",
       " ('scular repair in diabetes in mouse models of type 1 and type 2 diabetes lineage − sca 1 + c kit +',\n",
       "  1),\n",
       " ('t increased in cells number 1 5 fold p < 0 005 was found in the lineage negative sca1 + ckit +',\n",
       "  1),\n",
       " ('the smo − − cml mice showed a significant decrease in the populations of lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('tify hscs based on phenotypic marker in the mouse hscs are found in the lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('y ligation in the femur cd150 + cd48 − hsc figure 1a and lineage − sca 1 + c kit + cells',\n",
       "  1),\n",
       " ('yzed with antibodies against c kit sca 1 cd150 and cd48 figure 3b the lin − c kit + sca 1 +',\n",
       "  1),\n",
       " ('110β ablation could rescue hscs in pten δ δ mice the number of both the lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('beckman coulter brea ca on day 14 and day 21 the upper 30 of notch egfp lin − sca1 + c kit +',\n",
       "  1),\n",
       " ('d type and il 6 deficient mice n ≥ 7 group and flow cytometric analyses of lin sca 1 + c kit +',\n",
       "  1),\n",
       " ('es whereas both matn4 − − and matn 1 4 − − mice presented with decrease lin − sca 1 + ckit +',\n",
       "  1),\n",
       " ('ic cells moreover epcr + long term lt hscs were enriched in the n cad + lin sca 1 + c kit +',\n",
       "  1),\n",
       " ('ive lin and lin c kit + lk populations 85 90 of the cells in the lin sca 1 + c kit +',\n",
       "  1),\n",
       " ('lin − sca 1 + c kit lsk − cells populations derived from hsc containing lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('ls after 5 th round of plating d h transplantation of ure het msh2 − − lin − ckit + sca 1 +',\n",
       "  1),\n",
       " ('nd npy 21 36 to mobilization hspcs in the bm as expected the percentage of lin − sca 1 + c kit +',\n",
       "  1),\n",
       " ('sca 1 cd34 and cd16 32 hscs derived from donor were gfp + and lin − sca 1 + and c kit +',\n",
       "  1),\n",
       " ('that received either normal or high fat diet for 2 months quantification of l in − s ca1 + c k it +',\n",
       "  1),\n",
       " ('und that transcripts of g 0 g 1 switch gene 2 g0s2 are enriched in lineage − sca 1 + c kit +',\n",
       "  1),\n",
       " ('bm niche 4 5 and are large quiescent in general hspc are defined as l in s ca 1 + c k it +',\n",
       "  1),\n",
       " ('d il 6 deficient mice il 6 ko n ≥ 9 group and flow cytometric analyses of lin sca 1 + c kit +',\n",
       "  1),\n",
       " ('one way anova p < 0 05 p < 0 01 p < 0 005 p < 0 001 a analysis of pb progenitor',\n",
       "  1),\n",
       " ('ed as previously described 27 28 for lin sca 1 − c kit + c kit + and lin sca 1 + c kit +',\n",
       "  1),\n",
       " ('150 high ckit + sca 1 + lin − il7rα − flt3 − cd48 − myeloid biased hsc or lin − sca 1 + ckit +',\n",
       "  1),\n",
       " ('b facs analysis of bm cells of p53 − − mysm1 − − dko and wt mice for lin − sca 1 + c kit high',\n",
       "  1),\n",
       " ('on cells that are cd150 + lin − cd48 − cd4 − sca1 + c kit + in some cases lin − sca1 + c kit +',\n",
       "  1),\n",
       " ('1', 0.0),\n",
       " ('5', 0.0),\n",
       " ('compartment', 0.0),\n",
       " ('early', 0.0),\n",
       " ('enriched', 0.0),\n",
       " ('former', 0.0),\n",
       " ('fraction', 0.0),\n",
       " ('high', 0.0),\n",
       " ('hpsc', 0.0),\n",
       " ('hscp', 0.0),\n",
       " ('immature', 0.0),\n",
       " ('kernel', 0.0),\n",
       " ('kit', 0.0),\n",
       " ('kyphosis', 0.0),\n",
       " ('lsk', 0.0),\n",
       " ('number', 0.0),\n",
       " ('sk', 0.0),\n",
       " ('+ fraction', 0.0),\n",
       " ('1 +', 0.0),\n",
       " ('1 pos', 0.0),\n",
       " ('10 5', 0.0),\n",
       " ('4 cells', 0.0),\n",
       " ('and hspcs', 0.0),\n",
       " ('antibodies hsc', 0.0),\n",
       " ('b6 lsk', 0.0),\n",
       " ('both immature', 0.0),\n",
       " ('c kit', 0.0),\n",
       " ('cd117 +', 0.0),\n",
       " ('cells enriched', 0.0),\n",
       " ('cells number', 0.0),\n",
       " ('ckit hi', 0.0),\n",
       " ('fa hspc', 0.0),\n",
       " ('from hspcs', 0.0),\n",
       " ('hspc cells', 0.0),\n",
       " ('increased hpsc', 0.0),\n",
       " ('it +', 0.0),\n",
       " ('k +', 0.0),\n",
       " ('kit hi', 0.0),\n",
       " ('kit high', 0.0),\n",
       " ('kit pos', 0.0),\n",
       " ('lsk cells', 0.0),\n",
       " ('lsk compartment', 0.0),\n",
       " ('multipotent progenitor', 0.0),\n",
       " ('neg hspcs', 0.0),\n",
       " ('neg sk', 0.0),\n",
       " ('not early', 0.0),\n",
       " ('of hspc', 0.0),\n",
       " ('of progenitor', 0.0),\n",
       " ('pb progenitor', 0.0),\n",
       " ('perturbed hsc', 0.0),\n",
       " ('positive populations', 0.0),\n",
       " ('quantifying progenitor', 0.0),\n",
       " ('sca +', 0.0),\n",
       " ('sca 1', 0.0),\n",
       " ('sca positive', 0.0),\n",
       " ('sca1 +', 0.0),\n",
       " ('steering kernel', 0.0),\n",
       " ('stem progenitor', 0.0),\n",
       " ('that hspc', 0.0),\n",
       " ('the former', 0.0),\n",
       " ('− hscp', 0.0),\n",
       " ('− hspc', 0.0),\n",
       " ('− hspcs', 0.0),\n",
       " ('+ cd117 +', 0.0),\n",
       " ('+ ckit hi', 0.0),\n",
       " ('+ k +', 0.0),\n",
       " ('+ sca +', 0.0),\n",
       " ('+ sca1 +', 0.0),\n",
       " ('+ stem cells', 0.0),\n",
       " ('1 + cells', 0.0),\n",
       " ('1 + fraction', 0.0),\n",
       " ('10 4 cells', 0.0),\n",
       " ('1c kit +', 0.0),\n",
       " ('and c kit', 0.0),\n",
       " ('and ckit +', 0.0),\n",
       " ('and fa hspc', 0.0),\n",
       " ('and hspc cells', 0.0),\n",
       " ('and lumbosacral kyphosis', 0.0),\n",
       " ('and sca 1', 0.0),\n",
       " ('average cells number', 0.0),\n",
       " ('because the former', 0.0),\n",
       " ('but not early', 0.0),\n",
       " ('by quantifying progenitor', 0.0),\n",
       " ('c kit high', 0.0),\n",
       " ('c kit pos', 0.0),\n",
       " ('cy5 antibodies hsc', 0.0),\n",
       " ('demonstrated that hspc', 0.0),\n",
       " ('distinct from hspcs', 0.0),\n",
       " ('expansion of hspc', 0.0),\n",
       " ('for lsk cells', 0.0),\n",
       " ('frequency of hspcs', 0.0),\n",
       " ('harbored increased hpsc', 0.0),\n",
       " ('hematopoietic progenitor populations', 0.0),\n",
       " ('hsc multipotent progenitor', 0.0),\n",
       " ('hscs and hspcs', 0.0),\n",
       " ('k it +', 0.0),\n",
       " ('kit + cells', 0.0),\n",
       " ('kit positive populations', 0.0),\n",
       " ('lin neg hspcs', 0.0),\n",
       " ('lineage neg sk', 0.0),\n",
       " ('ll lumbosacral kyphosis', 0.0),\n",
       " ('local steering kernel', 0.0),\n",
       " ('multipotent progenitor populations', 0.0),\n",
       " ('native b6 lsk', 0.0),\n",
       " ('number of progenitor', 0.0),\n",
       " ('of pb progenitor', 0.0),\n",
       " ('of stem progenitor', 0.0),\n",
       " ('positive kit positive', 0.0),\n",
       " ('positive sca positive', 0.0),\n",
       " ('proliferation of hspcs', 0.0),\n",
       " ('sca 1 +', 0.0),\n",
       " ('sca 1 pos', 0.0),\n",
       " ('stem cells enriched', 0.0),\n",
       " ('stem progenitor cells', 0.0),\n",
       " ('that both immature', 0.0),\n",
       " ('the lsk compartment', 0.0),\n",
       " ('the perturbed hsc', 0.0),\n",
       " ('the stem cells', 0.0),\n",
       " ('× 10 5', 0.0),\n",
       " ('− − hscp', 0.0),\n",
       " ('− − hspc', 0.0),\n",
       " ('− − hspcs', 0.0),\n",
       " ('+ + kit +', 0.0),\n",
       " ('+ 1c kit +', 0.0),\n",
       " ('+ and ckit +', 0.0),\n",
       " ('+ c kit hi', 0.0),\n",
       " ('+ c kit high', 0.0),\n",
       " ('+ hematopoietic progenitor cells', 0.0),\n",
       " ('1 + cd117 +', 0.0),\n",
       " ('1 + ckit +', 0.0),\n",
       " ('1 + kit +', 0.0),\n",
       " ('1 and c kit', 0.0),\n",
       " ('1 × 10 5', 0.0),\n",
       " ('abl1 + stem cells', 0.0),\n",
       " ('affecting the stem cells', 0.0),\n",
       " ('also harbored increased hpsc', 0.0),\n",
       " ('analysis of pb progenitor', 0.0),\n",
       " ('and by quantifying progenitor', 0.0),\n",
       " ('and c kit +', 0.0),\n",
       " ('and c kit positive', 0.0),\n",
       " ('and stem cells enriched', 0.0),\n",
       " ('are distinct from hspcs', 0.0),\n",
       " ('area for lsk cells', 0.0),\n",
       " ('c k it +', 0.0),\n",
       " ('c kit + cells', 0.0),\n",
       " ('c kit positive populations', 0.0),\n",
       " ('cells because the former', 0.0),\n",
       " ('ckit + sca1 +', 0.0),\n",
       " ('committed lin neg hspcs', 0.0),\n",
       " ('conclude that both immature', 0.0),\n",
       " ('csfr3 − − hspcs', 0.0),\n",
       " ('dysfunctional expansion of hspc', 0.0),\n",
       " ('fancc − − hscp', 0.0),\n",
       " ('fancd2 − − hspc', 0.0),\n",
       " ('frequency of stem progenitor', 0.0),\n",
       " ('hematopoietic stem progenitor cells', 0.0),\n",
       " ('hi c kit hi', 0.0),\n",
       " ('in hsc multipotent progenitor', 0.0),\n",
       " ('in the perturbed hsc', 0.0),\n",
       " ('kit + sca +', 0.0),\n",
       " ('kit and sca 1', 0.0),\n",
       " ('kit positive sca positive', 0.0),\n",
       " ('lordosis ll lumbosacral kyphosis', 0.0),\n",
       " ('low and hspc cells', 0.0),\n",
       " ('lt hscs and hspcs', 0.0),\n",
       " ('of lineage neg sk', 0.0),\n",
       " ('our multipotent progenitor populations', 0.0),\n",
       " ('pe cy5 antibodies hsc', 0.0),\n",
       " ('pos c kit pos', 0.0),\n",
       " ('pos sca 1 pos', 0.0),\n",
       " ('positive but not early', 0.0),\n",
       " ('positive c kit +', 0.0),\n",
       " ('positive c kit positive', 0.0),\n",
       " ('s + k +', 0.0),\n",
       " ('sca + kit +', 0.0),\n",
       " ('sca 1 + cells', 0.0),\n",
       " ('sca 1 + fraction', 0.0),\n",
       " ('sca1 + ckit hi', 0.0),\n",
       " ('sca1 c kit +', 0.0),\n",
       " ('sca1 positive kit positive', 0.0),\n",
       " ('sfd and lumbosacral kyphosis', 0.0),\n",
       " ('the average cells number', 0.0),\n",
       " ('the expansion of hspcs', 0.0),\n",
       " ('the frequency of hspcs', 0.0),\n",
       " ('the hematopoietic progenitor populations', 0.0),\n",
       " ('the native b6 lsk', 0.0),\n",
       " ('the proliferation of hspcs', 0.0),\n",
       " ('to expansion of hspcs', 0.0),\n",
       " ('total number of progenitor', 0.0),\n",
       " ('used local steering kernel', 0.0),\n",
       " ('we demonstrated that hspc', 0.0),\n",
       " ('within the lsk compartment', 0.0),\n",
       " ('wt and fa hspc', 0.0),\n",
       " ('× 10 4 cells', 0.0),\n",
       " ('− c kit +', 0.0),\n",
       " ('+ and c kit +', 0.0),\n",
       " ('+ c kit + cells', 0.0),\n",
       " ('+ sca 1 + fraction', 0.0),\n",
       " ('1 + and ckit +', 0.0),\n",
       " ('1 + c kit +', 0.0),\n",
       " ('1 + c kit high', 0.0),\n",
       " ('1 and c kit positive', 0.0),\n",
       " ('1 hi c kit hi', 0.0),\n",
       " ('1 pos c kit pos', 0.0),\n",
       " ('1 positive c kit positive', 0.0),\n",
       " ('5 × 10 4 cells', 0.0),\n",
       " ('a analysis of pb progenitor', 0.0),\n",
       " ('and fancc − − hscp', 0.0),\n",
       " ('assays and by quantifying progenitor', 0.0),\n",
       " ('bcr abl1 + stem cells', 0.0),\n",
       " ('between lt hscs and hspcs', 0.0),\n",
       " ('both wt and fa hspc', 0.0),\n",
       " ('by dysfunctional expansion of hspc', 0.0),\n",
       " ('c kit and sca 1', 0.0),\n",
       " ('caused the expansion of hspcs', 0.0),\n",
       " ('ckit + hematopoietic progenitor cells', 0.0),\n",
       " ('ckit + sca 1 +', 0.0),\n",
       " ('cy5 pe cy5 antibodies hsc', 0.0),\n",
       " ('distance sfd and lumbosacral kyphosis', 0.0),\n",
       " ('enhance the proliferation of hspcs', 0.0),\n",
       " ('expansion of hematopoietic progenitor cells', 0.0),\n",
       " ('expression in hsc multipotent progenitor', 0.0),\n",
       " ('frequency of lineage neg sk', 0.0),\n",
       " ('gated area for lsk cells', 0.0),\n",
       " ('gfp low and hspc cells', 0.0),\n",
       " ('hscs are distinct from hspcs', 0.0),\n",
       " ('in our multipotent progenitor populations', 0.0),\n",
       " ('increased the hematopoietic progenitor populations', 0.0),\n",
       " ('kit + sca 1 +', 0.0),\n",
       " ('kit pos sca 1 pos', 0.0),\n",
       " ('kit positive c kit +', 0.0),\n",
       " ('l sca1 + kit +', 0.0),\n",
       " ('leads to expansion of hspcs', 0.0),\n",
       " ('lin kit + sca +', 0.0),\n",
       " ('lin sca + ckit +', 0.0),\n",
       " ('lin sca1 + ckit +', 0.0),\n",
       " ('lin sca1 + ckit hi', 0.0),\n",
       " ('lin sca1 c kit +', 0.0),\n",
       " ('lineage positive but not early', 0.0),\n",
       " ('linsca1 + c kit +', 0.0),\n",
       " ('low sca1 + ckit +', 0.0),\n",
       " ('lumbar lordosis ll lumbosacral kyphosis', 0.0),\n",
       " ('measured the frequency of hspcs', 0.0),\n",
       " ('mm cells because the former', 0.0),\n",
       " ('mnc and stem cells enriched', 0.0),\n",
       " ('mouse fancd2 − − hspc', 0.0),\n",
       " ('muris also harbored increased hpsc', 0.0),\n",
       " ('neg sca1 + ckit +', 0.0),\n",
       " ('neg sca1 + kit +', 0.0),\n",
       " ('negative kit positive sca positive', 0.0),\n",
       " ('negative sca1 + ckit +', 0.0),\n",
       " ('negative sca1 positive kit positive', 0.0),\n",
       " ('non committed lin neg hspcs', 0.0),\n",
       " ('of csfr3 − − hspcs', 0.0),\n",
       " ('panel the average cells number', 0.0),\n",
       " ('populations 1 × 10 5', 0.0),\n",
       " ('positive c kit positive populations', 0.0),\n",
       " ('progenitor within the lsk compartment', 0.0),\n",
       " ('proliferation of hematopoietic progenitor cells', 0.0),\n",
       " ('resident hematopoietic stem progenitor cells', 0.0),\n",
       " ('sca + 1c kit +', 0.0),\n",
       " ('sca + c kit +', 0.0),\n",
       " ('sca 1 + cd117 +', 0.0),\n",
       " ('sca1 + + kit +', 0.0),\n",
       " ('sca1 + c kit hi', 0.0),\n",
       " ('sca1 in the perturbed hsc', 0.0),\n",
       " ('sca1 − c kit +', 0.0),\n",
       " ('studies we demonstrated that hspc', 0.0),\n",
       " ('tensor used local steering kernel', 0.0),\n",
       " ('the frequency of stem progenitor', 0.0),\n",
       " ('the total number of progenitor', 0.0),\n",
       " ('we conclude that both immature', 0.0),\n",
       " ('with the native b6 lsk', 0.0),\n",
       " ('without affecting the stem cells', 0.0),\n",
       " ('− s + k +', 0.0),\n",
       " ('+ ckit + hematopoietic progenitor cells', 0.0),\n",
       " ('+ lin sca1 c kit +', 0.0),\n",
       " ('+ non committed lin neg hspcs', 0.0),\n",
       " ('001 a analysis of pb progenitor', 0.0),\n",
       " ('1 + and c kit +', 0.0),\n",
       " ('1 + c k it +', 0.0),\n",
       " ('1 + c kit + cells', 0.0),\n",
       " ('7rα − ckit + sca1 +', 0.0),\n",
       " ('aeruginosa caused the expansion of hspcs', 0.0),\n",
       " ('also enhance the proliferation of hspcs', 0.0),\n",
       " ('alter bcr abl1 + stem cells', 0.0),\n",
       " ('and mouse fancd2 − − hspc', 0.0),\n",
       " ('and the total number of progenitor', 0.0),\n",
       " ('at 5 × 10 4 cells', 0.0),\n",
       " ('both c kit and sca 1', 0.0),\n",
       " ('c kit pos sca 1 pos', 0.0),\n",
       " ('c kit positive c kit +', 0.0),\n",
       " ('ca1 + c k it +', 0.0),\n",
       " ('caused by dysfunctional expansion of hspc', 0.0),\n",
       " ('cells populations 1 × 10 5', 0.0),\n",
       " ('clustered with the native b6 lsk', 0.0),\n",
       " ('dapi lin kit + sca +', 0.0),\n",
       " ('depletion linsca1 + c kit +', 0.0),\n",
       " ('ehrlichia muris also harbored increased hpsc', 0.0),\n",
       " ('exposure leads to expansion of hspcs', 0.0),\n",
       " ('expression in our multipotent progenitor populations', 0.0),\n",
       " ('expression sca 1 and c kit', 0.0),\n",
       " ('expression sca1 in the perturbed hsc', 0.0),\n",
       " ('frequency between lt hscs and hspcs', 0.0),\n",
       " ('function resident hematopoietic stem progenitor cells', 0.0),\n",
       " ('gmp without affecting the stem cells', 0.0),\n",
       " ('hematopoietic progenitor within the lsk compartment', 0.0),\n",
       " ('hspcs lin sca + ckit +', 0.0),\n",
       " ('id1 expression in hsc multipotent progenitor', 0.0),\n",
       " ('in both wt and fa hspc', 0.0),\n",
       " ('in expansion of hematopoietic progenitor cells', 0.0),\n",
       " ('in lin sca1 + ckit hi', 0.0),\n",
       " ('in mnc and stem cells enriched', 0.0),\n",
       " ('in the frequency of stem progenitor', 0.0),\n",
       " ('increased proliferation of hematopoietic progenitor cells', 0.0),\n",
       " ('kit + sca 1 + cells', 0.0),\n",
       " ('kit + sca 1 + fraction', 0.0),\n",
       " ('l − s + k +', 0.0),\n",
       " ('left panel the average cells number', 0.0),\n",
       " ('lin neg sca1 + kit +', 0.0),\n",
       " ('lin negative sca1 positive kit positive', 0.0),\n",
       " ('lin sca + c kit +', 0.0),\n",
       " ('linckit + sca 1 + cells', 0.0),\n",
       " ('lineage negative kit positive sca positive', 0.0),\n",
       " ('lineage sca1 + c kit +', 0.0),\n",
       " ('lineage − sca + ckit +', 0.0),\n",
       " ('lineage − sca1 + ckit +', 0.0),\n",
       " ('lineage − sca1 + kit +', 0.0),\n",
       " ('low sca1 + c kit +', 0.0),\n",
       " ('lt hscs are distinct from hspcs', 0.0),\n",
       " ('mature lineage positive but not early', 0.0),\n",
       " ('mep mm cells because the former', 0.0),\n",
       " ('negative l sca1 + kit +', 0.0),\n",
       " ('negative sca 1 + ckit +', 0.0),\n",
       " ('negative sca1 + c kit +', 0.0),\n",
       " ('ns gfp low and hspc cells', 0.0),\n",
       " ('of sca 1 and c kit', 0.0),\n",
       " ('or lin sca1 + ckit +', 0.0),\n",
       " ('or tensor used local steering kernel', 0.0),\n",
       " ('phenotypic lin sca1 + kit +', 0.0),\n",
       " ('phycoerythrin cy5 pe cy5 antibodies hsc', 0.0),\n",
       " ('primitive lin sca + kit +', 0.0),\n",
       " ('reduction of csfr3 − − hspcs', 0.0),\n",
       " ('relative frequency of lineage neg sk', 0.0),\n",
       " ('response assays and by quantifying progenitor', 0.0),\n",
       " ('sacrofemoral distance sfd and lumbosacral kyphosis', 0.0),\n",
       " ('sca 1 + and ckit +', 0.0),\n",
       " ('sca 1 and c kit positive', 0.0),\n",
       " ('sca 1 hi c kit hi', 0.0),\n",
       " ('sca 1 pos c kit pos', 0.0),\n",
       " ('sca 1 positive c kit positive', 0.0),\n",
       " ('sca1 positive c kit positive populations', 0.0),\n",
       " ('significant increased the hematopoietic progenitor populations', 0.0),\n",
       " ('significant we conclude that both immature', 0.0),\n",
       " ('ss lumbar lordosis ll lumbosacral kyphosis', 0.0),\n",
       " ('the gated area for lsk cells', 0.0),\n",
       " ('the lin sca + kit +', 0.0),\n",
       " ('the lin sca1 + kit +', 0.0),\n",
       " ('this studies we demonstrated that hspc', 0.0),\n",
       " ('we measured the frequency of hspcs', 0.0),\n",
       " ('− and fancc − − hscp', 0.0),\n",
       " ('− ckit + sca 1 +', 0.0),\n",
       " ('− low sca1 + ckit +', 0.0),\n",
       " ('− sca + 1c kit +', 0.0),\n",
       " ('− sca1 + + kit +', 0.0),\n",
       " ('− sca1 + c kit hi', 0.0),\n",
       " ('− sca1 − c kit +', 0.0),\n",
       " ('+ c kit positive c kit +', 0.0),\n",
       " ('+ lineage sca 1 + cd117 +', 0.0),\n",
       " ('+ or lin sca1 + ckit +', 0.0),\n",
       " ('0 001 a analysis of pb progenitor', 0.0),\n",
       " ('27 exposure leads to expansion of hspcs', 0.0),\n",
       " ('37 in mnc and stem cells enriched', 0.0),\n",
       " ('and expression sca 1 and c kit', 0.0),\n",
       " ('and mep mm cells because the former', 0.0),\n",
       " ('and ns gfp low and hspc cells', 0.0),\n",
       " ('as lin − sca1 + kit +', 0.0),\n",
       " ('biosciences lineage sca 1 + cd117 +', 0.0),\n",
       " ('both lin − sca + ckit +', 0.0),\n",
       " ('c kit + sca 1 + cells', 0.0),\n",
       " ('c kit + sca 1 + fraction', 0.0),\n",
       " ('c lin − sca1 + ckit +', 0.0),\n",
       " ('ca 1 + c k it +', 0.0),\n",
       " ('can also enhance the proliferation of hspcs', 0.0),\n",
       " ('cells depletion linsca1 + c kit +', 0.0),\n",
       " ('cells expression sca1 in the perturbed hsc', 0.0),\n",
       " ('cells gmp without affecting the stem cells', 0.0),\n",
       " ('cells l − s + k +', 0.0),\n",
       " ('cells the lin sca + kit +', 0.0),\n",
       " ('change significant we conclude that both immature', 0.0),\n",
       " ('containing lineage − sca1 + kit +', 0.0),\n",
       " ('d the gated area for lsk cells', 0.0),\n",
       " ('depletion primitive lin sca + kit +', 0.0),\n",
       " ('difference in the frequency of stem progenitor', 0.0),\n",
       " ('dissimilar lineage negative sca1 + ckit +', 0.0),\n",
       " ('expression both c kit and sca 1', 0.0),\n",
       " ('expression of sca 1 and c kit', 0.0),\n",
       " ('facs left panel the average cells number', 0.0),\n",
       " ('fl lin sca 1 + kit +', 0.0),\n",
       " ('gfp + lin sca1 c kit +', 0.0),\n",
       " ('high lt hscs are distinct from hspcs', 0.0),\n",
       " ('highlight the lin sca1 + kit +', 0.0),\n",
       " ('hsc lin − sca1 + ckit +', 0.0),\n",
       " ('il 7rα − ckit + sca1 +', 0.0),\n",
       " ('in lin neg sca1 + ckit +', 0.0),\n",
       " ('in lin neg sca1 + kit +', 0.0),\n",
       " ('in this studies we demonstrated that hspc', 0.0),\n",
       " ('incidence and the total number of progenitor', 0.0),\n",
       " ('infection we measured the frequency of hspcs', 0.0),\n",
       " ('isolated linckit + sca 1 + cells', 0.0),\n",
       " ('isolated lineage − sca1 + ckit +', 0.0),\n",
       " ('lin + non committed lin neg hspcs', 0.0),\n",
       " ('lin c kit + sca 1 +', 0.0),\n",
       " ('lin low sca1 + c kit +', 0.0),\n",
       " ('lin negative sca 1 + ckit +', 0.0),\n",
       " ('lin − sca + 1c kit +', 0.0),\n",
       " ('lin − sca1 + c kit hi', 0.0),\n",
       " ('lin − sca1 − c kit +', 0.0),\n",
       " ('lineage c kit + sca 1 +', 0.0),\n",
       " ('lineage lin − sca1 + ckit +', 0.0),\n",
       " ('lineage negative l sca1 + kit +', 0.0),\n",
       " ('lineage − sca 1 + kit +', 0.0),\n",
       " ('lineage − sca1 + c kit +', 0.0),\n",
       " ('ll sacrofemoral distance sfd and lumbosacral kyphosis', 0.0),\n",
       " ('lo sca 1 + c kit +', 0.0),\n",
       " ('lsks lin − sca1 + kit +', 0.0),\n",
       " ('marked reduction of csfr3 − − hspcs', 0.0),\n",
       " ('marker sca 1 + c kit +', 0.0),\n",
       " ('marrow lin sca1 + c kit +', 0.0),\n",
       " ('marrow lin − ckit + sca1 +', 0.0),\n",
       " ('mice significant increased the hematopoietic progenitor populations', 0.0),\n",
       " ('murine lin sca1 + c kit +', 0.0),\n",
       " ('neg c kit pos sca 1 pos', 0.0),\n",
       " ('neg sca 1 + c kit +', 0.0),\n",
       " ('neg sca 1 + c kit high', 0.0),\n",
       " ('neg sca 1 pos c kit pos', 0.0),\n",
       " ('negative sca 1 + and ckit +', 0.0),\n",
       " ('negative sca 1 and c kit positive', 0.0),\n",
       " ('negative sca 1 positive c kit positive', 0.0),\n",
       " ('negative sca1 positive c kit positive populations', 0.0),\n",
       " ('not alter bcr abl1 + stem cells', 0.0),\n",
       " ('of hematopoietic progenitor within the lsk compartment', 0.0),\n",
       " ('of hspcs lin sca + ckit +', 0.0),\n",
       " ('of lin negative sca1 positive kit positive', 0.0),\n",
       " ('of lin − sca1 + ckit +', 0.0),\n",
       " ('of lineage sca1 + c kit +', 0.0),\n",
       " ('or frequency between lt hscs and hspcs', 0.0),\n",
       " ('or lin − sca1 + ckit +', 0.0),\n",
       " ('or phycoerythrin cy5 pe cy5 antibodies hsc', 0.0),\n",
       " ('pd20 and mouse fancd2 − − hspc', 0.0),\n",
       " ('peculiar function resident hematopoietic stem progenitor cells', 0.0),\n",
       " ('pseudomonas aeruginosa caused the expansion of hspcs', 0.0),\n",
       " ('reseeding at 5 × 10 4 cells', 0.0),\n",
       " ('resulted in expansion of hematopoietic progenitor cells', 0.0),\n",
       " ('s ca1 + c k it +', 0.0),\n",
       " ('sca 1 + and c kit +', 0.0),\n",
       " ('sca1 + ckit + hematopoietic progenitor cells', 0.0),\n",
       " ('signaling in both wt and fa hspc', 0.0),\n",
       " ('signaling in lin sca1 + ckit hi', 0.0),\n",
       " ('significant increased proliferation of hematopoietic progenitor cells',\n",
       "  0.0),\n",
       " ('slope ss lumbar lordosis ll lumbosacral kyphosis', 0.0),\n",
       " ('sorted cells populations 1 × 10 5', 0.0),\n",
       " ('sorted lineage negative kit positive sca positive', 0.0),\n",
       " ('studies lin − sca1 + ckit +', 0.0),\n",
       " ('subgated dapi lin kit + sca +', 0.0),\n",
       " ('template or tensor used local steering kernel', 0.0),\n",
       " ('that clustered with the native b6 lsk', 0.0),\n",
       " ('that id1 expression in hsc multipotent progenitor', 0.0),\n",
       " ('the lin neg sca1 + ckit +', 0.0),\n",
       " ('the lin sca1 + c kit +', 0.0),\n",
       " ('the lin − sca + ckit +', 0.0),\n",
       " ('the lineage negative sca1 + ckit +', 0.0),\n",
       " ('the phenotypic lin sca1 + kit +', 0.0),\n",
       " ('the relative frequency of lineage neg sk', 0.0),\n",
       " ('time response assays and by quantifying progenitor', 0.0),\n",
       " ('to lin − sca + ckit +', 0.0),\n",
       " ('transcripts expression in our multipotent progenitor populations', 0.0),\n",
       " ('type mature lineage positive but not early', 0.0),\n",
       " ('was caused by dysfunctional expansion of hspc', 0.0),\n",
       " ('with ehrlichia muris also harbored increased hpsc', 0.0),\n",
       " ('− lin − ckit + sca1 +', 0.0),\n",
       " ('− sca 1 + c kit high', 0.0),\n",
       " ('− sca 1 hi c kit hi', 0.0),\n",
       " ('− − and fancc − − hscp', 0.0),\n",
       " ('+ cells depletion linsca1 + c kit +', 0.0),\n",
       " ('+ lin − ckit + sca 1 +', 0.0),\n",
       " ('+ lin − sca 1 + ckit +', 0.0),\n",
       " ('+ lin − sca1 − c kit +', 0.0),\n",
       " ('000 transcripts expression in our multipotent progenitor populations', 0.0),\n",
       " ('1 + c kit positive c kit +', 0.0),\n",
       " ('5 sorted lineage negative kit positive sca positive', 0.0),\n",
       " ('5a c lin − sca1 + ckit +', 0.0),\n",
       " ('< 0 001 a analysis of pb progenitor', 0.0),\n",
       " ('a marked reduction of csfr3 − − hspcs', 0.0),\n",
       " ('a peculiar function resident hematopoietic stem progenitor cells', 0.0),\n",
       " ('a1 37 in mnc and stem cells enriched', 0.0),\n",
       " ('adult mice significant increased the hematopoietic progenitor populations',\n",
       "  0.0),\n",
       " ('allophycocyanin or phycoerythrin cy5 pe cy5 antibodies hsc', 0.0),\n",
       " ('although pseudomonas aeruginosa caused the expansion of hspcs', 0.0),\n",
       " ('analyzed the lin neg sca1 + ckit +', 0.0),\n",
       " ('and lin sca 1 + c kit +', 0.0),\n",
       " ('and time response assays and by quantifying progenitor', 0.0),\n",
       " ('apoptosis of lin sca 1 + kit +', 0.0),\n",
       " ('are lin low sca1 + c kit +', 0.0),\n",
       " ('are lineage − sca 1 + ckit +', 0.0),\n",
       " ('as lin − sca1 + c kit +', 0.0),\n",
       " ('b gfp + lin sca1 c kit +', 0.0),\n",
       " ('bd biosciences lineage sca 1 + cd117 +', 0.0),\n",
       " ('bearing the phenotypic lin sca1 + kit +', 0.0),\n",
       " ('before reseeding at 5 × 10 4 cells', 0.0),\n",
       " ('bm lineage − low sca1 + ckit +', 0.0),\n",
       " ('bone marrow lin sca1 + c kit +', 0.0),\n",
       " ('bone marrow lin − ckit + sca1 +', 0.0),\n",
       " ('by lin − sca1 + c kit +', 0.0),\n",
       " ('canonical signaling in lin sca1 + ckit hi', 0.0),\n",
       " ('cases lin − sca1 + c kit +', 0.0),\n",
       " ('cd166 + lineage sca 1 + cd117 +', 0.0),\n",
       " ('cd34 − lin − ckit + sca1 +', 0.0),\n",
       " ('cells expression both c kit and sca 1', 0.0),\n",
       " ('cells hsc lin − sca1 + ckit +', 0.0),\n",
       " ('circulating lin − sca 1 + ckit +', 0.0),\n",
       " ('cocultured lin − sca 1 + kit +', 0.0),\n",
       " ('committed lin + non committed lin neg hspcs', 0.0),\n",
       " ('containing lin − sca 1 + kit +', 0.0),\n",
       " ('decrease lin − sca 1 + ckit +', 0.0),\n",
       " ('deficient lin c kit + sca 1 +', 0.0),\n",
       " ('deficient lsks lin − sca1 + kit +', 0.0),\n",
       " ('did not alter bcr abl1 + stem cells', 0.0),\n",
       " ('display dissimilar lineage negative sca1 + ckit +', 0.0),\n",
       " ('egfp lin − sca1 + c kit +', 0.0),\n",
       " ('enriched lin sca 1 + c kit +', 0.0),\n",
       " ('enriched lin − sca 1 + kit +', 0.0),\n",
       " ('expansion of lin sca 1 + kit +', 0.0),\n",
       " ('fanca − − and fancc − − hscp', 0.0),\n",
       " ('following infection we measured the frequency of hspcs', 0.0),\n",
       " ('following lin − sca 1 + ckit +', 0.0),\n",
       " ('for sorted cells populations 1 × 10 5', 0.0),\n",
       " ('form lineage negative sca1 + c kit +', 0.0),\n",
       " ('frequency of lin − sca + ckit +', 0.0),\n",
       " ('frequency of lin − sca1 + ckit +', 0.0),\n",
       " ('from cells expression sca1 in the perturbed hsc', 0.0),\n",
       " ('gfp high lt hscs are distinct from hspcs', 0.0),\n",
       " ('high and ns gfp low and hspc cells', 0.0),\n",
       " ('hsc containing lineage − sca1 + kit +', 0.0),\n",
       " ('il 27 exposure leads to expansion of hspcs', 0.0),\n",
       " ('immature lineage c kit + sca 1 +', 0.0),\n",
       " ('immature lineage sca 1 + c kit +', 0.0),\n",
       " ('in d the gated area for lsk cells', 0.0),\n",
       " ('in fl lin sca 1 + kit +', 0.0),\n",
       " ('in isolated linckit + sca 1 + cells', 0.0),\n",
       " ('in lin − sca 1 + ckit +', 0.0),\n",
       " ('in lin − sca1 + c kit +', 0.0),\n",
       " ('in lineage sca 1 + c kit +', 0.0),\n",
       " ('in the lineage negative sca1 + ckit +', 0.0),\n",
       " ('including lineage sca 1 + c kit +', 0.0),\n",
       " ('increased lin sca 1 + c kit +', 0.0),\n",
       " ('increased the lin sca1 + c kit +', 0.0),\n",
       " ('infection with ehrlichia muris also harbored increased hpsc', 0.0),\n",
       " ('invitrogen lin − sca1 + c kit +', 0.0),\n",
       " ('kit + or lin sca1 + ckit +', 0.0),\n",
       " ('levels or frequency between lt hscs and hspcs', 0.0),\n",
       " ('lin c kit + sca 1 + cells', 0.0),\n",
       " ('lin lo sca 1 + c kit +', 0.0),\n",
       " ('lin low sca 1 + c kit +', 0.0),\n",
       " ('lin neg sca 1 + c kit high', 0.0),\n",
       " ('lin neg sca 1 pos c kit pos', 0.0),\n",
       " ('lin negative sca 1 + c kit +', 0.0),\n",
       " ('lin − c kit + sca 1 +', 0.0),\n",
       " ('lin − sca 1 + c kit high', 0.0),\n",
       " ('lin − sca 1 hi c kit hi', 0.0),\n",
       " ('line pd20 and mouse fancd2 − − hspc', 0.0),\n",
       " ('lineage lin − sca1 + c kit +', 0.0),\n",
       " ('lineage marker sca 1 + c kit +', 0.0),\n",
       " ('lineage neg c kit pos sca 1 pos', 0.0),\n",
       " ('lineage negative sca 1 + and ckit +', 0.0),\n",
       " ('lineage negative sca 1 positive c kit positive', 0.0),\n",
       " ('lineage negative sca1 positive c kit positive populations', 0.0),\n",
       " ('lineage − c kit + sca 1 +', 0.0),\n",
       " ('lordosis ll sacrofemoral distance sfd and lumbosacral kyphosis', 0.0),\n",
       " ('manipulated lineage − sca 1 + ckit +', 0.0),\n",
       " ('marker negative sca 1 + c kit +', 0.0),\n",
       " ('marker negative sca 1 and c kit positive', 0.0),\n",
       " ('marrow cells l − s + k +', 0.0),\n",
       " ('marrow lin negative sca 1 + ckit +', 0.0),\n",
       " ('marrow lin − sca 1 + ckit +', 0.0),\n",
       " ('marrow lineage lin − sca1 + ckit +', 0.0),\n",
       " ('mkp and mep mm cells because the former', 0.0),\n",
       " ('mmc lin − sca1 + c kit +', 0.0),\n",
       " ('mobilization lin sca 1 + c kit +', 0.0),\n",
       " ('mpl in lin neg sca1 + kit +', 0.0),\n",
       " ('multidimensional template or tensor used local steering kernel', 0.0),\n",
       " ('negative lin − sca 1 + kit +', 0.0),\n",
       " ('negative lin − sca1 + c kit hi', 0.0),\n",
       " ('not change significant we conclude that both immature', 0.0),\n",
       " ('notch signaling in both wt and fa hspc', 0.0),\n",
       " ('number of hspcs lin sca + ckit +', 0.0),\n",
       " ('number of lin negative sca1 positive kit positive', 0.0),\n",
       " ('number of lin − sca + ckit +', 0.0),\n",
       " ('number of lineage sca1 + c kit +', 0.0),\n",
       " ('of both lin − sca + ckit +', 0.0),\n",
       " ('of lin − sca 1 + kit +', 0.0),\n",
       " ('of lin − sca1 + + kit +', 0.0),\n",
       " ('of lineage − low sca1 + ckit +', 0.0),\n",
       " ('of lineage − sca 1 + ckit +', 0.0),\n",
       " ('of lineage − sca1 + c kit +', 0.0),\n",
       " ('of murine lin sca1 + c kit +', 0.0),\n",
       " ('or lin − sca + 1c kit +', 0.0),\n",
       " ('or lin − sca 1 + ckit +', 0.0),\n",
       " ('percentage of lin sca + c kit +', 0.0),\n",
       " ('plots of hematopoietic progenitor within the lsk compartment', 0.0),\n",
       " ('plots subgated dapi lin kit + sca +', 0.0),\n",
       " ('pool in this studies we demonstrated that hspc', 0.0),\n",
       " ('populations of lin sca + c kit +', 0.0),\n",
       " ('preferential depletion primitive lin sca + kit +', 0.0),\n",
       " ('primitive lin − sca1 + c kit +', 0.0),\n",
       " ('production lineage − sca 1 + ckit +', 0.0),\n",
       " ('profile that clustered with the native b6 lsk', 0.0),\n",
       " ('progenitor cells gmp without affecting the stem cells', 0.0),\n",
       " ('proliferation of lin sca 1 + kit +', 0.0),\n",
       " ('reduced the relative frequency of lineage neg sk', 0.0),\n",
       " ('reduction in lin neg sca1 + ckit +', 0.0),\n",
       " ('remarkably the lin − sca + ckit +', 0.0),\n",
       " ('representative facs left panel the average cells number', 0.0),\n",
       " ('revealed significant increased proliferation of hematopoietic progenitor cells',\n",
       "  0.0),\n",
       " ('s ca 1 + c k it +', 0.0),\n",
       " ('sacral slope ss lumbar lordosis ll lumbosacral kyphosis', 0.0),\n",
       " ('sepsis was caused by dysfunctional expansion of hspc', 0.0),\n",
       " ('shown that id1 expression in hsc multipotent progenitor', 0.0),\n",
       " ('stem cells the lin sca + kit +', 0.0),\n",
       " ...]"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "top"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [],
   "source": [
    "longforms0 = miners['LSK'].get_longforms()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('+', 83.90816326530613),\n",
       " ('cells', 12.588235294117647),\n",
       " ('hspcs', 5.5),\n",
       " ('progenitor', 4.0),\n",
       " ('hspc', 3.0),\n",
       " ('positive', 1.5),\n",
       " ('hi', 1.3333333333333333),\n",
       " ('populations', 1.3333333333333333)]"
      ]
     },
     "execution_count": 114,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "longforms0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [],
   "source": [
    "longforms0 = [x for i, x in enumerate(longforms0) if i in [0]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('lymphocyte specific protein tyrosine kinase', 23.083333333333332)]"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "longforms0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[(0, ('lymphocyte specific protein tyrosine kinase', 23.083333333333332)),\n",
       " (1, ('kinase', 9.076923076923077)),\n",
       " (2, ('specific protein tyrosine kinase', 8.733333333333334)),\n",
       " (3, ('tyrosine kinase', 7.235294117647058)),\n",
       " (4, ('of lymphocyte specific protein tyrosine kinase', 3.0)),\n",
       " (5, ('and lymphocyte specific protein tyrosine kinase', 2.0)),\n",
       " (6,\n",
       "  ('min 1 22 k and kp values for the alkylation of bovine trypsin with lysine chloromethyl ketone',\n",
       "   2)),\n",
       " (7, ('lymphocyte specific kinase', 1.3333333333333333)),\n",
       " (8, ('leukocyte specific protein tyrosine kinase', 1.3333333333333333)),\n",
       " (9, ('lck proto oncogene src family tyrosine kinase', 1.3333333333333333)),\n",
       " (10, ('cell kill', 1.0)),\n",
       " (11, ('of p56', 1.0)),\n",
       " (12, ('p56 lck', 1.0)),\n",
       " (13, ('the lymphocyte specific kinase', 1.0)),\n",
       " (14, ('lymphocyte cell specific protein tyrosine kinase', 1.0)),\n",
       " (15, ('the lymphocyte specific protein tyrosine kinase', 1.0)),\n",
       " (16, ('kinase such as lymphocyte specific protein tyrosine kinase', 1.0)),\n",
       " (17, ('levels of cd247 leukocyte specific protein tyrosine kinase', 1.0)),\n",
       " (18, ('of lck proto oncogene src family tyrosine kinase', 1.0)),\n",
       " (19,\n",
       "  ('recent transcriptomic analysis revealed upregulation of lymphocyte specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (20,\n",
       "  ('levels providing new insights and testable hypotheses lymphocyte specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (21,\n",
       "  ('ntigen presenting cell upon tcr activation activation lymphocyte specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (22,\n",
       "  ('ular receptor phosphorylation various kinase such as lymphocyte specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (23,\n",
       "  ('ene tyrosine protein kinase gene fyn together with lymphocyte specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (24,\n",
       "  ('jnk signaling and in parallel cisplatin resistance via lymphocyte specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (25,\n",
       "  ('mical derivation of 3 nitrotyrosine and identified that lymphocyte specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (26,\n",
       "  ('reiley et al 2006 since cyld positively regulated lymphocyte specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (27,\n",
       "  ('tion into the immunological synapse the interaction of lymphocyte specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (28,\n",
       "  ('y increase their signaling capacity by intracellular recruitment of the lymphocyte specific kinase',\n",
       "   1)),\n",
       " (29,\n",
       "  ('ases mapk3 in addition downregulated gene of lck proto oncogene src family tyrosine kinase',\n",
       "   1)),\n",
       " (30,\n",
       "  ('ations showed that it interaction with receptor tyrosine kinase such as lymphocyte specific kinase',\n",
       "   1)),\n",
       " (31,\n",
       "  ('b patches to become detergent sensitive as patches of colocalized protein were several disrupted',\n",
       "   1)),\n",
       " (32, ('cd8 + t cell from ot i rag2 − − mice expressing wild type', 1.0)),\n",
       " (33,\n",
       "  ('eness one src like tyrosine kinase trafficked by inf2 lymphocyte specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (34,\n",
       "  ('i recognition fig 4c starts with the recruitment of lymphocyte specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (35,\n",
       "  ('ion the distal proline motif pyap associates with lymphocyte cell specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (36,\n",
       "  ('ll lines harbor an activation and targetable sub network composed of lymphocytespecific tyrosine kinase',\n",
       "   1)),\n",
       " (37,\n",
       "  ('ohydrate intake previous research suggests that an ad libitum very low carbohydrate ketogenic diet',\n",
       "   1)),\n",
       " (38,\n",
       "  ('py783 plc γ1 cell signaling plc γ1 cell signaling lymphocyte specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (39,\n",
       "  ('rad23b and the mouse laminin receptor lamr1 and the lymphocyte specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (40,\n",
       "  ('rst identified by reverse phase protein arrays the lymphocyte cell specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (41,\n",
       "  ('transducer and activation of transcript 3 stat3 lck proto oncogene src family tyrosine kinase',\n",
       "   1)),\n",
       " (42,\n",
       "  ('carboxylates was reproducibly activation in the m5076 tumor producing mean maximum gross log cell kill',\n",
       "   1)),\n",
       " (43,\n",
       "  ('function s black n = 6 transcript levels of cd247 leukocyte specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (44,\n",
       "  ('gnaling is induced by il 18 specific il 18 induced phosphorylation of phosphorylation p56 lck',\n",
       "   1)),\n",
       " (45,\n",
       "  ('h t cell receptor tcr is fundamental to adaptive immune responses the lymphocyte specific kinase',\n",
       "   1)),\n",
       " (46,\n",
       "  ('has been reported to involve src family kinase such as lymphocyte specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (47,\n",
       "  ('her alterations in tcr signaling like in zap70 itk or lymphocyte specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (48,\n",
       "  ('inase activation 68 tcr induced stimulation of t cell led to simultaneous phosphorylation of p56',\n",
       "   1)),\n",
       " (49,\n",
       "  ('inase in the tcr signaling cascade is the membrane bound leukocyte specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (50,\n",
       "  ('mmary a non myristylated form lckm of the human t lymphocyte specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (51,\n",
       "  ('nced the binding of acetylated heat shock protein 90 to lymphocyte specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (52,\n",
       "  ('on state of the tcr proximal tyrosine kinase zap70 and lymphocyte specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (53,\n",
       "  ('r lrrc8a and that lrrc8a ligation activation akt via the lymphocyte specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (54,\n",
       "  ('several potent kinase inhibitors such as those described for the related tyrosine kinase p56 lck',\n",
       "   1)),\n",
       " (55,\n",
       "  ('the activation of receptor in immune responses in the ppi network src kinase lymphocyte kinase',\n",
       "   1)),\n",
       " (56,\n",
       "  ('this screen revealed a decrease in levels of ror2 and lymphocyte specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (57,\n",
       "  ('ng pathway such as t cell receptor ζ cd247 16 and lymphocyte specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (58,\n",
       "  ('o t cell expressing in t cell but not macrophages of lymphocyte specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (59,\n",
       "  ('osine sites on the cd28 cd3 ζ car is phosphorylation by lymphocyte specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (60,\n",
       "  ('s the most important of which in endogenous t cell is lymphocyte specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (61,\n",
       "  ('the t cell receptor to the nucleus interaction of lck proto oncogene src family tyrosine kinase',\n",
       "   1)),\n",
       " (62,\n",
       "  ('ce for vav3 p = 0 16 gene expressing levels of cd247 leukocyte specific protein tyrosine kinase',\n",
       "   1)),\n",
       " (63,\n",
       "  ('ng to the tcr leading to the recruitment and activation of the proto oncogene tyrosine protein kinase',\n",
       "   1)),\n",
       " (64,\n",
       "  ('concentration in the femoral bone marrow by a factor of about 7 the log leukaemic stem cell kill',\n",
       "   1)),\n",
       " (65,\n",
       "  ('hemoglobin hba1c > 6 0 and elevated body weight bmi > 25 to a very low carbohydrate ketogenic',\n",
       "   1)),\n",
       " (66,\n",
       "  ('ickly inhibit microbe growth leading to the death of microbe it is also a novel inhibitors of p56',\n",
       "   1)),\n",
       " (67,\n",
       "  ('implement tcr ligand discrimination cd8 + t cell from ot i rag2 − − mice expressing wild type',\n",
       "   1)),\n",
       " (68,\n",
       "  ('00 nm it still demonstrates high selectivity for egfr ic 50 stk10 ic 50 egfr = 45 ic 50',\n",
       "   1)),\n",
       " (69,\n",
       "  ('ive of three independent experiments cd8 + t cell from ot i rag2 − − mice expressing wild type',\n",
       "   1)),\n",
       " (70,\n",
       "  ('h itk k390r 1 μm lanes 4 6 8 serve as substrates for lck 200 nm lanes 1 and 2 are no enzyme',\n",
       "   1)),\n",
       " (71, ('50', 0.0)),\n",
       " (72, ('diet', 0.0)),\n",
       " (73, ('disrupted', 0.0)),\n",
       " (74, ('enzyme', 0.0)),\n",
       " (75, ('ketogenic', 0.0)),\n",
       " (76, ('ketone', 0.0)),\n",
       " (77, ('kill', 0.0)),\n",
       " (78, ('lck', 0.0)),\n",
       " (79, ('p56', 0.0)),\n",
       " (80, ('type', 0.0)),\n",
       " (81, ('carbohydrate ketogenic', 0.0)),\n",
       " (82, ('chloromethyl ketone', 0.0)),\n",
       " (83, ('ic 50', 0.0)),\n",
       " (84, ('ketogenic diet', 0.0)),\n",
       " (85, ('lymphocyte kinase', 0.0)),\n",
       " (86, ('no enzyme', 0.0)),\n",
       " (87, ('protein kinase', 0.0)),\n",
       " (88, ('several disrupted', 0.0)),\n",
       " (89, ('specific kinase', 0.0)),\n",
       " (90, ('wild type', 0.0)),\n",
       " (91, ('45 ic 50', 0.0)),\n",
       " (92, ('are no enzyme', 0.0)),\n",
       " (93, ('carbohydrate ketogenic diet', 0.0)),\n",
       " (94, ('expressing wild type', 0.0)),\n",
       " (95, ('family tyrosine kinase', 0.0)),\n",
       " (96, ('inhibitors of p56', 0.0)),\n",
       " (97, ('kinase lymphocyte kinase', 0.0)),\n",
       " (98, ('kinase p56 lck', 0.0)),\n",
       " (99, ('log cell kill', 0.0)),\n",
       " (100, ('low carbohydrate ketogenic', 0.0)),\n",
       " (101, ('lymphocytespecific tyrosine kinase', 0.0)),\n",
       " (102, ('lysine chloromethyl ketone', 0.0)),\n",
       " (103, ('phosphorylation of p56', 0.0)),\n",
       " (104, ('phosphorylation p56 lck', 0.0)),\n",
       " (105, ('protein tyrosine kinase', 0.0)),\n",
       " (106, ('stem cell kill', 0.0)),\n",
       " (107, ('tyrosine protein kinase', 0.0)),\n",
       " (108, ('were several disrupted', 0.0)),\n",
       " (109, ('2 are no enzyme', 0.0)),\n",
       " (110, ('= 45 ic 50', 0.0)),\n",
       " (111, ('as lymphocyte specific kinase', 0.0)),\n",
       " (112, ('gross log cell kill', 0.0)),\n",
       " (113, ('leukaemic stem cell kill', 0.0)),\n",
       " (114, ('low carbohydrate ketogenic diet', 0.0)),\n",
       " (115, ('mice expressing wild type', 0.0)),\n",
       " (116, ('novel inhibitors of p56', 0.0)),\n",
       " (117, ('of lymphocytespecific tyrosine kinase', 0.0)),\n",
       " (118, ('of phosphorylation p56 lck', 0.0)),\n",
       " (119, ('oncogene tyrosine protein kinase', 0.0)),\n",
       " (120, ('protein were several disrupted', 0.0)),\n",
       " (121, ('simultaneous phosphorylation of p56', 0.0)),\n",
       " (122, ('src family tyrosine kinase', 0.0)),\n",
       " (123, ('src kinase lymphocyte kinase', 0.0)),\n",
       " (124, ('tyrosine kinase p56 lck', 0.0)),\n",
       " (125, ('very low carbohydrate ketogenic', 0.0)),\n",
       " (126, ('with lysine chloromethyl ketone', 0.0)),\n",
       " (127, ('a novel inhibitors of p56', 0.0)),\n",
       " (128, ('a very low carbohydrate ketogenic', 0.0)),\n",
       " (129, ('and 2 are no enzyme', 0.0)),\n",
       " (130, ('cell specific protein tyrosine kinase', 0.0)),\n",
       " (131, ('colocalized protein were several disrupted', 0.0)),\n",
       " (132, ('composed of lymphocytespecific tyrosine kinase', 0.0)),\n",
       " (133, ('egfr = 45 ic 50', 0.0)),\n",
       " (134, ('log leukaemic stem cell kill', 0.0)),\n",
       " (135, ('maximum gross log cell kill', 0.0)),\n",
       " (136, ('network src kinase lymphocyte kinase', 0.0)),\n",
       " (137, ('of the lymphocyte specific kinase', 0.0)),\n",
       " (138, ('oncogene src family tyrosine kinase', 0.0)),\n",
       " (139, ('phosphorylation of phosphorylation p56 lck', 0.0)),\n",
       " (140, ('proto oncogene tyrosine protein kinase', 0.0)),\n",
       " (141, ('related tyrosine kinase p56 lck', 0.0)),\n",
       " (142, ('responses the lymphocyte specific kinase', 0.0)),\n",
       " (143, ('such as lymphocyte specific kinase', 0.0)),\n",
       " (144, ('to simultaneous phosphorylation of p56', 0.0)),\n",
       " (145, ('trypsin with lysine chloromethyl ketone', 0.0)),\n",
       " (146, ('very low carbohydrate ketogenic diet', 0.0)),\n",
       " (147, ('− mice expressing wild type', 0.0)),\n",
       " (148, ('1 and 2 are no enzyme', 0.0)),\n",
       " (149, ('50 egfr = 45 ic 50', 0.0)),\n",
       " (150, ('activation lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (151, ('also a novel inhibitors of p56', 0.0)),\n",
       " (152, ('as lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (153, ('bound leukocyte specific protein tyrosine kinase', 0.0)),\n",
       " (154, ('bovine trypsin with lysine chloromethyl ketone', 0.0)),\n",
       " (155, ('by lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (156, ('cd247 leukocyte specific protein tyrosine kinase', 0.0)),\n",
       " (157, ('hypotheses lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (158, ('immune responses the lymphocyte specific kinase', 0.0)),\n",
       " (159, ('induced phosphorylation of phosphorylation p56 lck', 0.0)),\n",
       " (160, ('inf2 lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (161, ('is lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (162, ('kinase such as lymphocyte specific kinase', 0.0)),\n",
       " (163, ('led to simultaneous phosphorylation of p56', 0.0)),\n",
       " (164, ('libitum very low carbohydrate ketogenic diet', 0.0)),\n",
       " (165, ('mean maximum gross log cell kill', 0.0)),\n",
       " (166, ('network composed of lymphocytespecific tyrosine kinase', 0.0)),\n",
       " (167, ('of colocalized protein were several disrupted', 0.0)),\n",
       " (168, ('or lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (169, ('ppi network src kinase lymphocyte kinase', 0.0)),\n",
       " (170, ('proto oncogene src family tyrosine kinase', 0.0)),\n",
       " (171, ('recruitment of the lymphocyte specific kinase', 0.0)),\n",
       " (172, ('regulated lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (173, ('signaling lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (174, ('t lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (175, ('that lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (176, ('the log leukaemic stem cell kill', 0.0)),\n",
       " (177, ('the proto oncogene tyrosine protein kinase', 0.0)),\n",
       " (178, ('the related tyrosine kinase p56 lck', 0.0)),\n",
       " (179, ('to a very low carbohydrate ketogenic', 0.0)),\n",
       " (180, ('to lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (181, ('via lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (182, ('with lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (183, ('− − mice expressing wild type', 0.0)),\n",
       " (184, ('16 and lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (185, ('18 induced phosphorylation of phosphorylation p56 lck', 0.0)),\n",
       " (186, ('25 to a very low carbohydrate ketogenic', 0.0)),\n",
       " (187, ('7 the log leukaemic stem cell kill', 0.0)),\n",
       " (188, ('90 to lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (189,\n",
       "  ('activation activation lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (190, ('ad libitum very low carbohydrate ketogenic diet', 0.0)),\n",
       " (191, ('adaptive immune responses the lymphocyte specific kinase', 0.0)),\n",
       " (192, ('and the lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (193, ('by inf2 lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (194, ('cell is lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (195, ('cell led to simultaneous phosphorylation of p56', 0.0)),\n",
       " (196, ('cell signaling lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (197, ('for the related tyrosine kinase p56 lck', 0.0)),\n",
       " (198, ('human t lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (199, ('ic 50 egfr = 45 ic 50', 0.0)),\n",
       " (200, ('identified that lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (201, ('interaction of lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (202, ('intracellular recruitment of the lymphocyte specific kinase', 0.0)),\n",
       " (203, ('is also a novel inhibitors of p56', 0.0)),\n",
       " (204, ('itk or lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (205, ('lanes 1 and 2 are no enzyme', 0.0)),\n",
       " (206, ('macrophages of lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (207, ('membrane bound leukocyte specific protein tyrosine kinase', 0.0)),\n",
       " (208, ('of bovine trypsin with lysine chloromethyl ketone', 0.0)),\n",
       " (209, ('of cd247 leukocyte specific protein tyrosine kinase', 0.0)),\n",
       " (210, ('of the proto oncogene tyrosine protein kinase', 0.0)),\n",
       " (211, ('patches of colocalized protein were several disrupted', 0.0)),\n",
       " (212,\n",
       "  ('phosphorylation by lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (213,\n",
       "  ('positively regulated lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (214, ('producing mean maximum gross log cell kill', 0.0)),\n",
       " (215, ('rag2 − − mice expressing wild type', 0.0)),\n",
       " (216, ('recruitment of lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (217, ('resistance via lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (218, ('ror2 and lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (219, ('sub network composed of lymphocytespecific tyrosine kinase', 0.0)),\n",
       " (220, ('such as lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (221,\n",
       "  ('testable hypotheses lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (222, ('the lymphocyte cell specific protein tyrosine kinase', 0.0)),\n",
       " (223, ('the ppi network src kinase lymphocyte kinase', 0.0)),\n",
       " (224, ('together with lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (225, ('tyrosine kinase such as lymphocyte specific kinase', 0.0)),\n",
       " (226, ('upregulation of lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (227, ('via the lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (228, ('with lymphocyte cell specific protein tyrosine kinase', 0.0)),\n",
       " (229, ('zap70 and lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (230, ('> 25 to a very low carbohydrate ketogenic', 0.0)),\n",
       " (231, ('about 7 the log leukaemic stem cell kill', 0.0)),\n",
       " (232, ('activation of the proto oncogene tyrosine protein kinase', 0.0)),\n",
       " (233, ('akt via the lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (234, ('alkylation of bovine trypsin with lysine chloromethyl ketone', 0.0)),\n",
       " (235, ('an ad libitum very low carbohydrate ketogenic diet', 0.0)),\n",
       " (236,\n",
       "  ('and identified that lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (237,\n",
       "  ('and testable hypotheses lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (238, ('arrays the lymphocyte cell specific protein tyrosine kinase', 0.0)),\n",
       " (239, ('as patches of colocalized protein were several disrupted', 0.0)),\n",
       " (240,\n",
       "  ('associates with lymphocyte cell specific protein tyrosine kinase', 0.0)),\n",
       " (241,\n",
       "  ('by intracellular recruitment of the lymphocyte specific kinase', 0.0)),\n",
       " (242, ('cd247 16 and lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (243,\n",
       "  ('cisplatin resistance via lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (244,\n",
       "  ('cyld positively regulated lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (245, ('described for the related tyrosine kinase p56 lck', 0.0)),\n",
       " (246, ('fyn together with lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (247, ('i rag2 − − mice expressing wild type', 0.0)),\n",
       " (248, ('il 18 induced phosphorylation of phosphorylation p56 lck', 0.0)),\n",
       " (249, ('in the ppi network src kinase lymphocyte kinase', 0.0)),\n",
       " (250,\n",
       "  ('is phosphorylation by lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (251, ('it is also a novel inhibitors of p56', 0.0)),\n",
       " (252, ('kinase zap70 and lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (253, ('lamr1 and the lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (254, ('nm lanes 1 and 2 are no enzyme', 0.0)),\n",
       " (255,\n",
       "  ('not macrophages of lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (256, ('of ror2 and lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (257, ('protein 90 to lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (258, ('receptor tyrosine kinase such as lymphocyte specific kinase', 0.0)),\n",
       " (259,\n",
       "  ('revealed upregulation of lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (260, ('stat3 lck proto oncogene src family tyrosine kinase', 0.0)),\n",
       " (261, ('stk10 ic 50 egfr = 45 ic 50', 0.0)),\n",
       " (262, ('t cell is lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (263, ('t cell led to simultaneous phosphorylation of p56', 0.0)),\n",
       " (264,\n",
       "  ('targetable sub network composed of lymphocytespecific tyrosine kinase',\n",
       "   0.0)),\n",
       " (265,\n",
       "  ('tcr activation activation lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (266, ('the human t lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (267,\n",
       "  ('the interaction of lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (268, ('the membrane bound leukocyte specific protein tyrosine kinase', 0.0)),\n",
       " (269,\n",
       "  ('the recruitment of lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (270, ('to adaptive immune responses the lymphocyte specific kinase', 0.0)),\n",
       " (271,\n",
       "  ('trafficked by inf2 lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (272, ('tumor producing mean maximum gross log cell kill', 0.0)),\n",
       " (273, ('zap70 itk or lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (274, ('γ1 cell signaling lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (275, ('200 nm lanes 1 and 2 are no enzyme', 0.0)),\n",
       " (276, ('3 stat3 lck proto oncogene src family tyrosine kinase', 0.0)),\n",
       " (277, ('50 stk10 ic 50 egfr = 45 ic 50', 0.0)),\n",
       " (278,\n",
       "  ('activation akt via the lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (279,\n",
       "  ('analysis revealed upregulation of lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (280, ('and activation of the proto oncogene tyrosine protein kinase', 0.0)),\n",
       " (281,\n",
       "  ('and targetable sub network composed of lymphocytespecific tyrosine kinase',\n",
       "   0.0)),\n",
       " (282, ('bmi > 25 to a very low carbohydrate ketogenic', 0.0)),\n",
       " (283,\n",
       "  ('but not macrophages of lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (284,\n",
       "  ('capacity by intracellular recruitment of the lymphocyte specific kinase',\n",
       "   0.0)),\n",
       " (285,\n",
       "  ('car is phosphorylation by lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (286,\n",
       "  ('endogenous t cell is lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (287,\n",
       "  ('expressing levels of cd247 leukocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (288,\n",
       "  ('family kinase such as lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (289,\n",
       "  ('fundamental to adaptive immune responses the lymphocyte specific kinase',\n",
       "   0.0)),\n",
       " (290,\n",
       "  ('gene fyn together with lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (291, ('gene of lck proto oncogene src family tyrosine kinase', 0.0)),\n",
       " (292, ('in zap70 itk or lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (293,\n",
       "  ('insights and testable hypotheses lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (294, ('interaction of lck proto oncogene src family tyrosine kinase', 0.0)),\n",
       " (295,\n",
       "  ('is the membrane bound leukocyte specific protein tyrosine kinase', 0.0)),\n",
       " (296,\n",
       "  ('kinase trafficked by inf2 lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (297,\n",
       "  ('levels of ror2 and lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (298, ('m5076 tumor producing mean maximum gross log cell kill', 0.0)),\n",
       " (299, ('microbe it is also a novel inhibitors of p56', 0.0)),\n",
       " (300,\n",
       "  ('nitrotyrosine and identified that lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (301, ('of about 7 the log leukaemic stem cell kill', 0.0)),\n",
       " (302, ('of t cell led to simultaneous phosphorylation of p56', 0.0)),\n",
       " (303, ('of the human t lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (304, ('ot i rag2 − − mice expressing wild type', 0.0)),\n",
       " (305,\n",
       "  ('parallel cisplatin resistance via lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (306,\n",
       "  ('plc γ1 cell signaling lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (307,\n",
       "  ('protein arrays the lymphocyte cell specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (308,\n",
       "  ('pyap associates with lymphocyte cell specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (309,\n",
       "  ('receptor lamr1 and the lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (310, ('responses in the ppi network src kinase lymphocyte kinase', 0.0)),\n",
       " (311,\n",
       "  ('sensitive as patches of colocalized protein were several disrupted', 0.0)),\n",
       " (312,\n",
       "  ('shock protein 90 to lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (313,\n",
       "  ('since cyld positively regulated lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (314,\n",
       "  ('specific il 18 induced phosphorylation of phosphorylation p56 lck', 0.0)),\n",
       " (315,\n",
       "  ('synapse the interaction of lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (316, ('that an ad libitum very low carbohydrate ketogenic diet', 0.0)),\n",
       " (317,\n",
       "  ('the alkylation of bovine trypsin with lysine chloromethyl ketone', 0.0)),\n",
       " (318, ('those described for the related tyrosine kinase p56 lck', 0.0)),\n",
       " (319,\n",
       "  ('transcript levels of cd247 leukocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (320,\n",
       "  ('tyrosine kinase zap70 and lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (321,\n",
       "  ('upon tcr activation activation lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (322,\n",
       "  ('various kinase such as lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (323,\n",
       "  ('with receptor tyrosine kinase such as lymphocyte specific kinase', 0.0)),\n",
       " (324,\n",
       "  ('with the recruitment of lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (325, ('ζ cd247 16 and lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (326,\n",
       "  ('18 specific il 18 induced phosphorylation of phosphorylation p56 lck',\n",
       "   0.0)),\n",
       " (327,\n",
       "  ('2006 since cyld positively regulated lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (328,\n",
       "  ('3 nitrotyrosine and identified that lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (329,\n",
       "  ('6 transcript levels of cd247 leukocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (330,\n",
       "  ('activation and targetable sub network composed of lymphocytespecific tyrosine kinase',\n",
       "   0.0)),\n",
       " (331, ('as those described for the related tyrosine kinase p56 lck', 0.0)),\n",
       " (332,\n",
       "  ('cascade is the membrane bound leukocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (333,\n",
       "  ('cell but not macrophages of lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (334,\n",
       "  ('cell upon tcr activation activation lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (335,\n",
       "  ('detergent sensitive as patches of colocalized protein were several disrupted',\n",
       "   0.0)),\n",
       " (336,\n",
       "  ('downregulated gene of lck proto oncogene src family tyrosine kinase',\n",
       "   0.0)),\n",
       " (337, ('factor of about 7 the log leukaemic stem cell kill', 0.0)),\n",
       " (338,\n",
       "  ('for the alkylation of bovine trypsin with lysine chloromethyl ketone',\n",
       "   0.0)),\n",
       " (339, ('from ot i rag2 − − mice expressing wild type', 0.0)),\n",
       " (340,\n",
       "  ('gene expressing levels of cd247 leukocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (341,\n",
       "  ('heat shock protein 90 to lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (342, ('ic 50 stk10 ic 50 egfr = 45 ic 50', 0.0)),\n",
       " (343,\n",
       "  ('immune responses in the ppi network src kinase lymphocyte kinase', 0.0)),\n",
       " (344,\n",
       "  ('immunological synapse the interaction of lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (345,\n",
       "  ('in endogenous t cell is lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (346,\n",
       "  ('in levels of ror2 and lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (347,\n",
       "  ('in parallel cisplatin resistance via lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (348,\n",
       "  ('interaction with receptor tyrosine kinase such as lymphocyte specific kinase',\n",
       "   0.0)),\n",
       " (349,\n",
       "  ('is fundamental to adaptive immune responses the lymphocyte specific kinase',\n",
       "   0.0)),\n",
       " (350,\n",
       "  ('kinase gene fyn together with lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (351,\n",
       "  ('laminin receptor lamr1 and the lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (352, ('lck 200 nm lanes 1 and 2 are no enzyme', 0.0)),\n",
       " (353,\n",
       "  ('lckm of the human t lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (354,\n",
       "  ('ligation activation akt via the lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (355,\n",
       "  ('like in zap70 itk or lymphocyte specific protein tyrosine kinase', 0.0)),\n",
       " (356,\n",
       "  ('motif pyap associates with lymphocyte cell specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (357,\n",
       "  ('new insights and testable hypotheses lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (358,\n",
       "  ('nucleus interaction of lck proto oncogene src family tyrosine kinase',\n",
       "   0.0)),\n",
       " (359, ('of microbe it is also a novel inhibitors of p56', 0.0)),\n",
       " (360,\n",
       "  ('phase protein arrays the lymphocyte cell specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (361,\n",
       "  ('phosphorylation various kinase such as lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (362,\n",
       "  ('proximal tyrosine kinase zap70 and lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (363,\n",
       "  ('receptor ζ cd247 16 and lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (364,\n",
       "  ('recruitment and activation of the proto oncogene tyrosine protein kinase',\n",
       "   0.0)),\n",
       " (365,\n",
       "  ('signaling capacity by intracellular recruitment of the lymphocyte specific kinase',\n",
       "   0.0)),\n",
       " (366,\n",
       "  ('signaling plc γ1 cell signaling lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (367,\n",
       "  ('src family kinase such as lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (368,\n",
       "  ('starts with the recruitment of lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (369,\n",
       "  ('stimulation of t cell led to simultaneous phosphorylation of p56', 0.0)),\n",
       " (370,\n",
       "  ('suggests that an ad libitum very low carbohydrate ketogenic diet', 0.0)),\n",
       " (371, ('the m5076 tumor producing mean maximum gross log cell kill', 0.0)),\n",
       " (372,\n",
       "  ('transcript 3 stat3 lck proto oncogene src family tyrosine kinase', 0.0)),\n",
       " (373,\n",
       "  ('transcriptomic analysis revealed upregulation of lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (374,\n",
       "  ('tyrosine kinase trafficked by inf2 lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (375, ('weight bmi > 25 to a very low carbohydrate ketogenic', 0.0)),\n",
       " (376,\n",
       "  ('ζ car is phosphorylation by lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (377,\n",
       "  ('16 gene expressing levels of cd247 leukocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (378,\n",
       "  ('4c starts with the recruitment of lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (379,\n",
       "  ('= 6 transcript levels of cd247 leukocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (380, ('a factor of about 7 the log leukaemic stem cell kill', 0.0)),\n",
       " (381,\n",
       "  ('acetylated heat shock protein 90 to lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (382,\n",
       "  ('addition downregulated gene of lck proto oncogene src family tyrosine kinase',\n",
       "   0.0)),\n",
       " (383,\n",
       "  ('al 2006 since cyld positively regulated lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (384,\n",
       "  ('an activation and targetable sub network composed of lymphocytespecific tyrosine kinase',\n",
       "   0.0)),\n",
       " (385,\n",
       "  ('and in parallel cisplatin resistance via lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (386,\n",
       "  ('become detergent sensitive as patches of colocalized protein were several disrupted',\n",
       "   0.0)),\n",
       " (387, ('body weight bmi > 25 to a very low carbohydrate ketogenic', 0.0)),\n",
       " (388,\n",
       "  ('cd3 ζ car is phosphorylation by lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (389, ('cell from ot i rag2 − − mice expressing wild type', 0.0)),\n",
       " (390,\n",
       "  ('cell receptor ζ cd247 16 and lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (391,\n",
       "  ('cell signaling plc γ1 cell signaling lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (392, ('death of microbe it is also a novel inhibitors of p56', 0.0)),\n",
       " (393,\n",
       "  ('decrease in levels of ror2 and lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (394, ('egfr ic 50 stk10 ic 50 egfr = 45 ic 50', 0.0)),\n",
       " (395, ('for lck 200 nm lanes 1 and 2 are no enzyme', 0.0)),\n",
       " (396,\n",
       "  ('form lckm of the human t lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (397,\n",
       "  ('il 18 specific il 18 induced phosphorylation of phosphorylation p56 lck',\n",
       "   0.0)),\n",
       " (398,\n",
       "  ('in immune responses in the ppi network src kinase lymphocyte kinase',\n",
       "   0.0)),\n",
       " (399, ('in the m5076 tumor producing mean maximum gross log cell kill', 0.0)),\n",
       " (400,\n",
       "  ('induced stimulation of t cell led to simultaneous phosphorylation of p56',\n",
       "   0.0)),\n",
       " (401,\n",
       "  ('involve src family kinase such as lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (402,\n",
       "  ('it interaction with receptor tyrosine kinase such as lymphocyte specific kinase',\n",
       "   0.0)),\n",
       " (403,\n",
       "  ('like tyrosine kinase trafficked by inf2 lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (404,\n",
       "  ('lrrc8a ligation activation akt via the lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (405,\n",
       "  ('mouse laminin receptor lamr1 and the lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (406,\n",
       "  ('of 3 nitrotyrosine and identified that lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (407,\n",
       "  ('of transcript 3 stat3 lck proto oncogene src family tyrosine kinase',\n",
       "   0.0)),\n",
       " (408,\n",
       "  ('presenting cell upon tcr activation activation lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (409,\n",
       "  ('proline motif pyap associates with lymphocyte cell specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (410,\n",
       "  ('protein kinase gene fyn together with lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (411,\n",
       "  ('providing new insights and testable hypotheses lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (412,\n",
       "  ('receptor phosphorylation various kinase such as lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (413,\n",
       "  ('research suggests that an ad libitum very low carbohydrate ketogenic diet',\n",
       "   0.0)),\n",
       " (414,\n",
       "  ('reverse phase protein arrays the lymphocyte cell specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (415,\n",
       "  ('signaling cascade is the membrane bound leukocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (416,\n",
       "  ('signaling like in zap70 itk or lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (417,\n",
       "  ('such as those described for the related tyrosine kinase p56 lck', 0.0)),\n",
       " (418,\n",
       "  ('t cell but not macrophages of lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (419,\n",
       "  ('tcr is fundamental to adaptive immune responses the lymphocyte specific kinase',\n",
       "   0.0)),\n",
       " (420,\n",
       "  ('tcr proximal tyrosine kinase zap70 and lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (421,\n",
       "  ('the immunological synapse the interaction of lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (422,\n",
       "  ('the nucleus interaction of lck proto oncogene src family tyrosine kinase',\n",
       "   0.0)),\n",
       " (423,\n",
       "  ('the recruitment and activation of the proto oncogene tyrosine protein kinase',\n",
       "   0.0)),\n",
       " (424,\n",
       "  ('their signaling capacity by intracellular recruitment of the lymphocyte specific kinase',\n",
       "   0.0)),\n",
       " (425,\n",
       "  ('values for the alkylation of bovine trypsin with lysine chloromethyl ketone',\n",
       "   0.0)),\n",
       " (426,\n",
       "  ('which in endogenous t cell is lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (427,\n",
       "  ('0 16 gene expressing levels of cd247 leukocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (428,\n",
       "  ('a decrease in levels of ror2 and lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (429,\n",
       "  ('activation in the m5076 tumor producing mean maximum gross log cell kill',\n",
       "   0.0)),\n",
       " (430,\n",
       "  ('activation of transcript 3 stat3 lck proto oncogene src family tyrosine kinase',\n",
       "   0.0)),\n",
       " (431, ('by a factor of about 7 the log leukaemic stem cell kill', 0.0)),\n",
       " (432,\n",
       "  ('by il 18 specific il 18 induced phosphorylation of phosphorylation p56 lck',\n",
       "   0.0)),\n",
       " (433,\n",
       "  ('by reverse phase protein arrays the lymphocyte cell specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (434,\n",
       "  ('cd28 cd3 ζ car is phosphorylation by lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (435,\n",
       "  ('derivation of 3 nitrotyrosine and identified that lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (436,\n",
       "  ('distal proline motif pyap associates with lymphocyte cell specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (437,\n",
       "  ('elevated body weight bmi > 25 to a very low carbohydrate ketogenic', 0.0)),\n",
       " (438,\n",
       "  ('et al 2006 since cyld positively regulated lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (439,\n",
       "  ('fig 4c starts with the recruitment of lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (440, ('for egfr ic 50 stk10 ic 50 egfr = 45 ic 50', 0.0)),\n",
       " (441,\n",
       "  ('harbor an activation and targetable sub network composed of lymphocytespecific tyrosine kinase',\n",
       "   0.0)),\n",
       " (442,\n",
       "  ('in addition downregulated gene of lck proto oncogene src family tyrosine kinase',\n",
       "   0.0)),\n",
       " (443,\n",
       "  ('in t cell but not macrophages of lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (444,\n",
       "  ('increase their signaling capacity by intracellular recruitment of the lymphocyte specific kinase',\n",
       "   0.0)),\n",
       " (445,\n",
       "  ('inhibitors such as those described for the related tyrosine kinase p56 lck',\n",
       "   0.0)),\n",
       " (446,\n",
       "  ('into the immunological synapse the interaction of lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (447,\n",
       "  ('kp values for the alkylation of bovine trypsin with lysine chloromethyl ketone',\n",
       "   0.0)),\n",
       " (448,\n",
       "  ('myristylated form lckm of the human t lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (449,\n",
       "  ('n = 6 transcript levels of cd247 leukocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (450,\n",
       "  ('of acetylated heat shock protein 90 to lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (451,\n",
       "  ('of which in endogenous t cell is lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (452,\n",
       "  ('previous research suggests that an ad libitum very low carbohydrate ketogenic diet',\n",
       "   0.0)),\n",
       " (453,\n",
       "  ('receptor in immune responses in the ppi network src kinase lymphocyte kinase',\n",
       "   0.0)),\n",
       " (454,\n",
       "  ('receptor tcr is fundamental to adaptive immune responses the lymphocyte specific kinase',\n",
       "   0.0)),\n",
       " (455,\n",
       "  ('signaling and in parallel cisplatin resistance via lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (456,\n",
       "  ('src like tyrosine kinase trafficked by inf2 lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (457, ('substrates for lck 200 nm lanes 1 and 2 are no enzyme', 0.0)),\n",
       " (458, ('t cell from ot i rag2 − − mice expressing wild type', 0.0)),\n",
       " (459,\n",
       "  ('t cell receptor ζ cd247 16 and lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (460,\n",
       "  ('tcr induced stimulation of t cell led to simultaneous phosphorylation of p56',\n",
       "   0.0)),\n",
       " (461,\n",
       "  ('tcr signaling cascade is the membrane bound leukocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (462,\n",
       "  ('tcr signaling like in zap70 itk or lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (463,\n",
       "  ('that it interaction with receptor tyrosine kinase such as lymphocyte specific kinase',\n",
       "   0.0)),\n",
       " (464,\n",
       "  ('that lrrc8a ligation activation akt via the lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (465, ('the death of microbe it is also a novel inhibitors of p56', 0.0)),\n",
       " (466,\n",
       "  ('the mouse laminin receptor lamr1 and the lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (467,\n",
       "  ('the tcr proximal tyrosine kinase zap70 and lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (468,\n",
       "  ('to become detergent sensitive as patches of colocalized protein were several disrupted',\n",
       "   0.0)),\n",
       " (469,\n",
       "  ('to involve src family kinase such as lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (470,\n",
       "  ('to the nucleus interaction of lck proto oncogene src family tyrosine kinase',\n",
       "   0.0)),\n",
       " (471,\n",
       "  ('to the recruitment and activation of the proto oncogene tyrosine protein kinase',\n",
       "   0.0)),\n",
       " (472,\n",
       "  ('tyrosine protein kinase gene fyn together with lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (473,\n",
       "  ('γ1 cell signaling plc γ1 cell signaling lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (474, ('+ t cell from ot i rag2 − − mice expressing wild type', 0.0)),\n",
       " (475,\n",
       "  ('68 tcr induced stimulation of t cell led to simultaneous phosphorylation of p56',\n",
       "   0.0)),\n",
       " (476,\n",
       "  ('= 0 16 gene expressing levels of cd247 leukocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (477,\n",
       "  ('and activation of transcript 3 stat3 lck proto oncogene src family tyrosine kinase',\n",
       "   0.0)),\n",
       " (478,\n",
       "  ('and elevated body weight bmi > 25 to a very low carbohydrate ketogenic',\n",
       "   0.0)),\n",
       " (479,\n",
       "  ('and kp values for the alkylation of bovine trypsin with lysine chloromethyl ketone',\n",
       "   0.0)),\n",
       " (480,\n",
       "  ('and that lrrc8a ligation activation akt via the lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (481,\n",
       "  ('and the mouse laminin receptor lamr1 and the lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (482, ('as substrates for lck 200 nm lanes 1 and 2 are no enzyme', 0.0)),\n",
       " (483,\n",
       "  ('as t cell receptor ζ cd247 16 and lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (484,\n",
       "  ('binding of acetylated heat shock protein 90 to lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (485,\n",
       "  ('black n = 6 transcript levels of cd247 leukocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (486,\n",
       "  ('cell receptor tcr is fundamental to adaptive immune responses the lymphocyte specific kinase',\n",
       "   0.0)),\n",
       " (487,\n",
       "  ('expressing in t cell but not macrophages of lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (488,\n",
       "  ('identified by reverse phase protein arrays the lymphocyte cell specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (489,\n",
       "  ('important of which in endogenous t cell is lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (490,\n",
       "  ('in tcr signaling like in zap70 itk or lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (491,\n",
       "  ('induced by il 18 specific il 18 induced phosphorylation of phosphorylation p56 lck',\n",
       "   0.0)),\n",
       " (492,\n",
       "  ('intake previous research suggests that an ad libitum very low carbohydrate ketogenic diet',\n",
       "   0.0)),\n",
       " (493,\n",
       "  ('kinase inhibitors such as those described for the related tyrosine kinase p56 lck',\n",
       "   0.0)),\n",
       " (494,\n",
       "  ('leading to the recruitment and activation of the proto oncogene tyrosine protein kinase',\n",
       "   0.0)),\n",
       " (495,\n",
       "  ('lines harbor an activation and targetable sub network composed of lymphocytespecific tyrosine kinase',\n",
       "   0.0)),\n",
       " (496,\n",
       "  ('mapk3 in addition downregulated gene of lck proto oncogene src family tyrosine kinase',\n",
       "   0.0)),\n",
       " (497,\n",
       "  ('marrow by a factor of about 7 the log leukaemic stem cell kill', 0.0)),\n",
       " (498,\n",
       "  ('non myristylated form lckm of the human t lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (499,\n",
       "  ('of receptor in immune responses in the ppi network src kinase lymphocyte kinase',\n",
       "   0.0)),\n",
       " (500,\n",
       "  ('of the tcr proximal tyrosine kinase zap70 and lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (501,\n",
       "  ('one src like tyrosine kinase trafficked by inf2 lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (502,\n",
       "  ('patches to become detergent sensitive as patches of colocalized protein were several disrupted',\n",
       "   0.0)),\n",
       " (503,\n",
       "  ('plc γ1 cell signaling plc γ1 cell signaling lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (504,\n",
       "  ('receptor to the nucleus interaction of lck proto oncogene src family tyrosine kinase',\n",
       "   0.0)),\n",
       " (505,\n",
       "  ('recognition fig 4c starts with the recruitment of lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (506,\n",
       "  ('reported to involve src family kinase such as lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (507,\n",
       "  ('reproducibly activation in the m5076 tumor producing mean maximum gross log cell kill',\n",
       "   0.0)),\n",
       " (508,\n",
       "  ('revealed a decrease in levels of ror2 and lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (509, ('selectivity for egfr ic 50 stk10 ic 50 egfr = 45 ic 50', 0.0)),\n",
       " (510,\n",
       "  ('showed that it interaction with receptor tyrosine kinase such as lymphocyte specific kinase',\n",
       "   0.0)),\n",
       " (511,\n",
       "  ('the cd28 cd3 ζ car is phosphorylation by lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (512,\n",
       "  ('the distal proline motif pyap associates with lymphocyte cell specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (513,\n",
       "  ('the tcr signaling cascade is the membrane bound leukocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (514, ('to the death of microbe it is also a novel inhibitors of p56', 0.0)),\n",
       " (515,\n",
       "  ('0 and elevated body weight bmi > 25 to a very low carbohydrate ketogenic',\n",
       "   0.0)),\n",
       " (516,\n",
       "  ('a non myristylated form lckm of the human t lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (517,\n",
       "  ('activation 68 tcr induced stimulation of t cell led to simultaneous phosphorylation of p56',\n",
       "   0.0)),\n",
       " (518,\n",
       "  ('activation of receptor in immune responses in the ppi network src kinase lymphocyte kinase',\n",
       "   0.0)),\n",
       " (519,\n",
       "  ('alterations in tcr signaling like in zap70 itk or lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (520,\n",
       "  ('been reported to involve src family kinase such as lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (521,\n",
       "  ('bone marrow by a factor of about 7 the log leukaemic stem cell kill',\n",
       "   0.0)),\n",
       " (522,\n",
       "  ('cell expressing in t cell but not macrophages of lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (523,\n",
       "  ('cell receptor to the nucleus interaction of lck proto oncogene src family tyrosine kinase',\n",
       "   0.0)),\n",
       " (524, ('high selectivity for egfr ic 50 stk10 ic 50 egfr = 45 ic 50', 0.0)),\n",
       " (525,\n",
       "  ('in the tcr signaling cascade is the membrane bound leukocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (526,\n",
       "  ('is induced by il 18 specific il 18 induced phosphorylation of phosphorylation p56 lck',\n",
       "   0.0)),\n",
       " (527,\n",
       "  ('k and kp values for the alkylation of bovine trypsin with lysine chloromethyl ketone',\n",
       "   0.0)),\n",
       " (528,\n",
       "  ('leading to the death of microbe it is also a novel inhibitors of p56',\n",
       "   0.0)),\n",
       " (529,\n",
       "  ('lrrc8a and that lrrc8a ligation activation akt via the lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (530,\n",
       "  ('most important of which in endogenous t cell is lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (531,\n",
       "  ('on the cd28 cd3 ζ car is phosphorylation by lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (532,\n",
       "  ('p = 0 16 gene expressing levels of cd247 leukocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (533,\n",
       "  ('potent kinase inhibitors such as those described for the related tyrosine kinase p56 lck',\n",
       "   0.0)),\n",
       " (534,\n",
       "  ('s black n = 6 transcript levels of cd247 leukocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (535,\n",
       "  ('screen revealed a decrease in levels of ror2 and lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (536,\n",
       "  ('serve as substrates for lck 200 nm lanes 1 and 2 are no enzyme', 0.0)),\n",
       " (537,\n",
       "  ('state of the tcr proximal tyrosine kinase zap70 and lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (538,\n",
       "  ('such as t cell receptor ζ cd247 16 and lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (539,\n",
       "  ('t cell receptor tcr is fundamental to adaptive immune responses the lymphocyte specific kinase',\n",
       "   0.0)),\n",
       " (540,\n",
       "  ('tcr leading to the recruitment and activation of the proto oncogene tyrosine protein kinase',\n",
       "   0.0)),\n",
       " (541,\n",
       "  ('the binding of acetylated heat shock protein 90 to lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (542,\n",
       "  ('was reproducibly activation in the m5076 tumor producing mean maximum gross log cell kill',\n",
       "   0.0)),\n",
       " (543,\n",
       "  ('22 k and kp values for the alkylation of bovine trypsin with lysine chloromethyl ketone',\n",
       "   0.0)),\n",
       " (544,\n",
       "  ('6 0 and elevated body weight bmi > 25 to a very low carbohydrate ketogenic',\n",
       "   0.0)),\n",
       " (545,\n",
       "  ('8 serve as substrates for lck 200 nm lanes 1 and 2 are no enzyme', 0.0)),\n",
       " (546,\n",
       "  ('demonstrates high selectivity for egfr ic 50 stk10 ic 50 egfr = 45 ic 50',\n",
       "   0.0)),\n",
       " (547,\n",
       "  ('discrimination cd8 + t cell from ot i rag2 − − mice expressing wild type',\n",
       "   0.0)),\n",
       " (548,\n",
       "  ('experiments cd8 + t cell from ot i rag2 − − mice expressing wild type',\n",
       "   0.0)),\n",
       " (549,\n",
       "  ('femoral bone marrow by a factor of about 7 the log leukaemic stem cell kill',\n",
       "   0.0)),\n",
       " (550,\n",
       "  ('growth leading to the death of microbe it is also a novel inhibitors of p56',\n",
       "   0.0)),\n",
       " (551,\n",
       "  ('pathway such as t cell receptor ζ cd247 16 and lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (552,\n",
       "  ('sites on the cd28 cd3 ζ car is phosphorylation by lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (553,\n",
       "  ('t cell expressing in t cell but not macrophages of lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (554,\n",
       "  ('t cell receptor to the nucleus interaction of lck proto oncogene src family tyrosine kinase',\n",
       "   0.0)),\n",
       " (555,\n",
       "  ('the most important of which in endogenous t cell is lymphocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (556,\n",
       "  ('the tcr leading to the recruitment and activation of the proto oncogene tyrosine protein kinase',\n",
       "   0.0)),\n",
       " (557,\n",
       "  ('vav3 p = 0 16 gene expressing levels of cd247 leukocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (558,\n",
       "  ('1 22 k and kp values for the alkylation of bovine trypsin with lysine chloromethyl ketone',\n",
       "   0.0)),\n",
       " (559,\n",
       "  ('6 8 serve as substrates for lck 200 nm lanes 1 and 2 are no enzyme', 0.0)),\n",
       " (560,\n",
       "  ('> 6 0 and elevated body weight bmi > 25 to a very low carbohydrate ketogenic',\n",
       "   0.0)),\n",
       " (561,\n",
       "  ('for vav3 p = 0 16 gene expressing levels of cd247 leukocyte specific protein tyrosine kinase',\n",
       "   0.0)),\n",
       " (562,\n",
       "  ('independent experiments cd8 + t cell from ot i rag2 − − mice expressing wild type',\n",
       "   0.0)),\n",
       " (563,\n",
       "  ('ligand discrimination cd8 + t cell from ot i rag2 − − mice expressing wild type',\n",
       "   0.0)),\n",
       " (564,\n",
       "  ('microbe growth leading to the death of microbe it is also a novel inhibitors of p56',\n",
       "   0.0)),\n",
       " (565,\n",
       "  ('still demonstrates high selectivity for egfr ic 50 stk10 ic 50 egfr = 45 ic 50',\n",
       "   0.0)),\n",
       " (566,\n",
       "  ('the femoral bone marrow by a factor of about 7 the log leukaemic stem cell kill',\n",
       "   0.0)),\n",
       " (567,\n",
       "  ('to the tcr leading to the recruitment and activation of the proto oncogene tyrosine protein kinase',\n",
       "   0.0)),\n",
       " (568,\n",
       "  ('4 6 8 serve as substrates for lck 200 nm lanes 1 and 2 are no enzyme',\n",
       "   0.0)),\n",
       " (569,\n",
       "  ('hba1c > 6 0 and elevated body weight bmi > 25 to a very low carbohydrate ketogenic',\n",
       "   0.0)),\n",
       " (570,\n",
       "  ('in the femoral bone marrow by a factor of about 7 the log leukaemic stem cell kill',\n",
       "   0.0)),\n",
       " (571,\n",
       "  ('inhibit microbe growth leading to the death of microbe it is also a novel inhibitors of p56',\n",
       "   0.0)),\n",
       " (572,\n",
       "  ('it still demonstrates high selectivity for egfr ic 50 stk10 ic 50 egfr = 45 ic 50',\n",
       "   0.0)),\n",
       " (573,\n",
       "  ('tcr ligand discrimination cd8 + t cell from ot i rag2 − − mice expressing wild type',\n",
       "   0.0)),\n",
       " (574,\n",
       "  ('three independent experiments cd8 + t cell from ot i rag2 − − mice expressing wild type',\n",
       "   0.0)),\n",
       " (575,\n",
       "  ('lanes 4 6 8 serve as substrates for lck 200 nm lanes 1 and 2 are no enzyme',\n",
       "   0.0)),\n",
       " (576,\n",
       "  ('nm it still demonstrates high selectivity for egfr ic 50 stk10 ic 50 egfr = 45 ic 50',\n",
       "   0.0)),\n",
       " (577,\n",
       "  ('of three independent experiments cd8 + t cell from ot i rag2 − − mice expressing wild type',\n",
       "   0.0)),\n",
       " (578,\n",
       "  ('μm lanes 4 6 8 serve as substrates for lck 200 nm lanes 1 and 2 are no enzyme',\n",
       "   0.0)),\n",
       " (579,\n",
       "  ('1 μm lanes 4 6 8 serve as substrates for lck 200 nm lanes 1 and 2 are no enzyme',\n",
       "   0.0)),\n",
       " (580,\n",
       "  ('k390r 1 μm lanes 4 6 8 serve as substrates for lck 200 nm lanes 1 and 2 are no enzyme',\n",
       "   0.0)),\n",
       " (581,\n",
       "  ('itk k390r 1 μm lanes 4 6 8 serve as substrates for lck 200 nm lanes 1 and 2 are no enzyme',\n",
       "   0.0))]"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(enumerate(top))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "longforms0.extend([x for i, x in enumerate(top) if i in []])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('+', 83.90816326530613),\n",
       " ('cells', 12.588235294117647),\n",
       " ('hspcs', 5.5),\n",
       " ('progenitor', 4.0),\n",
       " ('hspc', 3.0),\n",
       " ('positive', 1.5),\n",
       " ('hi', 1.3333333333333333),\n",
       " ('populations', 1.3333333333333333)]"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "longforms0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [],
   "source": [
    "longforms0.sort(key=lambda x: -x[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [],
   "source": [
    "longforms, scores = zip(*longforms0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('+', 'cells', 'hspcs', 'progenitor', 'hspc', 'positive', 'hi', 'populations')"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "longforms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [],
   "source": [
    "grounding_map = {}\n",
    "names = {}\n",
    "for longform in longforms:\n",
    "    grounding = gilda_ground(longform)\n",
    "    if grounding[0]:\n",
    "        grounding_map[longform] = f'{grounding[0]}:{grounding[1]}'\n",
    "        names[grounding_map[longform]] = grounding[2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'cells': 'GO:GO:0005623',\n",
       " 'hspcs': 'HGNC:9536',\n",
       " 'hspc': 'HGNC:9536',\n",
       " 'hi': 'CHEBI:CHEBI:141437',\n",
       " 'populations': 'MESH:D011153'}"
      ]
     },
     "execution_count": 120,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "grounding_map"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'MESH:D012156': 'Reticulocytes'}"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [],
   "source": [
    "grounding_map, names, pos_labels = ground_with_gui(longforms, scores, grounding_map=grounding_map, names=names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [],
   "source": [
    "result = (grounding_map, names, pos_labels)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "({'+': 'ungrounded',\n",
       "  'cells': 'ungrounded',\n",
       "  'hi': 'ungrounded',\n",
       "  'hspc': 'ungrounded',\n",
       "  'hspcs': 'ungrounded',\n",
       "  'populations': 'ungrounded',\n",
       "  'positive': 'ungrounded',\n",
       "  'progenitor': 'ungrounded'},\n",
       " {},\n",
       " [])"
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [],
   "source": [
    "grounding_map, names, pos_labels = ({'+': 'ungrounded',\n",
    "  'cells': 'ungrounded',\n",
    "  'hi': 'ungrounded',\n",
    "  'hspc': 'ungrounded',\n",
    "  'hspcs': 'ungrounded',\n",
    "  'populations': 'ungrounded',\n",
    "  'positive': 'ungrounded',\n",
    "  'progenitor': 'ungrounded'},\n",
    " {},\n",
    " [])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [],
   "source": [
    "grounding_dict = {'LSK': grounding_map}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [],
   "source": [
    "classifier = AdeftClassifier('LSK', pos_labels=pos_labels)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [],
   "source": [
    "param_grid = {'C': [100.0], 'max_features': [10000]}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [],
   "source": [
    "labeler = AdeftLabeler(grounding_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [],
   "source": [
    "corpus = labeler.build_from_texts(shortform_texts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [],
   "source": [
    "corpus.extend(entrez_texts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [],
   "source": [
    "texts, labels = zip(*corpus)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [],
   "source": [
    "names['HGNC:6524'] = 'LCK'\n",
    "pos_labels.append('HGNC:6524')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO: [2020-01-17 15:07:57] /Users/albertsteppi/adeft/adeft/modeling/classify.py - Beginning grid search in parameter space:\n",
      "{'C': [100.0], 'max_features': [10000]}\n"
     ]
    }
   ],
   "source": [
    "classifier.cv(texts, labels, param_grid, cv=5, n_jobs=8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'label_distribution': {'ungrounded': 92, 'HGNC:9967': 872},\n",
       " 'f1': {'mean': 0.9920518376807328, 'std': 0.004873855908121181},\n",
       " 'precision': {'mean': 0.984275512009176, 'std': 0.00961380996783067},\n",
       " 'recall': {'mean': 1.0, 'std': 0.0}}"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "classifier.stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "disamb = AdeftDisambiguator(classifier, grounding_dict, names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "d = disamb.disambiguate(shortform_texts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [],
   "source": [
    "a = [text for pred, text in zip(d, shortform_texts)if not pred[0].startswith('HGNC')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "113"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(a)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "y =[x for x in a if 'tyrosine kinase' in x]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Non-small-cell lung cancer (NSCLC) patients with mutated or rearranged oncogene drivers can be treated with upfront selective inhibitors achieving higher response rates and longer survival than chemotherapy. The RET gene can undergo chromosomal rearrangements in 1%–2% of all NSCLC patients, involving various upstream fusion partners such as KIF5B, CCDC6, NCOA4, and TRIM33. Many multikinase inhibitors are active against rearranged RET. Cabozantinib, vandetanib, sunitinib, lenvatinib, and nintedanib achieved tumor responses in about 30% of these patients in retrospective studies. Prospective phase II trials investigated the activity and toxicity of cabozantinib, vandetanib, sorafenib, and lenvatinib, and did not reach significantly higher response rates. VEGFR and EGFR inhibition represented the main ways of developing off-target toxicity. An intrinsic resistance emerged according to the type of RET fusion partners, as KIF5B-RET fusion is the most resistant. Also acquired mutations in rearranged RET oncogene developed as resistance to these multikinase inhibitors. Interestingly, RET fusions have been found as a resistance mechanism to EGFR-TKIs in EGFR-mutant NSCLC patients. The combination of EGFR and RET inhibition can overcome this resistance. The limitations in terms of activity and tolerability of the various multikinase inhibitors prompted the investigation of new highly selective RET inhibitors, such as RXDX-105, BLU-667, and LOXO-292. Some data emerged about intracranial antitumor activity of BLU-667 and LOXO-292. If these novel drugs will achieve high activity in RET rearranged NSCLC, also these oncogene-addicted tumors can undergo a significant survival improvement.\\nIn all these cases, even in those with longer progression-free survival, resistance occurs. To date many resistance mechanisms have been known and, as a consequence, novel targeted drugs have been developed to deal with the inefficacy of previous treatment. Resistance mutations can be detected through the re-biopsy of tumor tissue or the so-called “liquid biopsy”, intended as the analysis of blood samples to find DNA alterations. 3  Among the various oncogene drivers in NSCLC the  RET  gene is involved in various chromosomal rearrangements, which can be found in 1%–2% of all NSCLC patients. 4  The current availability of RET inhibitors raises the possibility to spare this small proportion of NSCLC patients from chemotherapy and offer the opportunity of a further effective targeted therapy after failure of chemotherapy.\\nIn this review, we discuss the biological significance of the  RET  gene, the available RET-directed drugs, and relative clinical trials for NSCLC patients and resistance mechanisms occurring during the treatment with RET inhibitors.\\nIn 1985,  RET  was identified as a novel transforming gene as result of transfection of the NIH 3T3 cell with high molecular weight DNA of a human T-cell lymphoma. The gene was activated by a DNA rearrangement in which two unlinked segments of human DNA recombined to generate a new transcriptional unit. 5  Subsequently, studies mapped RET to chromosome 10q11.2, where it encodes for a receptor tyrosine kinase. 6\\nMutations of RET determine the absence of enteric ganglia from the distal colon (Hirschsprung’s disease) and congenital megacolon, besides RET.k-/RET.k- mice lack all enteric ganglia, demonstrating a RET important role in the development of the enteric nervous system. 7  Still in the embryonic phase a RET-dependent cell rearrangement generates a specialized epithelial domain that later emerges as the tip of the primary ureteric bud. 8  As well as in the embryonic phase RET is important in homeostasis of several tissues including neural, neuroendocrine, hematopoietic, and male germ tissues. 9\\nRET is a single-pass transmembrane protein with a typical intracellular tyrosine kinase domain ( Figure 1 ). While a “classical” activation of a receptor tyrosine kinase (RTK) is due to the interaction ligand-receptor, the activation of RET requires an interaction between its ligands (the glial cell line-derived neurotrophic factor-family ligands, GFLs) and a co-receptor (GFLs family receptor-alfa). 5  The GFL–GFRα complex binds to the extracellular domain of RET, leading to the phosphorylation of the intracellular tyrosine kinase domain and consequently the activation of several pathways, including MAPK, PI3K, JAK-STAT, PKA, and PKC. 10\\nMultiple endocrine neoplasia (MEN) syndrome is defined as a disorder with neoplasms in two or more different hormonal tissues. Each of these syndromes also includes one or more additional neoplasms. MEN1 is characterized by specific hormonal tumors: parathyroid adenoma (90%), gastrinoma (40%), and prolactinoma (30%), plus additional hormone-producing tumor with a range between 1% and 10% (insulinoma, glucagonoma, VIPoma, somatostatinoma, pituitary tumors, thymic carcinoid, bronchial carcinoid, gastric carcinoid, adrenal cortex, and pheochromocytoma). 11  The  MEN1  gene, mapped at chromosome 11q13, encodes for a protein, menin, that is involved in organogenesis of neural tube, heart, and craniofacial structures and hematopoiesis. 12  A germline mutation of the MEN1 leads to tumor development mainly via a biallelic loss-of-function mechanism. 11  MEN2A are characterized by medullary thyroid cancer, pheochromocytoma, and hyperparathyroidism. MEN2B has the same features as MEN2A, plus intestinal ganglioneuromas and the mucosal neuroma phenotype, but a different hormonal disorder profile. 13  During the 90 seconds, the International RET Mutation Consortium enrolled 477 independent MEN2 families worldwide to investigate the association between the position and type of germline mutation in the RET proto-oncogene and the presence or absence of medullary thyroid carcinoma, pheochromocytoma, primary hyperparathyroidism, and other. Overall, 92% of MEN2 families had a germline RET mutation in one of eight codons. Over 95% of families with MEN2B had a mutation in codon 918, and each of the three MEN2A categories were found to have mutations at cysteine codons 609, 611, 618, 620, and 634. 14  These mutations determine a ligand-independent constitutive activation of the tyrosine kinase receptor leading to an uncontrolled activation of the MAPK and the PI3K pathways that results in uncontrolled growth and cell de-differentiation. 15  In addition to the RET point mutations, several gene rearrangements were identified in papillary thyroid carcinoma (PTC), known as RET/PTC rearrangements. Each distinct chromosomal translocation is characterized by the promoter and 5′ region of a heterologous gene encoding a thyrocyte-expressed protein fused, in frame, to the kinase-encoding 3′ end of the RET proto-oncogene. 16  To date, 13 different oncogenic RET/PTC fusion proteins (termed RET/PTC1-PTC9) have been discovered. These chimeric proteins are characterized by coiled-coil domains that induce dimerization and activation of the kinase domain. This capability to form dimers is required for oncogenic activation, leading to an uncontrolled activation of the MAPK and PI3K pathways, similarly to the result of activating RET point mutations. 15 , 17  Interestingly, RET rearrangement are largely thought to be somatic events, as opposed to RET mutations that can occur in the germ line or be acquired somatically.\\nFirst, in 2011 RET rearrangement was discovered in a young never-smoking male patient with lung adenocarcinoma. 18  As aforementioned, 1%–2% of NSCLCs harbor a RET rearrangement. 4  While the correlation between ionic radiation and RET rearrangements is clearly confirmed in PTC, in NSCLC this correlation remains unclear, even if in vitro experiments demonstrated the possibility to induce RET rearrangement in human lung cancer cells by radiation. 19  To deepen the knowledge of RET in NSCLC, Wang et al 20  examined the  RET  fusion gene in 936 patients with surgically resected NSCLC using a PCR strategy. These patients seemed to have identifiable clinicopathologic characteristics, including younger age, never-smoker status, early lymph node metastases, poor differentiation, and a solid-predominant subtype. As well as the other main driving-mutations, RET rearrangement seems to be mutually exclusive, suggesting that it might be a targetable oncogenic driver. 21\\nThe first RET rearrangement identified in NSCLC patients was an in-frame fusion transcript of the kinesin family 5B gene ( KIF5B ) with  RET  gene (KIF5B-RET). Other upstream fusion partners for RET rearrangement have been identified in NSCLC, such as the coiled-coil domain-containing protein 6 ( CCDC6 ), the nuclear receptor coactivator 4 ( NCOA4 ), the tripartite motif-containing 33 ( TRIM33 ), myosin VC gene ( MYO5C ), EPH receptor A5 gene ( EPHA5 ), CAP-Gly domain containing linker protein family member one gene ( CLIP1 ), ELKS/RAB6-interacting/CAST family member one gene ( ERC1 ), phosphatidylinositol binding clathrin assembly protein gene ( PICALM ), FERM domain containing 4A gene ( FRMD4A ) RUN and RYVE domain containing two gene ( RUFY2 ), and tripartite motif containing 24 gene ( TRIM24 ). All of these fusion counterparts have a dimerization domain that induces ligand-independent activation of the RET kinase ( Figure 1 ). 22 , 23\\nThe majority of drugs active against RET are multikinase inhibitors. The approval of these agents is not restricted to patients harboring alterations in  RET  gene. Among these drugs, we can mention those approved for thyroid cancer, such as cabozantinib, vandetanib, lenvatinib, and sorafenib, but also other multikinase inhibitors approved for other malignancies, including alectinib, sunitinib, nintedanib, regorafenib, and ponatinib. The action of these drugs against RET kinase allows a classification in type I and type II inhibitors. The first ones (ie, vandetanib and sunitinib) bind the ATP-binding domain in its active conformation. The latter ones bind the same domain, but in the inactive conformation. 24 , 25\\nThe effects of these drugs on RET rearrangements with various upstream partners or breakpoints were studied by means of engineered and patient-derived RET-rearranged cell lines and xenograft models. 26  These experimental models have been useful to identify the activity of other agents against RET rearranged kinase (eg, RXDX-105, BLU-667, LOXO-292). 27 – 30  These studies demonstrated the effect on rearranged RET kinase, but the multikinase inhibition also induced the blockade of many downstream pathways such as MAPK, PI3K, and PLCγ with a consequent decrease of cell proliferation.\\nIn 2015, a retrospective analysis was performed on data from the Global Multicenter RET Registry (GLORY), which collected the experiences of the treatment with multikinase inhibitors in RET-rearranged NSCLC patients. 31  Interesting findings emerge from this analysis. Multikinase RET inhibitors were administered in various lines of systemic therapy ranging from first to eighth, with a median of third line. In fact, median time from diagnosis to the beginning of anti-RET therapy was 12 months. Among the various drugs only cabozantinib, vandetanib, sunitinib, lenvatinib, and nintedanib achieved tumor responses, ~30%, whereas no responses were observed with alectinib, regorafenib, sorafenib, or ponatinib ( Figure 2A ). None of the outcome measures (response rate, progression-free survival [PFS], overall survival [OS]) changed depending on upstream fusion partners (eg, KIF5B, CCDC6, EPHA5) of  RET  gene. Median PFS of 2.3 months and median OS of 6.8 months were reported. The majority of patients (about 80%) received only one multikinase RET inhibitor. Moreover, this registry also provides information about the efficacy of first-line platinum-based chemotherapy in RET-rearranged NSCLC, which reached about 50%. These results in terms of response rate and PFS are partially concordant with those from phase II trials, which had studied or were studying the same drugs in RET-rearranged NSCLC patients.\\nTo date in this subpopulation, five phase II trials with multikinase RET inhibitors have been completed ( Figure 2B ). One single arm phase II trial studied cabozantinib, a multikinase inhibitor active against VEGFR2, MET, ROS1, AXL, KIT, and TIE2, but with low activity against RET (IC50=5.2 nM). 32  The patients in this study were not previously treated with RET inhibitors. About one-third of these patients responded to cabozantinib, but no complete responses were observed. Moreover, responses were early, with a high tumor shrinkage (≥30% tumor reduction in 70% of patients). The median overall survival reached 9.9 months.\\nVandetanib is another multikinase inhibitor against VEG-FRs, EGFR, and RET, with higher IC50 than cabozantinib. It was studied in two phase II trials conducted in Eastern countries (Korea and Japan). 33 , 34  The objective response rate (ORR) was 18% and 53%, for median PFS of 4.5 and 4.7 months, respectively. The most common adverse effects of vandetanib were hypertension because of VEGFR inhibition, skin toxicity and diarrhea because of EGFR inhibition, and also manageable prolonged QT interval. Interestingly in these studies the differences in the types of upstream fusion partners of  RET  gene were documented. In the Korean study, the KIF5B-RET rearrangement was associated with no objective response, unlike CCDC6-RET fusion. They also found the novel rearrangement MYO5C-RET, which is characterized by the exclusion of RET transmembrane domain with consequent ligand-independent RET activation. Accordingly in the Japanese trial higher ORR (83% vs 20%) and longer PFS (8.3 vs 2.9 months) were achieved with CCDC6-RET than KIF5B-RET rearrangements.\\nAlso sorafenib was studied in a phase II trial for NSCLC patients with RET rearrangement. 35  Sorafenib is a multikinase inhibitor which targets intracellular (ie, CRAF, BRAF and mutated BRAF) and cell surface (ie, KIT, FLT3, VEGFRs and RET) molecules. It has anti-RET activity with IC50=5.9–47 nM. In a preclinical model, sorafenib was active against KIF5B-RET fusion. 36  In the only three patients treated with sorafenib in this study, no significant response was reported, but tumor shrinkage and symptom improvement were observed along with durable stable disease in one patient.\\nFinally, lenvatinib, a multikinase inhibitor against VEGFRs, PDGFR-β, and RET, has an IC50 for anti-RET activity of 35 nM. It was evaluated for antitumor activity in RET fusion positive patients with lung adenocarcinoma within a phase II study. Among 25 patients, ORR was 16% and median PFS 7.3 months. Grade 3–4 adverse events were experienced in 92% of patients, with hypertension, nausea, anorexia, diarrhea, proteinuria, and vomiting as the most common toxicities. 37\\nThe results of these clinical trials with multikinase inhibitors have revealed that not all RET-rearranged patients are responsive to these drugs. Objective response rates of 28% and 47%, 32 , 34  respectively obtained with cabozantinib and vandetanib, suggest the existence of intrinsic resistance mechanisms. First of all, the type of RET fusion partner seems to influence the response to treatment, as documented with vandetanib, which induced worse outcomes in the presence of KIF5B-RET fusion than with CCDC6-RET fusion. However, the phase II study with cabozantinib did not confirm this finding. Some preclinical studies showed that the presence of KIF5B upstream of RET induces an increase of RET transcription, whereas the presence of other fusion partners, such as CCDC6, is associated with a lower degree of RET expression. This aspect could influence the response to RET inhibition, 38 , 39  but more studies should be performed to address this issue. Moreover, KIF5B-RET fusion can strongly activate a signaling hub of various kinases, such as RET, EGFR, SRC, FGFR. This effect was not observed for other RET rearrangements (ie, CCDC6-RET and NCOA4-RET) and can be overcome through inhibiting both RET and EGFR, as Das and Cagan 40  demonstrated in an in vitro study. These findings are corroborated in vivo, as reviewed by Ferrara et al. 23  Patients with KIF5B-RET fusion mainly achieved lower overall response rates than patients with other RET fusions.\\nRecently, through other in vitro experiments, some missense RET mutations have been found associated with resistance to cabozantinib, lenvatinib, and vandetanib in patients with medullary thyroid cancer. In particular, the authors found a number of TKI-resistant mutations located in the Gly-rich loop (L730, E732, and V738), the gatekeeper residue (V804), or the hinge strand (Y806, A807, and G810) that comprise about two-thirds of the drug binding pocket. They also demonstrated that different aminoacid substitutions of the same site could have different consequences for drug resistance, suggesting that not only the site of the kinase but also the type of amino acid should be considered when evaluating drug sensitivity. 41\\nHowever, to date these mutations have not been observed in tumor samples from patients with RET-rearranged or RET-mutant cancer that have developed an acquired resistance to RET inhibitors. Recently, a case report showed a RET-rearranged lung cancer patient that developed a secondary RET mutation (S904F) at the onset of resistance to vandetanib. The mutation was a serine-to-phenylalanine substitution at codon 904 in the activation loop of the RET kinase domain and conferred resistance to vandetanib by increasing the ATP affinity and autophosphorylation activity of RET kinase. This result indicates that missense mutations in the activation loop of the kinase domain are able to increase kinase activity and confer drug resistance through allosteric effects. 42  Another case report described the effect of V804M gatekeeper mutation on resistance to vandetanib. 43\\nAs we know, in ALK rearranged patients, the frequency of ALK mutations as a resistance mechanism is higher in those patients receiving second generation ALK-inhibitors (alectinib or ceritinib), more potent than first generation inhibitors such as crizotinib, 44  suggesting that the potency of the drug in inducing a block of the target is associated with induction of resistance mutations on the target. This could suggest that more potent anti-RET agents could induce RET mutation, also indicating a more effective anti-RET activity, with respect to the current used agents.\\nConcurrent genomic alterations could also have a role in conferring resistance. Preclinical studies in a RET-rearranged lung cancer cell line showed that the EGFR-mediated signaling could mediate resistance to multikinase inhibition providing a rationale to cotarget EGFR to reduce the risks of developing drug resistance. 45 , 46  In the study by Chang et al, 45  CCDC6-RET positive lung cancer cells were highly sensitive to RET inhibition, but EGFR signaling was responsible for resistance to sunitinib, vandetanib, and sorafenib, through the induction of ERK and AKT activity. Moreover, they demonstrated that endothelial cells, which are known to produce EGF, decreased the sensitivity of RET inhibitors. In addition, the results of Vaishnavi et al, 46  performed on different models of lung cancer cell lines carrying ALK, ROS1, RET, and NTRK1 fusions, confirm that EGFR signaling was involved at different levels in determining resistance to multikinase inhibitors, and that treatment with the EGFR-TKI gefinitb abrogated all the EGFR contributions.\\nAnother potential resistance mechanism, both primary and acquired, is the MDM2 amplification. Pre- and post tumor biopsy were obtained from RET-rearranged lung cancer patients treated with cabozantinib, and amplification of the gene was observed in 50% of patients undergoing resistance, highlighting the possibility to combine anti-MDM2 agents with RET inhibitors. 47\\nAnother potential resistant mechanism to RET inhibition, observed in in vitro studies, is the activation of the MAPK pathway. By studying RET-rearranged cell lines treated with ponatinib, cell clones resistant to the drug were obtained and the molecular characteristics of those was studied. Although cells retained the expression of the RET fusion, phosphorylation lacked. The activation of the MAPK signaling was seen in both cell lines, in one case due to the induction of NRAS mutation and in the other model due to the overexpression of EGFR and AXL. 48  These results should be confirmed in a clinical setting and, if confirmed, could open the possibility for combinational treatment using MAPK-inhibiting drugs ( Table 1 ).\\nThe results in terms of activity and tolerability of the various multikinase inhibitors, active but not specific against RET, led us to investigate a new anti-RET specific kinase inhibitor, RXDX-105. Its IC50 is 0.3, 0.3–0.8, and 5–15 nM against wild-type RET, RET rearrangements, and mutated RET, respectively. VEGFRs are spared from inhibition by this drug. It was evaluated in a phase I trial including 28 RET-fusion positive NSCLC patients. The most common G3 adverse events reached no more than 10% and no G4 toxicity was reported. None of the patients with KIF5B-RET fusion experienced a response, whereas among patients with non-KIF5B-RET fusions ORR was 75%, suggesting a relevant role of patient selection according to rearrangement type. 49\\nBLU-667 is a novel small-molecule RET inhibitor. It has been designed for high potency and selectivity against oncogenic RET alterations, including the most frequent RET rearrangements (eg, KIF5B–RET and CCDC6–RET). It was tested preclinically in RET-driven thyroid, lung, and colorectal cancers. KIF5B–RET autophosphorylation was inhibited by BLU-667 in vitro over 20-fold more potently than RXDX-105. Durable responses without significant off-target toxicity in patients with RET-altered NSCLC and medullary thyroid cancer prompted clinical investigation. 29  A phase one open-label first-in-human study was designed for BLU-667 in patients with medullary thyroid cancer, RET-altered NSCLC, and other RET-altered solid tumors, and has been recruiting currently (NCT03037385). 50\\nLOXO-292 is another highly selective ATP-competitive RET inhibitor, which has nanomolar potency against diverse RET alterations, including anticipated acquired resistance mutations. It has advantageous pharmacokinetic features, such as high bioavailability, significant penetration of central nervous system, and low potential for drug interactions. 30  Patients with advanced RET-altered tumors, treated with any prior multikinase inhibitors, including lung cancer, are under recruitment in a phase 1/2, open-label, first-in-human study with LOXO-292 (NCT03157128). 51  In the meantime, a study of patients treated with LOXO-292 showed a rapid clearance of RET variants in cell-free DNA ( Table 2 ). 52\\nFew studies reported information on the frequency, responsiveness, and overall outcomes in RET-rearranged advanced NSCLC patients with central nervous system (CNS) metastases. A recent paper by Drilon et al 53  focused on this topic. They showed that the frequency of CNS involvement in these patients is 25% at diagnosis, but lifetime prevalence can reach almost a half. Furthermore, the cumulative incidence of CNS lesions in RET-positive NSCLC patients is higher than ROS1-positive and lower than ALK-positive patients. They found a low intracranial response rate when these patients were treated with various multikinase inhibitors (alectinib, cabozantinib, ponatinib, sunitinib, vandetanib, vandetanib + everolimus): two of eleven patients (18%), one treated with alectinib and one with vandetanib + everolimus. In both these patients with responding CNS metastases, CCDC6-RET fusion was present. PFS and OS were short in patients with brain metastases: 2.1 months (95% CI =1.3–2.9 months) and 3.9 months (95% CI =1.9–5.4), respectively. However, these outcomes can echo the limited efficacy of multikinase inhibitors in RET-rearranged NSCLC patients. The combination of vandetanib and everolimus can represent an option to optimize blood–brain-barrier penetration as previously reported. 54  Some data about intracranial antitumor activity through the selective RET-directed inhibitors BLU-667 and LOXO-292 are emerging. 55 , 56\\nFirst, in 2015, some authors 57  reported two cases of EGFR-mutant NSCLC patients treated with erlotinib, who developed CCDC6-RET fusion as detected via a hybrid-capture-based comprehensive genomic profiling assay in tumor tissue from rebiopsy. In another case, a retrospective analysis of the Foundation Medicine database allowed the identification of an acquired NCOA4-RET fusion in a NSCLC patient progressing on EGFR-TKI. 57\\nMore recently, the Foundation Medicine database was explored to identify EGFR-mutant NSCLC patients. The tumor and blood samples were analyzed for BRAF or RTK fusions. In four patients, three RET fusions were found (ie, CCDC6-RET, NCOA4-RET, and TRIM24-RET). After the appearance of these RET rearrangements during an EGFR-TKI, RET inhibitors were delivered. One patient with CCDC6-RET fusion post-erlotinib had no benefit to single-agent alectinib. Another patient with NCOA4-RET fusion post-afatinib experienced stable disease through the combination of afatinib and cabozantib. 58\\nAn in vitro model with EGFR-mutant lung cancer cell lines expressing CCDC6-RET showed that the combination of an EGFR-TKI with the selective inhibitor BLU-667 achieved a decrease in cell viability. In the same paper by Piotrowska et al, 59  some cases of EGFR-mutant NSCLC patients treated with combined EGFR and RET inhibition after the occurrence of RET fusions are reported. One patient with CCDC6-RET fusion post-afatinib was treated with erlotinib plus cabozantinib, but obtained no significant benefit. In two other patients, one with CCDC6-RET post-osimertinib and one with NCOA4-RET post-afatinib/cetuximab, the combination of osimertinib and BLU-667 achieved significant tumor response with marked tumor shrinkage.\\nThese reports suggest that a selective RET inhibition combined with EGFR-TKI could help to manage acquired resistance to EGFR-TKIs when RET fusions are documented as a resistance mechanism. However, specific clinical trials are needed to recommend this as a standard approach.\\nThe  RET  gene is one of the already known oncogenes undergoing activating rearrangements in a small subpopulation of lung cancer patients. The availability of multikinase inhibitors, active against RET among various targets, encouraged us to find a target therapy also for these patients. By using multikinase inhibitors to target RET, the consequent concomitant VEGFR and EGFR inhibitions lead to off-target toxicity. As a consequence these mutikinase inhibitors cannot be delivered at the dose necessary for RET inhibition. Both retrospective and prospective studies showed a good activity of some multikinase inhibitors in RET fusion positive NSCLC patients, but not sufficient to consider these drugs as a valid alternative to chemotherapy as achieved by EGFR- and ALK-inhibitors in other oncogene-addicted tumors. From these studies an intrinsic resistance emerged according to the type of RET fusion partners. Moreover, some acquired resistance mutations in rearranged RET were found during the treatment with multikinase inhibitors. Nowadays encouraging prospects derive from the development of selective RET inhibitors with high potency, but without off-target toxicity. Some early phase clinical trials are ongoing, giving the hope that soon new drugs will be available to specifically treat those 1%–2% of NSCLC patients with a RET rearrangement, sparing them from first-line chemotherapy.\\nSchematic structure of wild-type and rearranged RET proteins in a cancer cell.\\nSummary of known mechanisms of resistance to RET inhibition\\nSummary of new RET-specific drugs\\n'"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "y[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [],
   "source": [
    "disamb.dump('RET', '../results')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from adeft.disambiguate import load_disambiguator"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "d = load_disambiguator('ARG', '../results')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "d.disambiguate(texts[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(d.info())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "a = load_disambiguator('AR')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "a.disambiguate('Androgen')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "logit = d.classifier.estimator.named_steps['logit']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "logit.classes_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [],
   "source": [
    "model_to_s3(disamb)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
