{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import pickle\n",
    "\n",
    "from indra.literature.adeft_tools import universal_extract_text\n",
    "from indra.databases.hgnc_client import get_hgnc_name, get_hgnc_id\n",
    "\n",
    "from indra_db.util.content_scripts import get_text_content_from_pmids\n",
    "from indra_db.util.content_scripts import get_stmts_with_agent_text_like\n",
    "from indra_db.util.content_scripts import get_text_content_from_stmt_ids\n",
    "\n",
    "from adeft.discover import AdeftMiner\n",
    "from adeft.gui import ground_with_gui\n",
    "from adeft.modeling.label import AdeftLabeler\n",
    "from adeft.modeling.classify import AdeftClassifier\n",
    "from adeft.disambiguate import AdeftDisambiguator\n",
    "\n",
    "from adeft_indra.s3 import model_to_s3\n",
    "from adeft_indra.ground import gilda_ground"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "shortforms = ['CLK']\n",
    "genes = ['CLOCK']\n",
    "families = {'CLK': ['CLK1', 'CLK2', 'CLK3', 'CLK4']}\n",
    "groundings = [f'HGNC:{get_hgnc_id(gene)}' for gene in genes]\n",
    "for family, members in families.items():\n",
    "    genes.extend(members)\n",
    "    groundings.append(f'FPLX:{family}')\n",
    "with open('../data/entrez_all_pmids.json', 'r') as f:\n",
    "    all_pmids = json.load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['CLOCK', 'CLK1', 'CLK2', 'CLK3', 'CLK4']"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "entrez_texts = []\n",
    "entrez_refs = set()\n",
    "for gene, grounding in zip(genes, groundings):\n",
    "    try:\n",
    "        pmids = all_pmids[gene]\n",
    "    except KeyError:\n",
    "        continue\n",
    "    _, content = get_text_content_from_pmids(pmids)\n",
    "    entrez_texts.extend([(universal_extract_text(text), grounding)\n",
    "                          for text in content.values() if text])\n",
    "    entrez_refs.update(content.keys())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "miners = dict()\n",
    "all_texts = set()\n",
    "for shortform in shortforms:\n",
    "    stmts = get_stmts_with_agent_text_like(shortform)[shortform]\n",
    "    _, content = get_text_content_from_stmt_ids(stmts)\n",
    "    shortform_texts = [universal_extract_text(text, contains=shortforms)\n",
    "                       for ref, text in content.items() if text and ref not in entrez_refs]\n",
    "    miners[shortform] = AdeftMiner(shortform)\n",
    "    miners[shortform].process_texts(shortform_texts)\n",
    "    all_texts |= set(shortform_texts)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "It's then necessary to check if Acromine produced the correct results. We must fix errors manually"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "top = miners['CLK'].top()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('clock', 89.51428571428572),\n",
       " ('transcription factors clock', 22.756756756756758),\n",
       " ('the transcription factors clock', 19.476190476190474),\n",
       " ('the clock', 9.09090909090909),\n",
       " ('cdc2 like kinase', 6.0),\n",
       " ('the transcription activators clock', 5.5),\n",
       " ('and clock', 4.666666666666667),\n",
       " ('of clock', 4.0),\n",
       " ('of the transcription factors clock', 4.0),\n",
       " ('kidney', 3.0),\n",
       " ('the proteins clock', 3.0),\n",
       " ('two transcription factors clock', 3.0),\n",
       " ('bhlh pas transcription factors clock', 2.5),\n",
       " ('of the transcription activators clock', 2.5),\n",
       " ('contralateral kidney', 2.0),\n",
       " ('by the transcription factors clock', 2.0),\n",
       " ('activators clock', 1.8000000000000007),\n",
       " ('transcription activators clock', 1.7777777777777777),\n",
       " ('like kinase', 1.75),\n",
       " ('proteins clock', 1.6),\n",
       " ('timeless tim clock', 1.5),\n",
       " ('by clock', 1.3333333333333333),\n",
       " ('by the clock', 1.3333333333333333),\n",
       " ('contra lateral kidney', 1.3333333333333333),\n",
       " ('per timeless tim clock', 1.3333333333333333),\n",
       " ('drosophila the transcription factors clock', 1.3333333333333333),\n",
       " ('contra lateral', 1.0),\n",
       " ('tfs clock', 1.0),\n",
       " ('e g clock', 1.0),\n",
       " ('impaired contralateral kidney', 1),\n",
       " ('loop the clock', 1.0),\n",
       " ('of the clock', 1.0),\n",
       " ('and contra lateral kidney', 1.0),\n",
       " ('domain transcription factors clock', 1.0),\n",
       " ('heterodimer formed by clock', 1.0),\n",
       " ('in drosophila melanogaster clock', 1.0),\n",
       " ('in mammals and clock', 1.0),\n",
       " ('the positive elements clock', 1),\n",
       " ('is activators by the clock', 1.0),\n",
       " ('period per timeless tim clock', 1.0),\n",
       " ('the circadian transcription factors clock', 1.0),\n",
       " ('in drosophila the transcription factors clock', 1.0),\n",
       " ('post translational regulated of the transcription factors clock', 1),\n",
       " ('the heterodimeric complex of the transcription factors clock', 1),\n",
       " ('in mammals a complex formed by the proteins clock', 1),\n",
       " ('in drosophila the activators is a heterodimeric complex of clock', 1),\n",
       " ('the core feedback loop consists of the transcription activators clock',\n",
       "  1.0),\n",
       " ('ock transcription translational feedback loop in drosophila the transcription factors clock',\n",
       "  1),\n",
       " ('similarly to srpk1 2 full length clones encoding cdc2 like kinase', 1),\n",
       " ('a transcription translational feedback loop consists of transcription activators e g clock',\n",
       "  1),\n",
       " ('autoregulate their expression by inhibit activators of their transcription activators the clock',\n",
       "  1),\n",
       " ('ila melanogaster molecular biological studies have shown that the transcription activators clock',\n",
       "  1),\n",
       " ('imeless tim transcription figure 1 in the per tim loop the clock', 1),\n",
       " ('as dexamethasone dex reduces concanavalin a con a induced product of chemotactic lymphokines',\n",
       "  1),\n",
       " ('briefly in drosophila the core molecular oscillator consists of two transcription factors clock',\n",
       "  1),\n",
       " ('chung 2010 hardin 2011 in drosophila the bhlh pas transcription factors heterodimer clock',\n",
       "  1),\n",
       " ('hardin 2011 in drosophila several transcription loop are interconnected by the tfs clock',\n",
       "  1),\n",
       " ('lar oscillator e75 repress expression of the genes encoding the transcription activators clock',\n",
       "  1),\n",
       " ('nal feedback loop underlies animal circadian rhythms in drosophila the transcription factors clock',\n",
       "  1),\n",
       " ('o keep temporal order and anticipate daily environmental changes in drosophila melanogaster clock',\n",
       "  1),\n",
       " ('rotein oscillator in constant darkness dd these rhythms require the transcription factors clock',\n",
       "  1),\n",
       " ('tional translational feedback loop ttfl in the drosophila ttfl the transcription factors clock',\n",
       "  1),\n",
       " ('ts control the transcription of the negative elements briefly the transcription activators clock',\n",
       "  1),\n",
       " ('two basic helix loop helix per arnt sim bhlh pas transcription factors clock',\n",
       "  1.0),\n",
       " ('activators the expression of their own inhibitors in the drosophila circadian oscillator the clock',\n",
       "  1),\n",
       " ('al 2003 within the nucleus per repress the activators of the transcription activators clock',\n",
       "  1),\n",
       " ('and tissues in drosophila the core feedback loop consists of the transcription activators clock',\n",
       "  1),\n",
       " ('anism relies on transcription activators and repressor that constitute a feedback loop the clock',\n",
       "  1),\n",
       " ('circadian pacemaker this molecular clock dependent on the activators of the transcription factors clock',\n",
       "  1),\n",
       " ('ck genes that formed intracellular molecular clock in these clock two transcription factors clock',\n",
       "  1),\n",
       " ('clock clk and cycle cyc and their product proteins period per timeless tim clock',\n",
       "  1),\n",
       " ('clock genes cryptochrome 2 cry2 which encoding the main transcription repressor of the clock',\n",
       "  1),\n",
       " ('cription translational feedback loop they include in flies four major proteins timeless tim clock',\n",
       "  1),\n",
       " ('cxcr4 expression in c ardiosphere derived l in − c k it + progenitor', 1.0),\n",
       " ('e core feedback loop period per and timeless tim transcription are activators when clock',\n",
       "  1),\n",
       " ('especially for myc driven cancer patients member of the evolutionarily conserved cdc2 like kinase',\n",
       "  1),\n",
       " ('h include transcription negative feedback loop that are conserved from insects to mammals clock',\n",
       "  1),\n",
       " ('ila circadian oscillator is comprised of transcription feedback loop that are activators by clock',\n",
       "  1),\n",
       " ('it relies on a negative transcription feedback loop that involves the transcription factors clock',\n",
       "  1),\n",
       " ('kinase is necessary for the phosphorylation of period per a transcription repressor and clock',\n",
       "  1),\n",
       " ('left and medullary right fibrosis were quantified from trichrome staining in sham contra lateral',\n",
       "  1),\n",
       " ('lock genes in the three neurons clusters mentioned above the circadian transcription factors clock',\n",
       "  1),\n",
       " ('one of the clock neurons show per immunoreactivity c representative staining results of clock',\n",
       "  1),\n",
       " ('ore of the circadian pathway is a transcription feedback loop in drosophila melanogaster clock',\n",
       "  1),\n",
       " ('phila the major loop involves a positive limb which include the transcription activators clock',\n",
       "  1),\n",
       " ('se phosphorylation and destabilizes the period per and timeless tim proteins which inhibit clock',\n",
       "  1),\n",
       " ('sensitive proteins and negative feedback loop figure 1 26 the heterodimer formed by clock',\n",
       "  1),\n",
       " ('tablish circadian rhythms in physiology and behavior in this species putative daphnia pulex clock',\n",
       "  1),\n",
       " ('the flies clock consists of a transcription feedback loop whereby the transcription factors clock',\n",
       "  1),\n",
       " ('these feedback loop are both positive and negative with positive factors e g clock',\n",
       "  1),\n",
       " ('uclear rnps hnrnps cleavage and polyadenylation factors proteins kinase such as cdc like kinase',\n",
       "  1),\n",
       " ('we identified inhibitors with surprising isoform specificity for clk1 over clk3 cdc2 like kinase',\n",
       "  1),\n",
       " ('1 a simplified version of the core feedback loop consists of the transcription activators clock',\n",
       "  1),\n",
       " ('20 per and tim repress their own expression by binding to their transcription activators clock',\n",
       "  1),\n",
       " ('activators of brm functions to recruit repress factors to limit the transcription output of clock',\n",
       "  1),\n",
       " ('as in vertebrates in the core feedback loop of the drosophila clock transcription factors clock',\n",
       "  1),\n",
       " ('brief the core clock of d melanogaster consists of an interactions feedback loop involves clock',\n",
       "  1),\n",
       " ('chanism of dysfunction of regulatory b cells is unclear the circadian locomotor output cycle kaput',\n",
       "  1),\n",
       " ('e important to their survival central to drosophila clockwork is a key transcription factors clock',\n",
       "  1),\n",
       " ('en the pseudo diffusivity and tubular injury score d stenotic kidney stk contra lateral kidney',\n",
       "  1),\n",
       " ('enes formed two distinct heterodimer period per and timeless tim formed one heterodimer and clock',\n",
       "  1),\n",
       " ('enes that underlie self sustained 24 hour feedback loop in flies two transcription factors clock',\n",
       "  1),\n",
       " ('example in vertebrates the basic helix loop helix per arnt sim domain transcription factors clock',\n",
       "  1),\n",
       " ('in the clock circuit of the fruit flies drosophila melanogaster a heterodimer consists of clock',\n",
       "  1),\n",
       " ('ional feedback loop that are highly conserved between flies and mammals in drosophila the clock',\n",
       "  1),\n",
       " ('is highly analog to it counterpart in mammalian cells in flies the transcription factors clock',\n",
       "  1),\n",
       " ('ker figure 1 for review see for example 2 17 the circadian transcription factors clock',\n",
       "  1),\n",
       " ('lecular clock in the pacemaker neurons in the brain of drosophila the transcription factors clock',\n",
       "  1),\n",
       " ('ls in clock neurons in drosophila this clock is composed of the transcription activators clock',\n",
       "  1),\n",
       " ('malian molecular clock 2 30 in the drosophila core feedback loop the clock proteins clock',\n",
       "  1),\n",
       " ('ngly supportive of calfi clk being a true clock family member putative calanus finmarchicus clock',\n",
       "  1),\n",
       " ('ock clk and cycle cyc 1 2 clk and cyc encoding the transcription factors clock',\n",
       "  1),\n",
       " ('ophila melanogaster two basic helix loop helix per arnt sim bhlh pas transcription factors clock',\n",
       "  1),\n",
       " ('or brain and muscle aryl hydrocarbon receptor nuclear translocator like proteins 1 bmal1 and clock',\n",
       "  1),\n",
       " ('oung and kay 2001 zheng and sehgal 2012 in drosophila period per timeless tim clock',\n",
       "  1),\n",
       " ('per and timeless tim genes their proteins per and tim and the transcription factors clock',\n",
       "  1),\n",
       " ('r clock in drosophila consists of an autoregulatory loop in which the transcription factors clock',\n",
       "  1),\n",
       " ('re several because atherogenic factors exacerbate damage in both the stk and the contralateral kidney',\n",
       "  1),\n",
       " ('rfering with formation of the clk pl bmal1 complex a immunoprecipitation ip of endogenous clock',\n",
       "  1),\n",
       " ('s by phosphorylation of clock components 43 in mammals gsk3 is proposed to phosphorylation clock',\n",
       "  1),\n",
       " ('s often mediate proteins proteins interactions and the same region in the drosophila clock ortholog',\n",
       "  1),\n",
       " ('scription of the clock components period per and timeless tim is activators by the clock',\n",
       "  1),\n",
       " ('se dependent stimulates the release of pgs from both the hnk and unobstructed contralateral kidney',\n",
       "  1),\n",
       " ('tified the brahma brm complex as a regulated of the drosophila clock in drosophila clock',\n",
       "  1),\n",
       " ('toreceptor cells occurs 47 the positive elements of the mammalian clock are the proteins clock',\n",
       "  1),\n",
       " ('transcription translational feedback loop ttfl at the core of the system are the activators clock',\n",
       "  1),\n",
       " ('ukaryotic mrna these phosphorylation events are dependent on sr proteins srpk and cdc2 like kinase',\n",
       "  1),\n",
       " ('via one or more transcription feedback loop 1 in drosophila a heterodimer formed by clock',\n",
       "  1),\n",
       " ('well known inhibitors for a number of different kinase cyclin dependent kinase cdc2 like kinase',\n",
       "  1),\n",
       " ('ylation sites indicates that o glcnacylation at per s942 reduces interactions between per and clock',\n",
       "  1),\n",
       " ('32 p phosphate into clk 32 p was analyzed by autoradiography and total clk proteins levels',\n",
       "  1.0),\n",
       " ('ar clock mechanism in this species include period d rosophila per d per timeless tim clock',\n",
       "  1),\n",
       " ('bhlh pas transcription factors called clock and bmal1 or it homologue npas2 in mammals and clock',\n",
       "  1),\n",
       " ('brates the circadian clock is composed of an integral set of proteins include period per clock',\n",
       "  1),\n",
       " ('clock and an output pathway 7 the clock itself is comprised of the transcription factors clock',\n",
       "  1),\n",
       " ('core clock genes the expression of per and tim is regulated by the transcription factors clock',\n",
       "  1),\n",
       " ('ed to the same two positive basic helix loop helix bhlh pas containing transcription factors clock',\n",
       "  1),\n",
       " ('ent review see 1 the flies clock is composed of intracellular feedback loop the proteins clock',\n",
       "  1),\n",
       " ('es their activators and subcellular availability we here show that inhibit of the cdc2 like kinase',\n",
       "  1),\n",
       " ('gative feedback loop figure 1 at the center of these loop are the transcription factors clock',\n",
       "  1),\n",
       " ('in part of two interconnected feedback loop 3 in the period timeless per tim loop clock',\n",
       "  1),\n",
       " ('l repressor we found that rna levels of genes directly activators by the transcription factors clock',\n",
       "  1),\n",
       " ('lating activators of the circadian clock is formed by a heterodimer of the transcription factors clock',\n",
       "  1),\n",
       " ('of the principal loop is composed of a heterodimeric complex of the two transcription factors clock',\n",
       "  1),\n",
       " ('olecular clock a recently review by allada and chung 2010 a heterodimer formed by the clock',\n",
       "  1),\n",
       " ('preconditioning increased cxcr4 expression in c ardiosphere derived l in − c k it + progenitor',\n",
       "  1),\n",
       " ('ptional feedback loop in mammals the transcription factors bmal1 acts as a dimer with either clock',\n",
       "  1),\n",
       " ('sivity correlated with distal tubular mtt in the stenotic kidney stk a and contra lateral kidney',\n",
       "  1),\n",
       " ('tions in the negative limb of the clockwork by presumably binding to the transcription factors clock',\n",
       "  1),\n",
       " ('were derived from cardiosphere cultures and were defined by as cardiosphere derived c kit + lin',\n",
       "  1),\n",
       " ('aily oscillator of the clock genes expression at mrna and proteins levels in the evening the clock',\n",
       "  1),\n",
       " ('cription of the per and tim genes is initiated during mid day by the transcription factors clock',\n",
       "  1),\n",
       " ('ction f p were calculated in cortex and medulla of the stenotic stk and contra lateral kidney',\n",
       "  1),\n",
       " ('kloss et al 1998 price et al 1998 he et al 2017 in mammals the proteins clock',\n",
       "  1),\n",
       " ('nd consists of a two arm negative feedback loop the positive arm proteins in drosophila are clock',\n",
       "  1),\n",
       " ('nto the early part of the night two basic helix loop helix bhlh pas transcription factors clock',\n",
       "  1),\n",
       " ('on with local tfs networks a prime example is the circadian clock although the conserved tfs clock',\n",
       "  1),\n",
       " ('oops in which the basic helix loop helix per arnt sim bhlh pas domain transcription factors clock',\n",
       "  1),\n",
       " ('orated 32 p phosphate into clk 32 p was analyzed by autoradiography and total clk proteins levels',\n",
       "  1),\n",
       " ('rated 32 p phosphate into clk 32 p was analyzed by autoradiography and total clk proteins levels',\n",
       "  1),\n",
       " ('scription of period per and timeless tim in the evening per and tim proteins then repress clock',\n",
       "  1),\n",
       " ('step in control the speed of the clock 40 dbt is also require for phosphorylation of clock',\n",
       "  1),\n",
       " ('zoni et al 2005 in the drosophila clock per and tim expression is activators by the clock',\n",
       "  1),\n",
       " ('ck2α and anti ha were used to reveal clk and ck2α respectively left western blots for clk alone',\n",
       "  1),\n",
       " ('dence that hypoxia induced cxcr4 expression in c ardiosphere derived l in − c k it + progenitor',\n",
       "  1),\n",
       " ('e positive limb of this loop is driven by a heterodimer of two bhlh pas transcription factors clock',\n",
       "  1),\n",
       " ('enous cycle 1 3 6 two basic helix loop helix per arnt sim bhlh pas transcription factors clock',\n",
       "  1),\n",
       " ('lk4 phosphorylation the ser 28 of a truncated gflb3 with a deleted rs motif nδ21 cdc2 like kinase',\n",
       "  1),\n",
       " ('ock has one positive and two negative loop that interconnected at clk cyc a heterodimer of the clock',\n",
       "  1),\n",
       " ('eters in the cortex a d and medulla e h of sham stenotic stk and non stenotic contra lateral',\n",
       "  1),\n",
       " ('iour to the 24h day in flies as in mammals the oscillator operates in cells that co expression clock',\n",
       "  1),\n",
       " ('on at the heart of the molecular clock clock or it analog npas2 and bmal1 in mammals and clock',\n",
       "  1),\n",
       " ('on as an input pin tgout input from the das while rb1 rb2 rb3 and rb4 are output pin for clock',\n",
       "  1),\n",
       " ('alone', 0.0),\n",
       " ('kaput', 0.0),\n",
       " ('kinase', 0.0),\n",
       " ('lateral', 0.0),\n",
       " ('levels', 0.0),\n",
       " ('lin', 0.0),\n",
       " ('lymphokines', 0.0),\n",
       " ('ortholog', 0.0),\n",
       " ('progenitor', 0.0),\n",
       " ('+ lin', 0.0),\n",
       " ('+ progenitor', 0.0),\n",
       " ('are clock', 0.0),\n",
       " ('chemotactic lymphokines', 0.0),\n",
       " ('clk alone', 0.0),\n",
       " ('clock ortholog', 0.0),\n",
       " ('cycle kaput', 0.0),\n",
       " ('drosophila clock', 0.0),\n",
       " ('either clock', 0.0),\n",
       " ('elements clock', 0.0),\n",
       " ('endogenous clock', 0.0),\n",
       " ('expression clock', 0.0),\n",
       " ('factors clock', 0.0),\n",
       " ('finmarchicus clock', 0.0),\n",
       " ('for clock', 0.0),\n",
       " ('g clock', 0.0),\n",
       " ('heterodimer clock', 0.0),\n",
       " ('inhibit clock', 0.0),\n",
       " ('involves clock', 0.0),\n",
       " ('lateral kidney', 0.0),\n",
       " ('loop clock', 0.0),\n",
       " ('mammals clock', 0.0),\n",
       " ('melanogaster clock', 0.0),\n",
       " ('per clock', 0.0),\n",
       " ('phosphorylation clock', 0.0),\n",
       " ('proteins levels', 0.0),\n",
       " ('pulex clock', 0.0),\n",
       " ('repress clock', 0.0),\n",
       " ('tim clock', 0.0),\n",
       " ('when clock', 0.0),\n",
       " ('activators by clock', 0.0),\n",
       " ('activators the clock', 0.0),\n",
       " ('activators when clock', 0.0),\n",
       " ('bmal1 and clock', 0.0),\n",
       " ('calanus finmarchicus clock', 0.0),\n",
       " ('cdc like kinase', 0.0),\n",
       " ('clk proteins levels', 0.0),\n",
       " ('clock proteins clock', 0.0),\n",
       " ('co expression clock', 0.0),\n",
       " ('complex of clock', 0.0),\n",
       " ('conserved tfs clock', 0.0),\n",
       " ('consists of clock', 0.0),\n",
       " ('daphnia pulex clock', 0.0),\n",
       " ('drosophila are clock', 0.0),\n",
       " ('drosophila clock ortholog', 0.0),\n",
       " ('drosophila melanogaster clock', 0.0),\n",
       " ('drosophila the clock', 0.0),\n",
       " ('evening the clock', 0.0),\n",
       " ('factors heterodimer clock', 0.0),\n",
       " ('for clk alone', 0.0),\n",
       " ('formed by clock', 0.0),\n",
       " ('heterodimer and clock', 0.0),\n",
       " ('in drosophila clock', 0.0),\n",
       " ('it + progenitor', 0.0),\n",
       " ('kit + lin', 0.0),\n",
       " ('loop involves clock', 0.0),\n",
       " ('mammals and clock', 0.0),\n",
       " ('of chemotactic lymphokines', 0.0),\n",
       " ('of endogenous clock', 0.0),\n",
       " ('oscillator the clock', 0.0),\n",
       " ('output cycle kaput', 0.0),\n",
       " ('output of clock', 0.0),\n",
       " ('per and clock', 0.0),\n",
       " ('period per clock', 0.0),\n",
       " ('phosphorylation of clock', 0.0),\n",
       " ('pin for clock', 0.0),\n",
       " ('positive elements clock', 0.0),\n",
       " ('repressor and clock', 0.0),\n",
       " ('results of clock', 0.0),\n",
       " ('sham contra lateral', 0.0),\n",
       " ('stenotic contra lateral', 0.0),\n",
       " ('the activators clock', 0.0),\n",
       " ('the contralateral kidney', 0.0),\n",
       " ('the tfs clock', 0.0),\n",
       " ('then repress clock', 0.0),\n",
       " ('tim loop clock', 0.0),\n",
       " ('to mammals clock', 0.0),\n",
       " ('to phosphorylation clock', 0.0),\n",
       " ('unobstructed contralateral kidney', 0.0),\n",
       " ('which inhibit clock', 0.0),\n",
       " ('with either clock', 0.0),\n",
       " ('1 bmal1 and clock', 0.0),\n",
       " ('activators by the clock', 0.0),\n",
       " ('activators e g clock', 0.0),\n",
       " ('and cdc2 like kinase', 0.0),\n",
       " ('and the contralateral kidney', 0.0),\n",
       " ('and unobstructed contralateral kidney', 0.0),\n",
       " ('are activators by clock', 0.0),\n",
       " ('are activators when clock', 0.0),\n",
       " ('are the activators clock', 0.0),\n",
       " ('are the proteins clock', 0.0),\n",
       " ('as cdc like kinase', 0.0),\n",
       " ('between per and clock', 0.0),\n",
       " ('blots for clk alone', 0.0),\n",
       " ('by the proteins clock', 0.0),\n",
       " ('by the tfs clock', 0.0),\n",
       " ('c kit + lin', 0.0),\n",
       " ('circadian oscillator the clock', 0.0),\n",
       " ('circadian transcription factors clock', 0.0),\n",
       " ('clk3 cdc2 like kinase', 0.0),\n",
       " ('clock in drosophila clock', 0.0),\n",
       " ('clock transcription factors clock', 0.0),\n",
       " ('conserved cdc2 like kinase', 0.0),\n",
       " ('containing transcription factors clock', 0.0),\n",
       " ('dimer with either clock', 0.0),\n",
       " ('encoding cdc2 like kinase', 0.0),\n",
       " ('factors e g clock', 0.0),\n",
       " ('feedback loop involves clock', 0.0),\n",
       " ('feedback loop the clock', 0.0),\n",
       " ('for phosphorylation of clock', 0.0),\n",
       " ('formed by the clock', 0.0),\n",
       " ('heterodimer consists of clock', 0.0),\n",
       " ('heterodimer of the clock', 0.0),\n",
       " ('heterodimeric complex of clock', 0.0),\n",
       " ('in drosophila are clock', 0.0),\n",
       " ('in drosophila the clock', 0.0),\n",
       " ('in sham contra lateral', 0.0),\n",
       " ('include period per clock', 0.0),\n",
       " ('insects to mammals clock', 0.0),\n",
       " ('ip of endogenous clock', 0.0),\n",
       " ('k it + progenitor', 0.0),\n",
       " ('key transcription factors clock', 0.0),\n",
       " ('kinase cdc2 like kinase', 0.0),\n",
       " ('locomotor output cycle kaput', 0.0),\n",
       " ('loop the proteins clock', 0.0),\n",
       " ('mammals the proteins clock', 0.0),\n",
       " ('non stenotic contra lateral', 0.0),\n",
       " ('nδ21 cdc2 like kinase', 0.0),\n",
       " ('one heterodimer and clock', 0.0),\n",
       " ('output pin for clock', 0.0),\n",
       " ('pas transcription factors clock', 0.0),\n",
       " ('per tim loop clock', 0.0),\n",
       " ('product of chemotactic lymphokines', 0.0),\n",
       " ('proposed to phosphorylation clock', 0.0),\n",
       " ('proteins then repress clock', 0.0),\n",
       " ('proteins timeless tim clock', 0.0),\n",
       " ('proteins which inhibit clock', 0.0),\n",
       " ('putative calanus finmarchicus clock', 0.0),\n",
       " ('putative daphnia pulex clock', 0.0),\n",
       " ('repressor of the clock', 0.0),\n",
       " ('staining results of clock', 0.0),\n",
       " ('stk contra lateral kidney', 0.0),\n",
       " ('that co expression clock', 0.0),\n",
       " ('the cdc2 like kinase', 0.0),\n",
       " ('the clock proteins clock', 0.0),\n",
       " ('the conserved tfs clock', 0.0),\n",
       " ('the drosophila clock ortholog', 0.0),\n",
       " ('the evening the clock', 0.0),\n",
       " ('their transcription activators clock', 0.0),\n",
       " ('tim loop the clock', 0.0),\n",
       " ('total clk proteins levels', 0.0),\n",
       " ('transcription activators the clock', 0.0),\n",
       " ('transcription factors heterodimer clock', 0.0),\n",
       " ('transcription output of clock', 0.0),\n",
       " ('transcription repressor and clock', 0.0),\n",
       " ('a and contra lateral kidney', 0.0),\n",
       " ('a dimer with either clock', 0.0),\n",
       " ('a feedback loop the clock', 0.0),\n",
       " ('a heterodimer consists of clock', 0.0),\n",
       " ('a heterodimer formed by clock', 0.0),\n",
       " ('a heterodimer of the clock', 0.0),\n",
       " ('a heterodimeric complex of clock', 0.0),\n",
       " ('a key transcription factors clock', 0.0),\n",
       " ('a transcription repressor and clock', 0.0),\n",
       " ('although the conserved tfs clock', 0.0),\n",
       " ('and non stenotic contra lateral', 0.0),\n",
       " ('and the transcription factors clock', 0.0),\n",
       " ('and total clk proteins levels', 0.0),\n",
       " ('are output pin for clock', 0.0),\n",
       " ('are the transcription factors clock', 0.0),\n",
       " ('bmal1 in mammals and clock', 0.0),\n",
       " ('briefly the transcription activators clock', 0.0),\n",
       " ('c k it + progenitor', 0.0),\n",
       " ('cells that co expression clock', 0.0),\n",
       " ('changes in drosophila melanogaster clock', 0.0),\n",
       " ('circadian locomotor output cycle kaput', 0.0),\n",
       " ('clock are the proteins clock', 0.0),\n",
       " ('clock two transcription factors clock', 0.0),\n",
       " ('clones encoding cdc2 like kinase', 0.0),\n",
       " ('d per timeless tim clock', 0.0),\n",
       " ('dependent kinase cdc2 like kinase', 0.0),\n",
       " ('derived c kit + lin', 0.0),\n",
       " ('drosophila circadian oscillator the clock', 0.0),\n",
       " ('drosophila clock in drosophila clock', 0.0),\n",
       " ('drosophila clock transcription factors clock', 0.0),\n",
       " ('encoding the transcription activators clock', 0.0),\n",
       " ('encoding the transcription factors clock', 0.0),\n",
       " ('evolutionarily conserved cdc2 like kinase', 0.0),\n",
       " ('feedback loop the proteins clock', 0.0),\n",
       " ('flies the transcription factors clock', 0.0),\n",
       " ('flies two transcription factors clock', 0.0),\n",
       " ('formed by the proteins clock', 0.0),\n",
       " ('formed one heterodimer and clock', 0.0),\n",
       " ('from insects to mammals clock', 0.0),\n",
       " ('heterodimer formed by the clock', 0.0),\n",
       " ('hnk and unobstructed contralateral kidney', 0.0),\n",
       " ('immunoprecipitation ip of endogenous clock', 0.0),\n",
       " ('in mammals the proteins clock', 0.0),\n",
       " ('in the drosophila clock ortholog', 0.0),\n",
       " ('in the evening the clock', 0.0),\n",
       " ('include the transcription activators clock', 0.0),\n",
       " ('induced product of chemotactic lymphokines', 0.0),\n",
       " ('interactions between per and clock', 0.0),\n",
       " ('interactions feedback loop involves clock', 0.0),\n",
       " ('interconnected by the tfs clock', 0.0),\n",
       " ('involves the transcription factors clock', 0.0),\n",
       " ('is proposed to phosphorylation clock', 0.0),\n",
       " ('kidney stk contra lateral kidney', 0.0),\n",
       " ('loop in drosophila melanogaster clock', 0.0),\n",
       " ('loop the clock proteins clock', 0.0),\n",
       " ('major proteins timeless tim clock', 0.0),\n",
       " ('mammals in drosophila the clock', 0.0),\n",
       " ('member putative calanus finmarchicus clock', 0.0),\n",
       " ('motif nδ21 cdc2 like kinase', 0.0),\n",
       " ('npas2 in mammals and clock', 0.0),\n",
       " ('of the cdc2 like kinase', 0.0),\n",
       " ('of two transcription factors clock', 0.0),\n",
       " ('over clk3 cdc2 like kinase', 0.0),\n",
       " ('pas containing transcription factors clock', 0.0),\n",
       " ('pas domain transcription factors clock', 0.0),\n",
       " ('pas transcription factors heterodimer clock', 0.0),\n",
       " ('per tim loop the clock', 0.0),\n",
       " ('positive factors e g clock', 0.0),\n",
       " ('proteins 1 bmal1 and clock', 0.0),\n",
       " ('proteins in drosophila are clock', 0.0),\n",
       " ('proteins include period per clock', 0.0),\n",
       " ('representative staining results of clock', 0.0),\n",
       " ('require for phosphorylation of clock', 0.0),\n",
       " ('require the transcription factors clock', 0.0),\n",
       " ('sim domain transcription factors clock', 0.0),\n",
       " ('species putative daphnia pulex clock', 0.0),\n",
       " ('srpk and cdc2 like kinase', 0.0),\n",
       " ('staining in sham contra lateral', 0.0),\n",
       " ('stk and contra lateral kidney', 0.0),\n",
       " ('stk and the contralateral kidney', 0.0),\n",
       " ('such as cdc like kinase', 0.0),\n",
       " ('system are the activators clock', 0.0),\n",
       " ('that are activators by clock', 0.0),\n",
       " ('that the transcription activators clock', 0.0),\n",
       " ('the heterodimer formed by clock', 0.0),\n",
       " ('the transcription output of clock', 0.0),\n",
       " ('the two transcription factors clock', 0.0),\n",
       " ('their transcription activators the clock', 0.0),\n",
       " ('tim proteins then repress clock', 0.0),\n",
       " ('tim proteins which inhibit clock', 0.0),\n",
       " ('timeless per tim loop clock', 0.0),\n",
       " ('to the transcription factors clock', 0.0),\n",
       " ('to their transcription activators clock', 0.0),\n",
       " ('transcription activators e g clock', 0.0),\n",
       " ('transcription are activators when clock', 0.0),\n",
       " ('transcription repressor of the clock', 0.0),\n",
       " ('ttfl the transcription factors clock', 0.0),\n",
       " ('western blots for clk alone', 0.0),\n",
       " ('whereby the transcription factors clock', 0.0),\n",
       " ('which the transcription factors clock', 0.0),\n",
       " ('17 the circadian transcription factors clock', 0.0),\n",
       " ('2017 in mammals the proteins clock', 0.0),\n",
       " ('26 the heterodimer formed by clock', 0.0),\n",
       " ('a heterodimer formed by the clock', 0.0),\n",
       " ('a immunoprecipitation ip of endogenous clock', 0.0),\n",
       " ('a induced product of chemotactic lymphokines', 0.0),\n",
       " ('above the circadian transcription factors clock', 0.0),\n",
       " ('activators by the transcription factors clock', 0.0),\n",
       " ('activators of the transcription activators clock', 0.0),\n",
       " ('activators of the transcription factors clock', 0.0),\n",
       " ('also require for phosphorylation of clock', 0.0),\n",
       " ('an interactions feedback loop involves clock', 0.0),\n",
       " ('and bmal1 in mammals and clock', 0.0),\n",
       " ('and mammals in drosophila the clock', 0.0),\n",
       " ('and tim proteins then repress clock', 0.0),\n",
       " ('are interconnected by the tfs clock', 0.0),\n",
       " ('arm proteins in drosophila are clock', 0.0),\n",
       " ('arnt sim domain transcription factors clock', 0.0),\n",
       " ('as a dimer with either clock', 0.0),\n",
       " ('autoradiography and total clk proteins levels', 0.0),\n",
       " ('bhlh pas containing transcription factors clock', 0.0),\n",
       " ('bhlh pas domain transcription factors clock', 0.0),\n",
       " ('bhlh pas transcription factors heterodimer clock', 0.0),\n",
       " ('binding to the transcription factors clock', 0.0),\n",
       " ('binding to their transcription activators clock', 0.0),\n",
       " ('c representative staining results of clock', 0.0),\n",
       " ('cardiosphere derived c kit + lin', 0.0),\n",
       " ('clk1 over clk3 cdc2 like kinase', 0.0),\n",
       " ('clock although the conserved tfs clock', 0.0),\n",
       " ('complex formed by the proteins clock', 0.0),\n",
       " ('complex of the transcription factors clock', 0.0),\n",
       " ('composed of the transcription activators clock', 0.0),\n",
       " ('comprised of the transcription factors clock', 0.0),\n",
       " ('conserved from insects to mammals clock', 0.0),\n",
       " ('consists of the transcription activators clock', 0.0),\n",
       " ('consists of two transcription factors clock', 0.0),\n",
       " ('constitute a feedback loop the clock', 0.0),\n",
       " ('cyc a heterodimer of the clock', 0.0),\n",
       " ('cyc encoding the transcription factors clock', 0.0),\n",
       " ('cyclin dependent kinase cdc2 like kinase', 0.0),\n",
       " ('day by the transcription factors clock', 0.0),\n",
       " ('drosophila a heterodimer formed by clock', 0.0),\n",
       " ('drosophila period per timeless tim clock', 0.0),\n",
       " ('drosophila ttfl the transcription factors clock', 0.0),\n",
       " ('elements briefly the transcription activators clock', 0.0),\n",
       " ('environmental changes in drosophila melanogaster clock', 0.0),\n",
       " ('expression is activators by the clock', 0.0),\n",
       " ('family member putative calanus finmarchicus clock', 0.0),\n",
       " ('feedback loop in drosophila melanogaster clock', 0.0),\n",
       " ('feedback loop the clock proteins clock', 0.0),\n",
       " ('four major proteins timeless tim clock', 0.0),\n",
       " ('genes encoding the transcription activators clock', 0.0),\n",
       " ('gsk3 is proposed to phosphorylation clock', 0.0),\n",
       " ('helix bhlh pas transcription factors clock', 0.0),\n",
       " ('heterodimer of the transcription factors clock', 0.0),\n",
       " ('homologue npas2 in mammals and clock', 0.0),\n",
       " ('in cells that co expression clock', 0.0),\n",
       " ('in flies the transcription factors clock', 0.0),\n",
       " ('in flies two transcription factors clock', 0.0),\n",
       " ('in which the transcription factors clock', 0.0),\n",
       " ('inhibit of the cdc2 like kinase', 0.0),\n",
       " ('intracellular feedback loop the proteins clock', 0.0),\n",
       " ('is a heterodimeric complex of clock', 0.0),\n",
       " ('is a key transcription factors clock', 0.0),\n",
       " ('kinase such as cdc like kinase', 0.0),\n",
       " ('left western blots for clk alone', 0.0),\n",
       " ('length clones encoding cdc2 like kinase', 0.0),\n",
       " ('levels in the evening the clock', 0.0),\n",
       " ('like proteins 1 bmal1 and clock', 0.0),\n",
       " ('limit the transcription output of clock', 0.0),\n",
       " ('loop are the transcription factors clock', 0.0),\n",
       " ('loop that are activators by clock', 0.0),\n",
       " ('loop whereby the transcription factors clock', 0.0),\n",
       " ('main transcription repressor of the clock', 0.0),\n",
       " ('mammalian clock are the proteins clock', 0.0),\n",
       " ('melanogaster a heterodimer consists of clock', 0.0),\n",
       " ('of drosophila the transcription factors clock', 0.0),\n",
       " ('of proteins include period per clock', 0.0),\n",
       " ('of the two transcription factors clock', 0.0),\n",
       " ('of their transcription activators the clock', 0.0),\n",
       " ('of transcription activators e g clock', 0.0),\n",
       " ('per a transcription repressor and clock', 0.0),\n",
       " ('per d per timeless tim clock', 0.0),\n",
       " ('period timeless per tim loop clock', 0.0),\n",
       " ('proteins period per timeless tim clock', 0.0),\n",
       " ('proteins srpk and cdc2 like kinase', 0.0),\n",
       " ('rb4 are output pin for clock', 0.0),\n",
       " ('reduces interactions between per and clock', 0.0),\n",
       " ('region in the drosophila clock ortholog', 0.0),\n",
       " ('regulated by the transcription factors clock', 0.0),\n",
       " ('regulated of the transcription factors clock', 0.0),\n",
       " ('rhythms require the transcription factors clock', 0.0),\n",
       " ('rs motif nδ21 cdc2 like kinase', 0.0),\n",
       " ('shown that the transcription activators clock', 0.0),\n",
       " ('sim bhlh pas transcription factors clock', 0.0),\n",
       " ('stenotic kidney stk contra lateral kidney', 0.0),\n",
       " ('stenotic stk and contra lateral kidney', 0.0),\n",
       " ('stk a and contra lateral kidney', 0.0),\n",
       " ('stk and non stenotic contra lateral', 0.0),\n",
       " ('that involves the transcription factors clock', 0.0),\n",
       " ('the circadian locomotor output cycle kaput', 0.0),\n",
       " ('the drosophila circadian oscillator the clock', 0.0),\n",
       " ('the drosophila clock in drosophila clock', 0.0),\n",
       " ('the drosophila clock transcription factors clock', 0.0),\n",
       " ('the evolutionarily conserved cdc2 like kinase', 0.0),\n",
       " ('the hnk and unobstructed contralateral kidney', 0.0),\n",
       " ('the per tim loop the clock', 0.0),\n",
       " ('the stk and the contralateral kidney', 0.0),\n",
       " ('the system are the activators clock', 0.0),\n",
       " ('these clock two transcription factors clock', 0.0),\n",
       " ('this species putative daphnia pulex clock', 0.0),\n",
       " ('tim and the transcription factors clock', 0.0),\n",
       " ('tim formed one heterodimer and clock', 0.0),\n",
       " ('tim is activators by the clock', 0.0),\n",
       " ('tim transcription are activators when clock', 0.0),\n",
       " ('timeless tim proteins which inhibit clock', 0.0),\n",
       " ('trichrome staining in sham contra lateral', 0.0),\n",
       " ('two bhlh pas transcription factors clock', 0.0),\n",
       " ('which include the transcription activators clock', 0.0),\n",
       " ('with positive factors e g clock', 0.0),\n",
       " ('− c k it + progenitor', 0.0),\n",
       " ('1 26 the heterodimer formed by clock', 0.0),\n",
       " ('2 17 the circadian transcription factors clock', 0.0),\n",
       " ('2010 a heterodimer formed by the clock', 0.0),\n",
       " ('a complex formed by the proteins clock', 0.0),\n",
       " ('a heterodimer of the transcription factors clock', 0.0),\n",
       " ('acts as a dimer with either clock', 0.0),\n",
       " ('activators is a heterodimeric complex of clock', 0.0),\n",
       " ('activators of their transcription activators the clock', 0.0),\n",
       " ('al 2017 in mammals the proteins clock', 0.0),\n",
       " ('and cyc encoding the transcription factors clock', 0.0),\n",
       " ('and rb4 are output pin for clock', 0.0),\n",
       " ('and tim and the transcription factors clock', 0.0),\n",
       " ('and timeless tim proteins which inhibit clock', 0.0),\n",
       " ('are conserved from insects to mammals clock', 0.0),\n",
       " ('arnt sim bhlh pas transcription factors clock', 0.0),\n",
       " ('as cardiosphere derived c kit + lin', 0.0),\n",
       " ('both the hnk and unobstructed contralateral kidney', 0.0),\n",
       " ('both the stk and the contralateral kidney', 0.0),\n",
       " ('brain of drosophila the transcription factors clock', 0.0),\n",
       " ('by autoradiography and total clk proteins levels', 0.0),\n",
       " ('by binding to their transcription activators clock', 0.0),\n",
       " ('cells in flies the transcription factors clock', 0.0),\n",
       " ('circadian clock although the conserved tfs clock', 0.0),\n",
       " ('clk cyc a heterodimer of the clock', 0.0),\n",
       " ('clock family member putative calanus finmarchicus clock', 0.0),\n",
       " ('clockwork is a key transcription factors clock', 0.0),\n",
       " ('complex a immunoprecipitation ip of endogenous clock', 0.0),\n",
       " ('complex of the two transcription factors clock', 0.0),\n",
       " ('con a induced product of chemotactic lymphokines', 0.0),\n",
       " ('consists of transcription activators e g clock', 0.0),\n",
       " ('core feedback loop the clock proteins clock', 0.0),\n",
       " ('d stenotic kidney stk contra lateral kidney', 0.0),\n",
       " ('daily environmental changes in drosophila melanogaster clock', 0.0),\n",
       " ('deleted rs motif nδ21 cdc2 like kinase', 0.0),\n",
       " ('directly activators by the transcription factors clock', 0.0),\n",
       " ('drosophila melanogaster a heterodimer consists of clock', 0.0),\n",
       " ('feedback loop that are activators by clock', 0.0),\n",
       " ('feedback loop whereby the transcription factors clock', 0.0),\n",
       " ('flies and mammals in drosophila the clock', 0.0),\n",
       " ('flies four major proteins timeless tim clock', 0.0),\n",
       " ('for clk1 over clk3 cdc2 like kinase', 0.0),\n",
       " ('from trichrome staining in sham contra lateral', 0.0),\n",
       " ('full length clones encoding cdc2 like kinase', 0.0),\n",
       " ('have shown that the transcription activators clock', 0.0),\n",
       " ('helix bhlh pas containing transcription factors clock', 0.0),\n",
       " ('heterodimeric complex of the transcription factors clock', 0.0),\n",
       " ('immunoreactivity c representative staining results of clock', 0.0),\n",
       " ('in drosophila a heterodimer formed by clock', 0.0),\n",
       " ('in drosophila period per timeless tim clock', 0.0),\n",
       " ('in the drosophila circadian oscillator the clock', 0.0),\n",
       " ('in the per tim loop the clock', 0.0),\n",
       " ('in these clock two transcription factors clock', 0.0),\n",
       " ('in this species putative daphnia pulex clock', 0.0),\n",
       " ('in − c k it + progenitor', 0.0),\n",
       " ('is also require for phosphorylation of clock', 0.0),\n",
       " ('is composed of the transcription activators clock', 0.0),\n",
       " ('is comprised of the transcription factors clock', 0.0),\n",
       " ('is regulated by the transcription factors clock', 0.0),\n",
       " ('it homologue npas2 in mammals and clock', 0.0),\n",
       " ('kidney stk a and contra lateral kidney', 0.0),\n",
       " ('kinase cyclin dependent kinase cdc2 like kinase', 0.0),\n",
       " ('limb which include the transcription activators clock', 0.0),\n",
       " ('loop are interconnected by the tfs clock', 0.0),\n",
       " ('loop consists of the transcription activators clock', 0.0),\n",
       " ('loop helix bhlh pas transcription factors clock', 0.0),\n",
       " ('loop in drosophila the transcription factors clock', 0.0),\n",
       " ('loop in flies two transcription factors clock', 0.0),\n",
       " ('loop in which the transcription factors clock', 0.0),\n",
       " ('loop that involves the transcription factors clock', 0.0),\n",
       " ('mammals gsk3 is proposed to phosphorylation clock', 0.0),\n",
       " ('mentioned above the circadian transcription factors clock', 0.0),\n",
       " ('mid day by the transcription factors clock', 0.0),\n",
       " ('negative elements briefly the transcription activators clock', 0.0),\n",
       " ('negative with positive factors e g clock', 0.0),\n",
       " ('npas2 and bmal1 in mammals and clock', 0.0),\n",
       " ('of an interactions feedback loop involves clock', 0.0),\n",
       " ('of intracellular feedback loop the proteins clock', 0.0),\n",
       " ('of the drosophila clock in drosophila clock', 0.0),\n",
       " ('of the drosophila clock transcription factors clock', 0.0),\n",
       " ('of the evolutionarily conserved cdc2 like kinase', 0.0),\n",
       " ('of the system are the activators clock', 0.0),\n",
       " ('of two bhlh pas transcription factors clock', 0.0),\n",
       " ('operates in cells that co expression clock', 0.0),\n",
       " ('oscillator consists of two transcription factors clock', 0.0),\n",
       " ('per and tim proteins then repress clock', 0.0),\n",
       " ('per arnt sim domain transcription factors clock', 0.0),\n",
       " ('period per a transcription repressor and clock', 0.0),\n",
       " ('positive arm proteins in drosophila are clock', 0.0),\n",
       " ('presumably binding to the transcription factors clock', 0.0),\n",
       " ('product proteins period per timeless tim clock', 0.0),\n",
       " ('proteins kinase such as cdc like kinase', 0.0),\n",
       " ('proteins levels in the evening the clock', 0.0),\n",
       " ('respectively left western blots for clk alone', 0.0),\n",
       " ('rhythms in drosophila the transcription factors clock', 0.0),\n",
       " ('rosophila per d per timeless tim clock', 0.0),\n",
       " ('s942 reduces interactions between per and clock', 0.0),\n",
       " ('same region in the drosophila clock ortholog', 0.0),\n",
       " ('set of proteins include period per clock', 0.0),\n",
       " ('sim bhlh pas domain transcription factors clock', 0.0),\n",
       " ('sr proteins srpk and cdc2 like kinase', 0.0),\n",
       " ('stenotic stk and non stenotic contra lateral', 0.0),\n",
       " ('that constitute a feedback loop the clock', 0.0),\n",
       " ('that inhibit of the cdc2 like kinase', 0.0),\n",
       " ('the activators of the transcription activators clock', 0.0),\n",
       " ('the activators of the transcription factors clock', 0.0),\n",
       " ('the bhlh pas transcription factors heterodimer clock', 0.0),\n",
       " ('the drosophila ttfl the transcription factors clock', 0.0),\n",
       " ('the genes encoding the transcription activators clock', 0.0),\n",
       " ('the main transcription repressor of the clock', 0.0),\n",
       " ('the mammalian clock are the proteins clock', 0.0),\n",
       " ('the period timeless per tim loop clock', 0.0),\n",
       " ('the stenotic stk and contra lateral kidney', 0.0),\n",
       " ('these loop are the transcription factors clock', 0.0),\n",
       " ('these rhythms require the transcription factors clock', 0.0),\n",
       " ('tim expression is activators by the clock', 0.0),\n",
       " ('timeless tim formed one heterodimer and clock', 0.0),\n",
       " ('timeless tim is activators by the clock', 0.0),\n",
       " ('timeless tim transcription are activators when clock', 0.0),\n",
       " ('to limit the transcription output of clock', 0.0),\n",
       " ('transcription feedback loop in drosophila melanogaster clock', 0.0),\n",
       " ('translational regulated of the transcription factors clock', 0.0),\n",
       " ('translocator like proteins 1 bmal1 and clock', 0.0),\n",
       " ('unclear the circadian locomotor output cycle kaput', 0.0),\n",
       " ('1 in drosophila a heterodimer formed by clock', 0.0),\n",
       " ('1 in the per tim loop the clock', 0.0),\n",
       " ('2 full length clones encoding cdc2 like kinase', 0.0),\n",
       " ('2012 in drosophila period per timeless tim clock', 0.0),\n",
       " ('a con a induced product of chemotactic lymphokines', 0.0),\n",
       " ('a deleted rs motif nδ21 cdc2 like kinase', 0.0),\n",
       " ('a transcription feedback loop in drosophila melanogaster clock', 0.0),\n",
       " ('analog npas2 and bmal1 in mammals and clock', 0.0),\n",
       " ('analyzed by autoradiography and total clk proteins levels', 0.0),\n",
       " ('and negative with positive factors e g clock', 0.0),\n",
       " ('and proteins levels in the evening the clock', 0.0),\n",
       " ('and tim expression is activators by the clock', 0.0),\n",
       " ('and timeless tim formed one heterodimer and clock', 0.0),\n",
       " ('and timeless tim is activators by the clock', 0.0),\n",
       " ('and timeless tim transcription are activators when clock', 0.0),\n",
       " ('anticipate daily environmental changes in drosophila melanogaster clock',\n",
       "  0.0),\n",
       " ('arnt sim bhlh pas domain transcription factors clock', 0.0),\n",
       " ('at clk cyc a heterodimer of the clock', 0.0),\n",
       " ('autoregulatory loop in which the transcription factors clock', 0.0),\n",
       " ('behavior in this species putative daphnia pulex clock', 0.0),\n",
       " ('between flies and mammals in drosophila the clock', 0.0),\n",
       " ('bmal1 acts as a dimer with either clock', 0.0),\n",
       " ('bmal1 complex a immunoprecipitation ip of endogenous clock', 0.0),\n",
       " ('by a heterodimer of the transcription factors clock', 0.0),\n",
       " ('by as cardiosphere derived c kit + lin', 0.0),\n",
       " ('by presumably binding to the transcription factors clock', 0.0),\n",
       " ('chung 2010 a heterodimer formed by the clock', 0.0),\n",
       " ('circadian rhythms in drosophila the transcription factors clock', 0.0),\n",
       " ('ck2α respectively left western blots for clk alone', 0.0),\n",
       " ('clk and cyc encoding the transcription factors clock', 0.0),\n",
       " ('clock in these clock two transcription factors clock', 0.0),\n",
       " ('clock is composed of the transcription activators clock', 0.0),\n",
       " ('clusters mentioned above the circadian transcription factors clock', 0.0),\n",
       " ('composed of intracellular feedback loop the proteins clock', 0.0),\n",
       " ('consists of an interactions feedback loop involves clock', 0.0),\n",
       " ('core of the system are the activators clock', 0.0),\n",
       " ('d rosophila per d per timeless tim clock', 0.0),\n",
       " ('dbt is also require for phosphorylation of clock', 0.0),\n",
       " ('dd these rhythms require the transcription factors clock', 0.0),\n",
       " ('different kinase cyclin dependent kinase cdc2 like kinase', 0.0),\n",
       " ('drosophila clockwork is a key transcription factors clock', 0.0),\n",
       " ('drosophila core feedback loop the clock proteins clock', 0.0),\n",
       " ('drosophila the bhlh pas transcription factors heterodimer clock', 0.0),\n",
       " ('during mid day by the transcription factors clock', 0.0),\n",
       " ('encoding the main transcription repressor of the clock', 0.0),\n",
       " ('et al 2017 in mammals the proteins clock', 0.0),\n",
       " ('evening per and tim proteins then repress clock', 0.0),\n",
       " ('example 2 17 the circadian transcription factors clock', 0.0),\n",
       " ('expression by binding to their transcription activators clock', 0.0),\n",
       " ('factors proteins kinase such as cdc like kinase', 0.0),\n",
       " ('factors to limit the transcription output of clock', 0.0),\n",
       " ('feedback loop consists of the transcription activators clock', 0.0),\n",
       " ('feedback loop in drosophila the transcription factors clock', 0.0),\n",
       " ('feedback loop in flies two transcription factors clock', 0.0),\n",
       " ('feedback loop that involves the transcription factors clock', 0.0),\n",
       " ('figure 1 26 the heterodimer formed by clock', 0.0),\n",
       " ('flies drosophila melanogaster a heterodimer consists of clock', 0.0),\n",
       " ('from both the hnk and unobstructed contralateral kidney', 0.0),\n",
       " ('genes directly activators by the transcription factors clock', 0.0),\n",
       " ('helix loop helix bhlh pas transcription factors clock', 0.0),\n",
       " ('helix per arnt sim domain transcription factors clock', 0.0),\n",
       " ('heterodimer of two bhlh pas transcription factors clock', 0.0),\n",
       " ('heterodimeric complex of the two transcription factors clock', 0.0),\n",
       " ('in both the stk and the contralateral kidney', 0.0),\n",
       " ('in flies four major proteins timeless tim clock', 0.0),\n",
       " ('in mammals gsk3 is proposed to phosphorylation clock', 0.0),\n",
       " ('in the drosophila ttfl the transcription factors clock', 0.0),\n",
       " ('in the period timeless per tim loop clock', 0.0),\n",
       " ('inhibit activators of their transcription activators the clock', 0.0),\n",
       " ('inhibitors in the drosophila circadian oscillator the clock', 0.0),\n",
       " ('integral set of proteins include period per clock', 0.0),\n",
       " ('is unclear the circadian locomotor output cycle kaput', 0.0),\n",
       " ('itself is comprised of the transcription factors clock', 0.0),\n",
       " ('l in − c k it + progenitor', 0.0),\n",
       " ('loop consists of transcription activators e g clock', 0.0),\n",
       " ('loop helix bhlh pas containing transcription factors clock', 0.0),\n",
       " ('loop of the drosophila clock transcription factors clock', 0.0),\n",
       " ('mammals a complex formed by the proteins clock', 0.0),\n",
       " ('mammalian cells in flies the transcription factors clock', 0.0),\n",
       " ('member of the evolutionarily conserved cdc2 like kinase', 0.0),\n",
       " ('molecular oscillator consists of two transcription factors clock', 0.0),\n",
       " ('nuclear translocator like proteins 1 bmal1 and clock', 0.0),\n",
       " ('of period per a transcription repressor and clock', 0.0),\n",
       " ('of the genes encoding the transcription activators clock', 0.0),\n",
       " ('of the mammalian clock are the proteins clock', 0.0),\n",
       " ('of the stenotic stk and contra lateral kidney', 0.0),\n",
       " ('of these loop are the transcription factors clock', 0.0),\n",
       " ('on sr proteins srpk and cdc2 like kinase', 0.0),\n",
       " ('on the activators of the transcription factors clock', 0.0),\n",
       " ('or it homologue npas2 in mammals and clock', 0.0),\n",
       " ('oscillator operates in cells that co expression clock', 0.0),\n",
       " ('per and tim and the transcription factors clock', 0.0),\n",
       " ('per and timeless tim proteins which inhibit clock', 0.0),\n",
       " ('per arnt sim bhlh pas transcription factors clock', 0.0),\n",
       " ('per immunoreactivity c representative staining results of clock', 0.0),\n",
       " ('per s942 reduces interactions between per and clock', 0.0),\n",
       " ('positive limb which include the transcription activators clock', 0.0),\n",
       " ('quantified from trichrome staining in sham contra lateral', 0.0),\n",
       " ('rb3 and rb4 are output pin for clock', 0.0),\n",
       " ('regulated of the drosophila clock in drosophila clock', 0.0),\n",
       " ('repress the activators of the transcription activators clock', 0.0),\n",
       " ('repressor that constitute a feedback loop the clock', 0.0),\n",
       " ('score d stenotic kidney stk contra lateral kidney', 0.0),\n",
       " ('sham stenotic stk and non stenotic contra lateral', 0.0),\n",
       " ('show that inhibit of the cdc2 like kinase', 0.0),\n",
       " ('specificity for clk1 over clk3 cdc2 like kinase', 0.0),\n",
       " ('stenotic kidney stk a and contra lateral kidney', 0.0),\n",
       " ('studies have shown that the transcription activators clock', 0.0),\n",
       " ('that are conserved from insects to mammals clock', 0.0),\n",
       " ('the activators is a heterodimeric complex of clock', 0.0),\n",
       " ('the brain of drosophila the transcription factors clock', 0.0),\n",
       " ('the circadian clock although the conserved tfs clock', 0.0),\n",
       " ('the negative elements briefly the transcription activators clock', 0.0),\n",
       " ('the positive arm proteins in drosophila are clock', 0.0),\n",
       " ('the same region in the drosophila clock ortholog', 0.0),\n",
       " ('their product proteins period per timeless tim clock', 0.0),\n",
       " ('tim is regulated by the transcription factors clock', 0.0),\n",
       " ('transcription feedback loop that are activators by clock', 0.0),\n",
       " ('transcription feedback loop whereby the transcription factors clock', 0.0),\n",
       " ('transcription loop are interconnected by the tfs clock', 0.0),\n",
       " ('true clock family member putative calanus finmarchicus clock', 0.0),\n",
       " ('2 clk and cyc encoding the transcription factors clock', 0.0),\n",
       " ('3 in the period timeless per tim loop clock', 0.0),\n",
       " ('40 dbt is also require for phosphorylation of clock', 0.0),\n",
       " ('43 in mammals gsk3 is proposed to phosphorylation clock', 0.0),\n",
       " ('a heterodimer of two bhlh pas transcription factors clock', 0.0),\n",
       " ('a heterodimeric complex of the two transcription factors clock', 0.0),\n",
       " ('a positive limb which include the transcription activators clock', 0.0),\n",
       " ('a regulated of the drosophila clock in drosophila clock', 0.0),\n",
       " ('a transcription feedback loop whereby the transcription factors clock',\n",
       "  0.0),\n",
       " ('a true clock family member putative calanus finmarchicus clock', 0.0),\n",
       " ('an autoregulatory loop in which the transcription factors clock', 0.0),\n",
       " ('an integral set of proteins include period per clock', 0.0),\n",
       " ('and anticipate daily environmental changes in drosophila melanogaster clock',\n",
       "  0.0),\n",
       " ('and behavior in this species putative daphnia pulex clock', 0.0),\n",
       " ('and chung 2010 a heterodimer formed by the clock', 0.0),\n",
       " ('and ck2α respectively left western blots for clk alone', 0.0),\n",
       " ('and repressor that constitute a feedback loop the clock', 0.0),\n",
       " ('and the same region in the drosophila clock ortholog', 0.0),\n",
       " ('and their product proteins period per timeless tim clock', 0.0),\n",
       " ('and tim is regulated by the transcription factors clock', 0.0),\n",
       " ('animal circadian rhythms in drosophila the transcription factors clock',\n",
       "  0.0),\n",
       " ('at per s942 reduces interactions between per and clock', 0.0),\n",
       " ('basic helix loop helix bhlh pas transcription factors clock', 0.0),\n",
       " ('biological studies have shown that the transcription activators clock',\n",
       "  0.0),\n",
       " ('bmal1 or it homologue npas2 in mammals and clock', 0.0),\n",
       " ('by inhibit activators of their transcription activators the clock', 0.0),\n",
       " ('cells is unclear the circadian locomotor output cycle kaput', 0.0),\n",
       " ('center of these loop are the transcription factors clock', 0.0),\n",
       " ('clock itself is comprised of the transcription factors clock', 0.0),\n",
       " ('clockwork by presumably binding to the transcription factors clock', 0.0),\n",
       " ('concanavalin a con a induced product of chemotactic lymphokines', 0.0),\n",
       " ('conserved between flies and mammals in drosophila the clock', 0.0),\n",
       " ('core feedback loop consists of the transcription activators clock', 0.0),\n",
       " ('core molecular oscillator consists of two transcription factors clock',\n",
       "  0.0),\n",
       " ('damage in both the stk and the contralateral kidney', 0.0),\n",
       " ('darkness dd these rhythms require the transcription factors clock', 0.0),\n",
       " ('defined by as cardiosphere derived c kit + lin', 0.0),\n",
       " ('dependent on sr proteins srpk and cdc2 like kinase', 0.0),\n",
       " ('dependent on the activators of the transcription factors clock', 0.0),\n",
       " ('derived l in − c k it + progenitor', 0.0),\n",
       " ('drosophila the activators is a heterodimeric complex of clock', 0.0),\n",
       " ('elements of the mammalian clock are the proteins clock', 0.0),\n",
       " ('expression of the genes encoding the transcription activators clock', 0.0),\n",
       " ('factors bmal1 acts as a dimer with either clock', 0.0),\n",
       " ('feedback loop consists of transcription activators e g clock', 0.0),\n",
       " ('feedback loop of the drosophila clock transcription factors clock', 0.0),\n",
       " ('figure 1 in the per tim loop the clock', 0.0),\n",
       " ('for example 2 17 the circadian transcription factors clock', 0.0),\n",
       " ('formed by a heterodimer of the transcription factors clock', 0.0),\n",
       " ('fruit flies drosophila melanogaster a heterodimer consists of clock', 0.0),\n",
       " ('he et al 2017 in mammals the proteins clock', 0.0),\n",
       " ('helix loop helix bhlh pas containing transcription factors clock', 0.0),\n",
       " ('helix per arnt sim bhlh pas transcription factors clock', 0.0),\n",
       " ('here show that inhibit of the cdc2 like kinase', 0.0),\n",
       " ('hour feedback loop in flies two transcription factors clock', 0.0),\n",
       " ('in drosophila the bhlh pas transcription factors heterodimer clock', 0.0),\n",
       " ('in mammalian cells in flies the transcription factors clock', 0.0),\n",
       " ('in the brain of drosophila the transcription factors clock', 0.0),\n",
       " ('include in flies four major proteins timeless tim clock', 0.0),\n",
       " ('initiated during mid day by the transcription factors clock', 0.0),\n",
       " ('injury score d stenotic kidney stk contra lateral kidney', 0.0),\n",
       " ('interconnected at clk cyc a heterodimer of the clock', 0.0),\n",
       " ('is a transcription feedback loop in drosophila melanogaster clock', 0.0),\n",
       " ('is composed of intracellular feedback loop the proteins clock', 0.0),\n",
       " ('is the circadian clock although the conserved tfs clock', 0.0),\n",
       " ('isoform specificity for clk1 over clk3 cdc2 like kinase', 0.0),\n",
       " ('it analog npas2 and bmal1 in mammals and clock', 0.0),\n",
       " ('loop 1 in drosophila a heterodimer formed by clock', 0.0),\n",
       " ('loop figure 1 26 the heterodimer formed by clock', 0.0),\n",
       " ('loop helix per arnt sim domain transcription factors clock', 0.0),\n",
       " ('loop that are conserved from insects to mammals clock', 0.0),\n",
       " ('loop the positive arm proteins in drosophila are clock', 0.0),\n",
       " ('medulla of the stenotic stk and contra lateral kidney', 0.0),\n",
       " ('melanogaster consists of an interactions feedback loop involves clock',\n",
       "  0.0),\n",
       " ('molecular clock in these clock two transcription factors clock', 0.0),\n",
       " ('mrna and proteins levels in the evening the clock', 0.0),\n",
       " ('neurons clusters mentioned above the circadian transcription factors clock',\n",
       "  0.0),\n",
       " ('of different kinase cyclin dependent kinase cdc2 like kinase', 0.0),\n",
       " ('of genes directly activators by the transcription factors clock', 0.0),\n",
       " ('of sham stenotic stk and non stenotic contra lateral', 0.0),\n",
       " ('of the negative elements briefly the transcription activators clock', 0.0),\n",
       " ('of transcription feedback loop that are activators by clock', 0.0),\n",
       " ('own expression by binding to their transcription activators clock', 0.0),\n",
       " ('own inhibitors in the drosophila circadian oscillator the clock', 0.0),\n",
       " ('patients member of the evolutionarily conserved cdc2 like kinase', 0.0),\n",
       " ('per and tim expression is activators by the clock', 0.0),\n",
       " ('per and timeless tim formed one heterodimer and clock', 0.0),\n",
       " ('per and timeless tim is activators by the clock', 0.0),\n",
       " ('per and timeless tim transcription are activators when clock', 0.0),\n",
       " ('per arnt sim bhlh pas domain transcription factors clock', 0.0),\n",
       " ('per repress the activators of the transcription activators clock', 0.0),\n",
       " ('period d rosophila per d per timeless tim clock', 0.0),\n",
       " ('period per and timeless tim proteins which inhibit clock', 0.0),\n",
       " ('pgs from both the hnk and unobstructed contralateral kidney', 0.0),\n",
       " ('phosphorylation of period per a transcription repressor and clock', 0.0),\n",
       " ('pl bmal1 complex a immunoprecipitation ip of endogenous clock', 0.0),\n",
       " ('polyadenylation factors proteins kinase such as cdc like kinase', 0.0),\n",
       " ('positive and negative with positive factors e g clock', 0.0),\n",
       " ('proteins per and tim and the transcription factors clock', 0.0),\n",
       " ('rb2 rb3 and rb4 are output pin for clock', 0.0),\n",
       " ('receptor nuclear translocator like proteins 1 bmal1 and clock', 0.0),\n",
       " ('repress factors to limit the transcription output of clock', 0.0),\n",
       " ('sehgal 2012 in drosophila period per timeless tim clock', 0.0),\n",
       " ('several transcription loop are interconnected by the tfs clock', 0.0),\n",
       " ('show per immunoreactivity c representative staining results of clock', 0.0),\n",
       " ('srpk1 2 full length clones encoding cdc2 like kinase', 0.0),\n",
       " ('the core of the system are the activators clock', 0.0),\n",
       " ('the drosophila core feedback loop the clock proteins clock', 0.0),\n",
       " ('the evening per and tim proteins then repress clock', 0.0),\n",
       " ('the oscillator operates in cells that co expression clock', 0.0),\n",
       " ('the stenotic kidney stk a and contra lateral kidney', 0.0),\n",
       " ('this clock is composed of the transcription activators clock', 0.0),\n",
       " ('to drosophila clockwork is a key transcription factors clock', 0.0),\n",
       " ('transcription feedback loop that involves the transcription factors clock',\n",
       "  0.0),\n",
       " ('translational feedback loop in drosophila the transcription factors clock',\n",
       "  0.0),\n",
       " ('ttfl in the drosophila ttfl the transcription factors clock', 0.0),\n",
       " ('was analyzed by autoradiography and total clk proteins levels', 0.0),\n",
       " ('were quantified from trichrome staining in sham contra lateral', 0.0),\n",
       " ('which encoding the main transcription repressor of the clock', 0.0),\n",
       " ('with a deleted rs motif nδ21 cdc2 like kinase', 0.0),\n",
       " ('1 2 clk and cyc encoding the transcription factors clock', 0.0),\n",
       " ('1998 he et al 2017 in mammals the proteins clock', 0.0),\n",
       " ('2011 in drosophila the bhlh pas transcription factors heterodimer clock',\n",
       "  0.0),\n",
       " ('24 hour feedback loop in flies two transcription factors clock', 0.0),\n",
       " ('activators and repressor that constitute a feedback loop the clock', 0.0),\n",
       " ('allada and chung 2010 a heterodimer formed by the clock', 0.0),\n",
       " ('and bmal1 or it homologue npas2 in mammals and clock', 0.0),\n",
       " ('and medulla of the stenotic stk and contra lateral kidney', 0.0),\n",
       " ('and polyadenylation factors proteins kinase such as cdc like kinase', 0.0),\n",
       " ('and sehgal 2012 in drosophila period per timeless tim clock', 0.0),\n",
       " ('ardiosphere derived l in − c k it + progenitor', 0.0),\n",
       " ('are dependent on sr proteins srpk and cdc2 like kinase', 0.0),\n",
       " ('as a regulated of the drosophila clock in drosophila clock', 0.0),\n",
       " ('at mrna and proteins levels in the evening the clock', 0.0),\n",
       " ('at the core of the system are the activators clock', 0.0),\n",
       " ('b cells is unclear the circadian locomotor output cycle kaput', 0.0),\n",
       " ('basic helix loop helix bhlh pas containing transcription factors clock',\n",
       "  0.0),\n",
       " ('being a true clock family member putative calanus finmarchicus clock', 0.0),\n",
       " ('both positive and negative with positive factors e g clock', 0.0),\n",
       " ('by a heterodimer of two bhlh pas transcription factors clock', 0.0),\n",
       " ('cancer patients member of the evolutionarily conserved cdc2 like kinase',\n",
       "  0.0),\n",
       " ('central to drosophila clockwork is a key transcription factors clock', 0.0),\n",
       " ('clk and ck2α respectively left western blots for clk alone', 0.0),\n",
       " ('clk pl bmal1 complex a immunoprecipitation ip of endogenous clock', 0.0),\n",
       " ('clock 40 dbt is also require for phosphorylation of clock', 0.0),\n",
       " ('clock dependent on the activators of the transcription factors clock', 0.0),\n",
       " ('clock is composed of intracellular feedback loop the proteins clock', 0.0),\n",
       " ('clock per and tim expression is activators by the clock', 0.0),\n",
       " ('components 43 in mammals gsk3 is proposed to phosphorylation clock', 0.0),\n",
       " ('comprised of transcription feedback loop that are activators by clock',\n",
       "  0.0),\n",
       " ('constant darkness dd these rhythms require the transcription factors clock',\n",
       "  0.0),\n",
       " ('core feedback loop of the drosophila clock transcription factors clock',\n",
       "  0.0),\n",
       " ('counterpart in mammalian cells in flies the transcription factors clock',\n",
       "  0.0),\n",
       " ('cry2 which encoding the main transcription repressor of the clock', 0.0),\n",
       " ('cyc and their product proteins period per timeless tim clock', 0.0),\n",
       " ('d melanogaster consists of an interactions feedback loop involves clock',\n",
       "  0.0),\n",
       " ('drosophila several transcription loop are interconnected by the tfs clock',\n",
       "  0.0),\n",
       " ('drosophila this clock is composed of the transcription activators clock',\n",
       "  0.0),\n",
       " ('exacerbate damage in both the stk and the contralateral kidney', 0.0),\n",
       " ('example is the circadian clock although the conserved tfs clock', 0.0),\n",
       " ('expression by inhibit activators of their transcription activators the clock',\n",
       "  0.0),\n",
       " ('feedback loop 1 in drosophila a heterodimer formed by clock', 0.0),\n",
       " ('feedback loop figure 1 26 the heterodimer formed by clock', 0.0),\n",
       " ('feedback loop that are conserved from insects to mammals clock', 0.0),\n",
       " ('feedback loop the positive arm proteins in drosophila are clock', 0.0),\n",
       " ('fibrosis were quantified from trichrome staining in sham contra lateral',\n",
       "  0.0),\n",
       " ('gflb3 with a deleted rs motif nδ21 cdc2 like kinase', 0.0),\n",
       " ('glcnacylation at per s942 reduces interactions between per and clock', 0.0),\n",
       " ('h of sham stenotic stk and non stenotic contra lateral', 0.0),\n",
       " ('helix loop helix per arnt sim domain transcription factors clock', 0.0),\n",
       " ('helix per arnt sim bhlh pas domain transcription factors clock', 0.0),\n",
       " ('highly conserved between flies and mammals in drosophila the clock', 0.0),\n",
       " ('hydrocarbon receptor nuclear translocator like proteins 1 bmal1 and clock',\n",
       "  0.0),\n",
       " ('in the drosophila core feedback loop the clock proteins clock', 0.0),\n",
       " ('in the evening per and tim proteins then repress clock', 0.0),\n",
       " ('in the stenotic kidney stk a and contra lateral kidney', 0.0),\n",
       " ('include period d rosophila per d per timeless tim clock', 0.0),\n",
       " ('interactions and the same region in the drosophila clock ortholog', 0.0),\n",
       " ('intracellular molecular clock in these clock two transcription factors clock',\n",
       "  0.0),\n",
       " ('involves a positive limb which include the transcription activators clock',\n",
       "  0.0),\n",
       " ('is formed by a heterodimer of the transcription factors clock', 0.0),\n",
       " ('is initiated during mid day by the transcription factors clock', 0.0),\n",
       " ('levels of genes directly activators by the transcription factors clock',\n",
       "  0.0),\n",
       " ('loop 3 in the period timeless per tim loop clock', 0.0),\n",
       " ('loop helix per arnt sim bhlh pas transcription factors clock', 0.0),\n",
       " ('loop ttfl in the drosophila ttfl the transcription factors clock', 0.0),\n",
       " ('mammals the oscillator operates in cells that co expression clock', 0.0),\n",
       " ('molecular biological studies have shown that the transcription activators clock',\n",
       "  0.0),\n",
       " ('negative transcription feedback loop that involves the transcription factors clock',\n",
       "  0.0),\n",
       " ('neurons in the brain of drosophila the transcription factors clock', 0.0),\n",
       " ('neurons show per immunoreactivity c representative staining results of clock',\n",
       "  0.0),\n",
       " ('nucleus per repress the activators of the transcription activators clock',\n",
       "  0.0),\n",
       " ('number of different kinase cyclin dependent kinase cdc2 like kinase', 0.0),\n",
       " ('of a heterodimeric complex of the two transcription factors clock', 0.0),\n",
       " ('of a transcription feedback loop whereby the transcription factors clock',\n",
       "  0.0),\n",
       " ('of an autoregulatory loop in which the transcription factors clock', 0.0),\n",
       " ('of an integral set of proteins include period per clock', 0.0),\n",
       " ('of pgs from both the hnk and unobstructed contralateral kidney', 0.0),\n",
       " ...]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "top"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "longforms = miners['CLK'].get_longforms(abs_decay_param=0.001)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('clock', 106, 0.9246367609300921),\n",
       " ('cdc2 like kinase', 7, 0.9975062395963412),\n",
       " ('contra lateral kidney', 3, 0.9981129987413702),\n",
       " ('contralateral kidney', 3, 0.8827786327110382),\n",
       " ('contra lateral', 2, 0.7457923787155183),\n",
       " ('clk proteins levels', 2, 0.637207331456189),\n",
       " ('cdc like kinase', 1, 0.9990628514746259),\n",
       " ('cycle kaput', 1, 0.9434263027606444),\n",
       " ('chemotactic lymphokines', 1, 0.8842122898223528),\n",
       " ('c kit + lin', 1, 0.846736156908713),\n",
       " ('clk alone', 1, 0.7494061755245505),\n",
       " ('clock ortholog', 1, 0.7086398896955698)]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "longforms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('clock', 106, 0.9246367609300921),\n",
       " ('cdc2 like kinase', 7, 0.9975062395963412),\n",
       " ('contra lateral kidney', 3, 0.9981129987413702),\n",
       " ('contralateral kidney', 3, 0.8827786327110382),\n",
       " ('contra lateral', 2, 0.7457923787155183),\n",
       " ('clk proteins levels', 2, 0.637207331456189),\n",
       " ('cdc like kinase', 1, 0.9990628514746259),\n",
       " ('cycle kaput', 1, 0.9434263027606444),\n",
       " ('chemotactic lymphokines', 1, 0.8842122898223528),\n",
       " ('c kit + lin', 1, 0.846736156908713),\n",
       " ('clk alone', 1, 0.7494061755245505),\n",
       " ('clock ortholog', 1, 0.7086398896955698)]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "longforms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "from adeft.ground import AdeftGrounder\n",
    "grounder = AdeftGrounder()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [],
   "source": [
    "longforms0 = [x for i, x in enumerate(longforms0) if i in [0, 1]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('clock', 86.3921568627451), ('cdc2 like kinase', 6.0)]"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "longforms0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [],
   "source": [
    "longforms0.extend([(longform, score) for i, (longform, score) in enumerate(top)\n",
    "                   if i in [13]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('clock', 86.3921568627451),\n",
       " ('cdc2 like kinase', 6.0),\n",
       " ('contralateral kidney', 2.0)]"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "longforms0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [],
   "source": [
    "longforms0.sort(key=lambda x: -x[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [],
   "source": [
    "longforms, scores = zip(*longforms0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('clock', 'cdc2 like kinase', 'contralateral kidney')"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "longforms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [],
   "source": [
    "grounding_map = {}\n",
    "names = {}\n",
    "for longform in longforms:\n",
    "    grounding = gilda_ground(longform)\n",
    "    if grounding[0]:\n",
    "        grounding_map[longform] = f'{grounding[0]}:{grounding[1]}'\n",
    "        names[grounding_map[longform]] = grounding[2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'clock': 'HGNC:2082'}"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "grounding_map"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'HGNC:2082': 'CLOCK'}"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [],
   "source": [
    "grounding_map, names, pos_labels = ground_with_gui(longforms, scores, grounding_map=grounding_map, names=names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'FPLX:CLK': 'CLK', 'HGNC:2082': 'CLOCK'}"
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'cdc2 like kinase': 'FPLX:CLK',\n",
       " 'clock': 'HGNC:2082',\n",
       " 'contralateral kidney': 'ungrounded'}"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "grounding_map"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "result = grounding_map, names, pos_labels"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "({'cdc2 like kinase': 'FPLX:CLK',\n",
       "  'clock': 'HGNC:2082',\n",
       "  'contralateral kidney': 'ungrounded'},\n",
       " {'FPLX:CLK': 'CLK', 'HGNC:2082': 'CLOCK'},\n",
       " ['FPLX:CLK', 'HGNC:2082'])"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": [
    "grounding_map, names, pos_labels = ({'cdc2 like kinase': 'FPLX:CLK',\n",
    "  'clock': 'HGNC:2082',\n",
    "  'contralateral kidney': 'ungrounded'},\n",
    " {'FPLX:CLK': 'CLK', 'HGNC:2082': 'CLOCK'},\n",
    " ['FPLX:CLK', 'HGNC:2082'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [],
   "source": [
    "grounding_dict = {'CLK': grounding_map}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [],
   "source": [
    "classifier = AdeftClassifier('CLK', pos_labels=pos_labels)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [],
   "source": [
    "param_grid = {'C': [100.0], 'max_features': [10000]}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [],
   "source": [
    "labeler = AdeftLabeler(grounding_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [],
   "source": [
    "corpus = labeler.build_from_texts(shortform_texts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [],
   "source": [
    "corpus.extend(entrez_texts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [],
   "source": [
    "texts, labels = zip(*corpus)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO: [2020-01-17 14:31:29] /Users/albertsteppi/adeft/adeft/modeling/classify.py - Beginning grid search in parameter space:\n",
      "{'C': [100.0], 'max_features': [10000]}\n",
      "/Users/albertsteppi/.virtualenvs/py37/lib/python3.7/site-packages/sklearn/model_selection/_split.py:657: Warning: The least populated class in y has only 3 members, which is too few. The minimum number of members in any class cannot be less than n_splits=5.\n",
      "  % (min_groups, self.n_splits)), Warning)\n",
      "INFO: [2020-01-17 14:31:34] /Users/albertsteppi/adeft/adeft/modeling/classify.py - Best f1 score of 0.9507457227323768 found for parameter values:\n",
      "{'logit__C': 100.0, 'tfidf__max_features': 10000}\n"
     ]
    }
   ],
   "source": [
    "classifier.cv(texts, labels, param_grid, cv=5, n_jobs=8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'label_distribution': {'ungrounded': 3, 'HGNC:2082': 295, 'FPLX:CLK': 49},\n",
       " 'f1': {'mean': 0.9507457227323768, 'std': 0.029135992296673168},\n",
       " 'precision': {'mean': 0.947456159148541, 'std': 0.03093001570141328},\n",
       " 'recall': {'mean': 0.9564791133844842, 'std': 0.02587786533592789}}"
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "classifier.stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [],
   "source": [
    "disamb = AdeftDisambiguator(classifier, grounding_dict, names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [],
   "source": [
    "d = disamb.disambiguate(shortform_texts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [],
   "source": [
    "a = [(text, pred) for pred, text in zip(d, shortform_texts)if pred[0].startswith('ung')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3"
      ]
     },
     "execution_count": 128,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(a)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('Unilateral ureteral obstruction (UUO) results in increased renal resistance as well as in exaggerated prostaglandin (PG) release from the obstructed hydronephrotic kidney (HNK). We have reported previously that platelet-activating factor (PAF) dose-dependently stimulates the release of PGs from both the HNK and unobstructed contralateral kidney (CLK), with CLK release being 10% that of the HNK. In the present report, we studied the interaction of PAF with its receptor by examining the effects of PAF-receptor antagonists on the release of PGs from the isolated perfused rabbit HNK and CLK stimulated by PAF; angiotensin II (AII), and bradykinin (BK) were also used as agonists. In the HNK, kadsurenone (3 microM) inhibited PAF-stimulated PGE2 and thromboxane B2 (TxB2) release by 28.2 and 62.5% respectively. CV-3988 (20 microM) and triazolam (5 microM) also preferentially diminished PAF-stimulated TxB2 release. In addition, all three drugs significantly diminished BK- and AII-stimulated TxB2 release, while CV-3988 was the only antagonist to affect peptide-stimulated PGE2 release. While effective against agonist-stimulated PG synthesis, these drugs had no direct effect on arachidonic acid metabolism to PGs. Furthermore, in the CLK, CV-3988 had no effect on BK- or AII-stimulated PGE2 release, whereas it totally inhibited PAF-stimulated release of PGE2. These results show that PAF-receptor antagonists in the HNK preferentially inhibit TxB2 release whether stimulated by PAF, AII or BK; in the CLK only PAF-stimulated PG release is affected. This biochemical difference may be of physiological significance and explain some of the functional differences between the HNK and CLK. Therefore, PAF may be an important mediator of some of the biochemical and functional changes associated with UUO.\\n',\n",
       "  ('ungrounded',\n",
       "   None,\n",
       "   {'FPLX:CLK': 0.0, 'ungrounded': 1.0, 'HGNC:2082': 0.0})),\n",
       " ('Domestic pigs divided into three groups (n=6–7 each) were studied for 16 weeks. Unilateral RAS was induced in two groups, of which one was fed with a high cholesterol diet to induce early atherosclerosis (ARAS), while the other (RAS) consumed regular diet. The third group included normal pigs which served as control sham. Renal function, hemodynamics, tubular function and morphology were assessed using multi-detector computed tomography (MDCT) and histology. Diffusion-weighted MR images were acquired at 3T and analyzed using mono and bi-exponential models. ADC and IVIM parameters (Diffusivity (D t ), flow-dependent pseudo-diffusivity (D p ), and fluid fraction (f p )) were calculated in cortex and medulla of the stenotic (STK) and contra-lateral kidneys (CLK). Results were analyzed using ANOVA, student t-test, and regression analysis.\\nIn both RAS and ARAS the STK shrank and the CLK underwent hypertrophy. GFR and renal blood flow declined in STKs, and CLKs manifested hyperfiltration. Additionally, ARAS kidneys showed reduced mean transit time in distal tubular segments. ADC and diffusivity both decreased in STK of RAS and ARAS. D p  and f p  were elevated in both the STK and CLK of RAS and more prominently in ARAS. STK cortical ADC and D t  correlated inversely with the degree of fibrosis, and directly with GFR. Furthermore, D p  correlated with tubular injury score in all kidneys.\\nRenal artery stenosis (RAS) is a vascular disorder, which may result in chronic kidney disease (CKD) and lead to end stage renal failure 1 . Significant RAS compromises renal blood flow (RBF), induces renovascular hypertension, impairs glomerular filtration rate (GFR), and induces interstitial fibrosis in the post-stenotic kidney (STK). While in the Western world atherosclerotic RAS (ARAS) is the predominant etiology in adults, RAS may be caused by other causes like fibromuscular dysplasia 1 . Renal injury in ARAS is more severe because atherogenic factors exacerbate damage in both the STK and the contralateral kidney (CLK) 2 , 3 . Early diagnosis and effective treatment may potentially slow further deterioration of renal function. Different imaging techniques and modalities have been used to detect and investigate mechanisms of renal functional and hemodynamic deterioration, yet assessment of tissue injury and morphological changes in the kidney is difficult without invasive methods 4 – 7 .\\nWe hypothesize that functional and morphological alterations in the swine STK and CLK, illustrated by multi-detector computed tomography (MDCT) and histology, would be accompanied by corresponding changes in DWI-derived diffusivity and pseudo-diffusivity calculated using the IVIM model. We further hypothesize that DWI-derived parameters would depict greater renal impairment in pigs with ARAS compared to RAS.\\nResults are presented as mean±SEM. ANOVA was used for STK/CLK comparison and among groups, followed by unpaired t-test. Regression analysis was performed using the least square method, in some cases on a compiled set of corresponding kidneys from all three groups of RAS, ARAS and Sham. For p values ≤ 0.05, results were considered significant.\\nTen weeks after RAS induction the degree of stenosis in the RAS and ARAS groups were comparable and MAP was similarly elevated ( Table. 1 ). The cortical and medullary regions of the STKs underwent atrophy and a decrease in RBF, whereas in the CLKs, the cortex showed hypertrophy, and medullary volumes remained unchanged. MDCT-derived GFR was reduced in the STKs and elevated in the CLKs compared to the sham group. Cortical vascular and proximal tubular MTT values were similar among all groups, whereas distal MTT was significantly shorter in the ARAS STK and CLK, and tended to be shorter in RAS as well (p=0.063 and p=0.072 in the STK and CLK, respectively). Cortical perfusion and FBV remained unchanged, while medullary perfusion and FBV declined in all STKs.\\nThe degree of fibrosis was significantly higher in the cortex and medulla of both the STK and CLK of ARAS compared to those in RAS or shams, suggesting greater tissue injury in ARAS. Interstitial fibrosis (trichrome-staining) in the cortex of RAS STK was also greater than Sham ( Figure 2.A–B ). Medullary fibrosis was elevated compared to sham in all RAS and ARAS kidneys, but again in ARAS more than in RAS. Tubular injury score was significantly higher in the RAS STK and in both ARAS kidneys compared to Sham ( Figure 2.C–D ).\\nRAS STK cortical diffusivity was significantly lower than that in its CLK or in sham, while the RAS CLK and Sham were similar. In contrast, ARAS STK diffusivity was not different than its CLK, yet was reduced compared to the Sham, and tended to decrease in the CLK as well (p= 0.084 vs. sham) ( Figure 3 ). Similarly, in RAS f p  was higher than Sham in the STK and tended to be elevated in the CLK (p=0.060), whereas in ARAS, both were significantly higher compared to sham Control. Finally, cortical pseudo-diffusivity (flow velocity) was elevated in STK and CLK of both RAS and ARAS, except that in the RAS STK the increase has not reached statistical significance (p=0.067).\\nA similar pattern was observed in the medulla, where diffusivity tended to decline in the RAS STK (p=0.091) and in both ARAS STK and CLK (p=0.072 and p=0.095 in RAS and ARAS, respectively and p=0.020 in RAS CLK vs. ARAS CLK) while pseudo-diffusivity was significantly elevated in all STKs and CLKs, except in RAS STK (p=0.058 vs. Sham). f p  was only increased in both ARAS but not RAS kidneys.\\nIn both the cortex and medulla, ADC values were significantly lower in the STK compared to the CLK and Sham, whereas CLK ADC were not different than Sham ( Figure 3.d & 3.h ).\\nCortical and medullary diffusivity and ADC showed strong correlation in all kidneys ( Figure 4 ). In the STK, cortical and medullary diffusivity and ADC, all correlated inversely with the histological degree of fibrosis and in the cortex also with GFR ( Figure 5.A–B ). Pseudo-diffusion in the cortex of all kidneys correlated inversely with MDCT-derived distal tubular MTT ( Figure 5.C ). Furthermore, STK (but not CLK) cortical f p  showed a modest but significant correlation with MDCT-derived FBV, and inversely correlated with the degree of fibrosis. Tubular injury score correlated with cortical pseudo-diffusivity ( Figure 5.D ).\\nThis study shows that IVIM-based analysis of DWI can detect and distinguish functional and morphological changes in atherosclerotic (ARAS) and non-atherosclerotic RAS kidneys, which agree with comparable assessments, obtained using independent methods. Diffusivity decreased in the STKs and inversely correlated to the histological degree of fibrosis. Fluid fraction and pseudo-diffusivity increased in the RAS STK cortex, and remarkably in both ARAS STK and CLK, suggesting tubular functional and structural alterations, as confirmed by MDCT and histology. Therefore, this study suggests that DWI parameters may serve as biomarkers of kidney injury.\\nThe results of this study indicated that IVIM is capable of unraveling renal morphological and fluid dynamic alterations in RAS and ARAS subjects. Ten weeks after induction of stenosis, we observed hallmarks of RAS, including a decline in STK volume RBF, and GFR, and development of systemic hypertension. Because of the relatively moderate degree of stenosis, the main changes were related to renal volume, so that STK atrophy and compensatory CLK hypertrophy largely preserved regional GFR, FBV, and perfusion 25 . Nevertheless, cortical diffusivity fell in both RAS and ARAS STK and tended to decrease in their medulla as well. Indeed, diffusivity correlated with the level of fibrosis, which was elevated in the RAS STK and in both ARAS kidneys. Smaller diffusivity of ARAS vs. RAS medullary CLK (which was not different than the ARAS STK) was likely attributed to greater fibrosis and structural alterations found in ARAS CLK by histology. Notably, this was consistent with our previous finding that the morphology of the RAS CLK remains largely preserved, whereas in ARAS the additive systemic effects of atherosclerosis and hypertension adversely impact CLK tissue 2 , 3 . Cortical diffusivity also correlated with GFR in the STK. Lack of such correlation in the CLK is particularly interesting, considering that CLK GFR was significantly increased to compensate for the reduced STK GFR. This may imply that the increase in CLK GFR is driven mostly by renal functional rather than structural changes, whereas in the STK, both morphological (e.g. fibrosis, glomerulosclerosis) and hemodynamic factors contribute to the diminished GFR.\\nIn addition to diffusivity, which is often attributed to morphological changes in the tissue 26 , 27 , IVIM analysis yields D p  and f p . D p  has been shown to be sensitive to both vascular blood and tubular fluid velocities 17 , 28 , while f p  is the ratio of vascular and tubular fluid volume to the total fluid in the tissue 19 . Our study suggests that cortical and medullary fluid velocities increased in both ARAS kidneys and in RAS CLK, and tended to rise in the RAS STK. The use of MDCT allowed attributing the increased D p  to tubular flow velocity. Notably, in our model with moderate stenosis and sustained STK perfusion, MDCT showed that vascular MTT remained unchanged, whereas in ARAS kidneys shorter distal tubular MTT implied higher flow velocity in this tubular segment in ARAS kidneys. Remarkably, DWI MRI-derived D p  correlated significantly and inversely with distal tubular MTT, but not with vascular MTT, suggesting that in this study, increases in D p  reflects primarily faster tubular flow variations. In support of this stipulation, D p  also directly correlated with the histological tubular injury score, which established tubular morphological alterations. Some tubular injuries such as cell detachment and intratubular cast formation may affect tubular fluid dynamic by increasing tubular resistance and fluid velocity, yet reducing overall tubular flow and urine volume 29 . Moreover, this speculation is concordant with findings in ischemic tubular injury which shows collecting ducts and particularly distal tubules are the primary cites of obstruction by tubular debris, cells and polymerized Tamm-Horsfall protein 30 . On the other hand, the increase in f p  might have resulted from extracellular matrix accumulation in the atrophic STK 2  that decreases interstitial fluid to a greater extent than tubular fluid volume. This is suggested by the inverse correlation of STK cortical fibrosis and f p .\\nHigher f p  and D p  in the CLK, which is subjected to increased filtration and pressure natriuresis, are concordant with purported increases in healthy subjects during diuresis 19 , as a result of elevated GFR and tubular flow. In our study, CLK hyperfiltration due to the increased arterial pressure also serves as a compensatory mechanism that increases CLK GFR but not the number of nephrons. Hence, an increase in f p  may reflect an increase in tubular flow in response to the CLK fluid overload. These observations suggest that an increase of f p  alone may not distinguish functional from structural underlying causes.\\nTherefore, this study demonstrates that IVIM analysis affords information beyond that provided by ADC, and allows detecting discernible renal alterations in RAS and ARAS. Diffusivity was linked to morphological changes, which were likely driven mainly by fibrosis. D p  might depend on tubular injury in the STK, and on increased tubular flow in the CLK. Similarly, f p  was sensitive to morphological changes in the STK and functional alterations in the CLK. Although currently incapable of differentiating the contribution of the vascular from tubular sub-compartments to f p  and D p , our study implies that IVIM can provide useful markers of renal tubular function. Conceivably a combination of arterial spin labeling and DWI scans may help distinguish these two functional aspects, based on different times-of-flight, and thereby facilitate their interpretation and provide more specific markers for tubular function and hemodynamics. Notably, we also found that morphological and functional alterations in the STK alone could be adequately depicted using ADC. Yet as previous studies have shown, b-value-dependency and inconsistent selection of b-values have complicated derivation of reliable and reproducible values of ADC 14 . Furthermore, the mono-exponential model cannot differentiate the contribution of functional from morphological alterations or distinguish subtle differences in injury severity (ARAS from RAS). Nevertheless, the relative simplicity of use may make analysis of ADC attractive for assessment of renal injury, when considerable in extent.\\nIn conclusion, this study shows that IVIM parameters are sensitive to fibrosis (e.g. mild degree of fibrosis in the CLK) and alterations in tubular flow as a result of functional or structural tubular alterations in unilateral RAS and ARAS. DWI is a powerful method to assess selective pathological markers in the kidney. Independence from exogenous contrast-agents makes this imaging technique appealing to use for patients with chronic kidney disease. Further studies are needed to determine the utility of these techniques in patients with ARAS.\\nTissue Morphology. Cortical (left) and medullary (right) fibrosis were quantified from trichrome staining in Sham, contra-lateral (CLK) and stenotic (STK) kidneys in RAS and ARAS (A), as well as representative 20× images in Sham (a), STK and CLK of RAS (b,d) and ARAS (c,e), respectively (B). Semi-quantitative tubular injury scores (C) and hematoxylin-eosin images (40×) in Sham, RAS and ARAS (D) (*p<0.05 vs. Sham; †p<0.05 vs. CLK; $p<0.05 RAS vs. ARAS). Cell detachment and cast formation are visible in RAS and ARAS tubules (arrows).\\nDWI-derived ADC and IVIM parameters. In the cortex (a–d) and medulla (e–h) of Sham, stenotic (STK) and non-stenotic contra-lateral (CLK) kidneys of RAS and ARAS pigs. (*p<0.05 vs. Sham; †p<0.05 vs. CLK, correspondingly).\\nAnalyses of regression. Renal interstitial fibrosis (A). Cortical (a,b) and medullary (c,d) fibrosis levels correlated with diffusivity and ADC in corresponding compartments. The two markers also showed significant correlations with GFR (B). Furthermore, perfusion-dependent IVIM parameters showed correlation with renal functional parameters (C). Cortical pseudo-diffusivity correlated with distal tubular MTT in the stenotic kidney (STK) (a) and contra-lateral kidney (CLK) (b). Cortical fluid fraction in the STK showed a moderate but significant correlation with blood volume (c). These two parameters did not correlate in CLK (d). A significant correlation was observed between the pseudo-diffusivity and tubular injury score (D).\\nStenotic Kidney (STK), Contra-lateral Kidney (CLK)\\n',\n",
       "  ('ungrounded',\n",
       "   None,\n",
       "   {'FPLX:CLK': 0.0, 'ungrounded': 1.0, 'HGNC:2082': 0.0})),\n",
       " ('Impaired contralateral kidney (CLK) function is important in the maintenance of hypertension in the two-kidney, one-clip (2K, 1C) Goldblatt rat model. Since glomerular filtration rate (GFR) is influenced by the products of arachidonic acid metabolism, we investigated the potential role of eicosanoids as mediators of impaired CLK pressure-volume regulation. At 4 wk following right renal artery clipping, GFR of hypertensive rats was significantly reduced. This decrease was due to the fixed reduction in GFR of the clipped kidney and failure of the CLK to increase its GFR. Thromboxane (Tx) production by isolated perfused CLK was significantly elevated, whereas prostacyclin production remained unchanged. Furthermore, CLK GFR was inversely proportional to Tx production. Treatment of 4-wk hypertensive animals with either the Tx synthase inhibitor UK-38,485 or the Tx receptor antagonist GR 32191 produced a significant increase in CLK GFR. In addition, treatment with either the Tx synthase inhibitor or the Tx receptor antagonist significantly reduced systemic blood pressure. Thus, in this 2K, 1C model of hypertension, increased renal Tx production prevents functional hypertrophy of the contralateral kidney. As a result, CLK pressure-volume regulation is impaired and systemic hypertension is maintained. Furthermore, Tx antagonists restore CLK function and acutely lower systemic blood pressure. Therefore, increased renal Tx production by the CLK appears to be an important mediator of hypertension in the 2K, 1C model.\\n',\n",
       "  ('ungrounded',\n",
       "   None,\n",
       "   {'FPLX:CLK': 0.0, 'ungrounded': 1.0, 'HGNC:2082': 0.0}))]"
      ]
     },
     "execution_count": 129,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [],
   "source": [
    "disamb.dump('CLK', '../results')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from adeft.disambiguate import load_disambiguator"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "d = load_disambiguator('ARG', '../results')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "d.disambiguate(texts[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(d.info())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "a = load_disambiguator('AR')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "a.disambiguate('Androgen')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "logit = d.classifier.estimator.named_steps['logit']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "logit.classes_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "model_to_s3(disamb)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
